---
title: "Transcan"
author: "Laura Villain, Rodolphe Thiébaut, Thomas Ferte, Boris Hejblum"
output: 
  bookdown::html_document2:
    toc: yes
    toc_float: true
    number_sections: FALSE
    fig_caption: TRUE
    table_caption: TRUE
---

 
```{r include=FALSE}
 knitr::opts_chunk$set(cache=TRUE)

```
 

```{r setup, include=FALSE, eval=TRUE,message=FALSE,warning=FALSE,cache=FALSE}

library(knitr)
library(readxl)
library(survival)
library("survminer")
library(randomForestSRC)
library(dearseq)
library(limma)
# library(xlsx)
library(GSA)
library(verification)
library(dynpred)
library(pec)
library(DESeq2)
library(base)
library(reshape2)
library(ggplot2)
library(stats)
library(patchwork)
library(blandr)
library(kableExtra)
library(dplyr)
library(FactoMineR)
library(scales)
library(ggrepel)
library(viridis)

```
# Transcan project
The Glioma are the most common primary brain tumors, among them are the Lower Grade Glioma (LLG), that are characterized by a less aggressive biological behaviour. Their incidence in the European population is increasing, and they mainly occur in the younger population, with an invariable progression to a more malignant phenotype. The Transcan project aims to predict its progression based on imaging (PET-scan and MRI), but also to identify predictive signature liked to the patient risk of recurrence. There are two types of tumors identified by the PET-scan, "cold" patients, with a tumor showing little intensity in the image, and "diffuse" patients when the tumor have an uniform intensity. We are also interested in identifying the difference between those two groups at the gene level, and to link this difference to the recurrence risk. 


# Transcan data
A total of 163 patients with lower grade glioma were followed for this study. Among them, we have RNAseq data on 56 patients, proteomics data on 98 patients and imaging data on 158 patients. The following figure shows the flowchart of the total data with the cold and diffuse patients :
![Flowchart of Transcan data](data/Transcan_DiffCold_FlowChart.png)
![Flowchart of Transcan data](data/Transcan_AstroOligo_FlowChart.png)
However, 5 patients identified as GBM-IDH1 mutated could change the results, with 4 patients with RNA measures and 4 patients with proteomics measures. All of them were characterized as "astro", but 2 were "cold", 3 "diffuse". For that, we compare the different results, with or without them 
```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE}


  
data_total <- read_excel("data/13_Transcan_All_survival.xlsx",sheet = "All_survie")
data_total=as.data.frame(data_total)
data_total=data_total[,c(4,5,16,18,10,11,6,8,9,3)]
data_total=data_total[-1,]
data_total=data_total[!is.na(data_total[,2]),]
data_total=data_total[!data_total[,2]=="SPOTS",]
data_total=data_total[!data_total[,2]=="SPOT",]

colnames(data_total)=c("ID","cold_diffuse","Reccurence","Time","Age","Sexe","oligo_astro","RNA","proteo","ID_proteo")
data_total=data_total[,c(1,2,3,4,6,5,7,8,9,10)]
for (i in 1:length(data_total[,1])){
  
  if (data_total[i,2]=="diffuse"){
    data_total[i,2]="DIFFUSE"
  }
    if (data_total[i,2]=="cold"){
    data_total[i,2]="COLD"
    }
  if (data_total[i,7]=="."){
    data_total[i,7]=NA
  }else{
    
  
  if (data_total[i,7]=="1"){
    data_total[i,7]="oligo"
  }
    if (data_total[i,7]=="0"){
    data_total[i,7]="astro"
  }}
}

data_total=data_total[!duplicated(data_total[,1]),]


data_total[,6]=as.numeric(data_total[,6])
data_total[,3]=as.numeric(data_total[,3])
data_total[,4]=as.numeric(data_total[,4])/365
data_total[,2]=as.factor(data_total[,2])
data_total[,5]=as.factor(data_total[,5])
data_total[,7]=as.factor(data_total[,7])
data_total[,8]=as.factor(data_total[,8])
data_total[,9]=as.factor(data_total[,9])


oligo=c(as.data.frame( read_excel("data/Transcan_Liste_Astro_Oligo.xlsx",sheet = "Oligo",col_names = FALSE)))$"...1"
astro=as.data.frame( read_excel("data/Transcan_Liste_Astro_Oligo.xlsx",sheet = "Astro",col_names = FALSE))
G4=astro[astro[,2]=="G4",]
G4=ifelse(data_total[,1]%in%G4[,1],"1","0")

data_total=cbind(data_total,G4)

data_total[,11]=as.factor(data_total[,11])
data_total_avec=data_total
data_total_sans=data_total[data_total[,11]=="0",]

cold_oligo=numeric(length(data_total[,1]))
for (i in 1:length(data_total[,1])){
  if (!is.na(data_total[i,7])){
    
  
  if (data_total[i,2]=="COLD"&data_total[i,7]=="oligo"){
    cold_oligo[i]=1
  }
    if (data_total[i,2]=="DIFFUSE"&data_total[i,7]=="oligo"){
    cold_oligo[i]=2
    }
    if (data_total[i,2]=="COLD"&data_total[i,7]=="astro"){
    cold_oligo[i]=3
    }
    if (data_total[i,2]=="DIFFUSE"&data_total[i,7]=="astro"){
    cold_oligo[i]=4
  }}
}

```

We can look at a Cox model for the base variables:

##  {- .tabset .tabset-pills .unlisted .unnumbered}
### With G4 patients 
```{r, message=FALSE,warning=FALSE}
modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=data_total_sans)
summary(modele_base)

```

With the G4 patients, the cold diffuse status is significant, not the other variables.

### Without G4 patients 
```{r, message=FALSE,warning=FALSE}
modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=data_total_avec)
summary(modele_base)

```

Without the G4 patients, the cold diffuse status is still significant, not the other variables.


## {- .tabset .tabset-pills .unlisted .unnumbered}
We can look at the link between oligo/astro and cold/diffuse: 

### With G4 patients
```{r,echo=F,message=FALSE,warning=FALSE}
data_total=data_total_avec
Oligo_test=c(dim(data_total[data_total[,2]=="COLD"&data_total[,7]=="oligo",])[1],dim(data_total[data_total[,2]=="DIFFUSE"&data_total[,7]=="oligo",])[1])
Astro_test=c(dim(data_total[data_total[,2]=="COLD"&data_total[,7]=="astro",])[1],dim(data_total[data_total[,2]=="DIFFUSE"&data_total[,7]=="astro",])[1])
table_comparaison=cbind(Oligo_test,Astro_test)
colnames(table_comparaison)=c("Oligo","Astro")
rownames(table_comparaison)=c("Cold","Diffuse")

 kable_classic(kbl(x=table_comparaison,caption = "Cold vs Diffuse and Astro vs Oligo"),full_width = F, html_font = "Cambria")
 


```

### Without G4 patients
```{r,echo=F,message=FALSE,warning=FALSE}
data_total=data_total_sans

Oligo_test=c(dim(data_total[data_total[,2]=="COLD"&data_total[,7]=="oligo",])[1],dim(data_total[data_total[,2]=="DIFFUSE"&data_total[,7]=="oligo",])[1])
Astro_test=c(dim(data_total[data_total[,2]=="COLD"&data_total[,7]=="astro",])[1],dim(data_total[data_total[,2]=="DIFFUSE"&data_total[,7]=="astro",])[1])
table_comparaison=cbind(Oligo_test,Astro_test)
colnames(table_comparaison)=c("Oligo","Astro")
rownames(table_comparaison)=c("Cold","Diffuse")

 kable_classic(kbl(x=table_comparaison,caption = "Cold vs Diffuse and Astro vs Oligo"),full_width = F, html_font = "Cambria")

```

The Cold patients are more often also Astro, while the Diffuse patients are more often also Oligo. The chi test is significant (pvalue=  `r round(chisq.test(table_comparaison)[3]$p.value,6)`)


## Cold vs Diffuse {.tabset .tabset-pills}

We have a total of 68 cold patients (2 of them being G4) and 95 diffuse patients (three of them being G4). The following table shows the different characteristics of these patients. 

### With G4
```{r, eval=T,echo=FALSE,message=FALSE,warning=FALSE}
data_total=data_total_avec
Age=c(round(mean(data_total[data_total[,2]=="COLD"&!is.na(data_total[,6]),6]),1),round(mean(data_total[data_total[,2]=="DIFFUSE",6]),1))
Female=c(table(data_total[data_total[,2]=="COLD",5])[1],table(data_total[data_total[,2]=="DIFFUSE",5])[1])
Male=c(table(data_total[data_total[,2]=="COLD",5])[2],table(data_total[data_total[,2]=="DIFFUSE",5])[2])
Event=c(table(data_total[data_total[,2]=="COLD",3])[2],table(data_total[data_total[,2]=="DIFFUSE",3])[2])
No_event=c(table(data_total[data_total[,2]=="COLD",3])[1],table(data_total[data_total[,2]=="DIFFUSE",3])[1])
Median_followup=c(round(summary(data_total[data_total[,2]=="COLD",4])[3],1),round(summary(data_total[data_total[,2]=="DIFFUSE",4])[3],1))
cold_diffuse_table=rbind(Age,Female,Male,Event,No_event,Median_followup)
colnames(cold_diffuse_table)=c("Cold (n=68)","Diffuse (n=95)")
rownames(cold_diffuse_table)=c("Age (mean)","Female","Male","Event","No event", "Median follow up (years)")
```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=cold_diffuse_table,caption = "Cold vs Diffuse patients"),full_width = F, html_font = "Cambria")

```

We can look at the Kaplan Meier curves for cold vs diffuse: 

```{r,echo=F,message=FALSE,warning=FALSE, fig.cap="Kaplan Meier curves, cold vs diffuse, with G4 patients"}
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total_avec)


p_cd_total= ggsurvplot(fit, data = data_total_avec,
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Cold vs Diffuse",
            legend.labs = c("Cold", "Diffuse"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            palette = c("#E7B800", "#2E9FDF"),
            
            title="C vs D, Total",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
p_cd_total


data_test=cbind(data_total,cold_oligo)
data_test[,12]=factor(data_test[,12])
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_oligo, data = data_test)


p_interaction= ggsurvplot(fit, data = data_test,
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Cold vs Diffuse",
            legend.labs = c("Cold and oligo","Diffuse and oligo","Cold and astro","Diffuse and astro"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
           # palette = c("#E7B800", "#2E9FDF"),
            
            title="Interactions, Total",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
#p_interaction
 
```
The curves seems to overlap, with no clear differences between the cold and diffuse 

### Without G4 
```{r, eval=T,echo=FALSE,message=FALSE,warning=FALSE}
data_total=data_total_sans
Age=c(round(mean(data_total[data_total[,2]=="COLD"&!is.na(data_total[,6]),6]),1),round(mean(data_total[data_total[,2]=="DIFFUSE",6]),1))
Female=c(table(data_total[data_total[,2]=="COLD",5])[1],table(data_total[data_total[,2]=="DIFFUSE",5])[1])
Male=c(table(data_total[data_total[,2]=="COLD",5])[2],table(data_total[data_total[,2]=="DIFFUSE",5])[2])
Event=c(table(data_total[data_total[,2]=="COLD",3])[2],table(data_total[data_total[,2]=="DIFFUSE",3])[2])
No_event=c(table(data_total[data_total[,2]=="COLD",3])[1],table(data_total[data_total[,2]=="DIFFUSE",3])[1])
Median_followup=c(round(summary(data_total[data_total[,2]=="COLD",4])[3],1),round(summary(data_total[data_total[,2]=="DIFFUSE",4])[3],1))
cold_diffuse_table=rbind(Age,Female,Male,Event,No_event,Median_followup)
colnames(cold_diffuse_table)=c("Cold (n=66)","Diffuse (n=92)")
rownames(cold_diffuse_table)=c("Age (mean)","Female","Male","Event","No event", "Median follow up (years)")
```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=cold_diffuse_table,caption = "Cold vs Diffuse patients"),full_width = F, html_font = "Cambria")

```

We can look at the Kaplan Meier curves for cold vs diffuse: 


```{r,echo=F,message=FALSE,warning=FALSE, fig.cap="Kaplan Meier curves, cold vs diffuse, without G4 patients"}
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total_sans)


p_cd_total= ggsurvplot(fit, data = data_total_sans,
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Cold vs Diffuse",
            legend.labs = c("Cold", "Diffuse"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            palette = c("#E7B800", "#2E9FDF"),
            
            title="C vs D, Total",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
p_cd_total
 
```

Again, there is no clear difference between the trajectories of cold and diffuse


## Astro vs Oligo {.tabset .tabset-pills}



We also have the Astro or Oligo status of the patients, with a total of 73 Astro patients (4 of them being G4) and 87 Oligo patients. The following table presents the different characteristics: 

### With G4
```{r, eval=T,echo=FALSE,message=FALSE,warning=FALSE}
data_total=data_total_avec
Age=c(round(mean(data_total[data_total[,7]=="oligo"&!is.na(data_total[,6])&!is.na(data_total[,7]),6]),1),round(mean(data_total[data_total[,7]=="astro"&!is.na(data_total[,6])&!is.na(data_total[,7]),6]),1))
Female=c(table(data_total[data_total[,7]=="oligo",5])[1],table(data_total[data_total[,7]=="astro",5])[1])
Male=c(table(data_total[data_total[,7]=="oligo",5])[2],table(data_total[data_total[,7]=="astro",5])[2])
Event=c(table(data_total[data_total[,7]=="oligo",3])[2],table(data_total[data_total[,7]=="astro",3])[2])
No_event=c(table(data_total[data_total[,7]=="oligo",3])[1],table(data_total[data_total[,7]=="astro",3])[1])
Median_followup=c(round(summary(data_total[data_total[,7]=="oligo",4])[3],1),round(summary(data_total[data_total[,7]=="astro",4])[3],1))
oligo_astro_table=rbind(Age,Female,Male,Event,No_event,Median_followup)
colnames(oligo_astro_table)=c("oligo (n=87)","astro (n=73)")
rownames(oligo_astro_table)=c("Age (mean)","Female","Male","Event","No event", "Median follow up (years)")
```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE}
library(kableExtra)

 kable_classic(kbl(x=oligo_astro_table,caption = "Oligo vs Astro patients"),full_width = F, html_font = "Cambria")

```


We can look at the Kaplan Meier curves: 

```{r,echo=F,message=FALSE,warning=FALSE, fig.cap="Kaplan Meier curves, oligo vs astro, with G4 patients"}
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro, data = data_total_avec)


 p_oa_total=ggsurvplot(fit, data = data_total_avec,
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Oligo vs Astro",
            legend.labs = c("Astro", "Oligo"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            #palette = c("#E7B800", "#2E9FDF"),
            
            title="O vs A, Total",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 p_oa_total
```


### Without G4
```{r, eval=T,echo=FALSE,message=FALSE,warning=FALSE}
data_total=data_total_sans
Age=c(round(mean(data_total[data_total[,7]=="oligo"&!is.na(data_total[,6])&!is.na(data_total[,7]),6]),1),round(mean(data_total[data_total[,7]=="astro"&!is.na(data_total[,6])&!is.na(data_total[,7]),6]),1))
Female=c(table(data_total[data_total[,7]=="oligo",5])[1],table(data_total[data_total[,7]=="astro",5])[1])
Male=c(table(data_total[data_total[,7]=="oligo",5])[2],table(data_total[data_total[,7]=="astro",5])[2])
Event=c(table(data_total[data_total[,7]=="oligo",3])[2],table(data_total[data_total[,7]=="astro",3])[2])
No_event=c(table(data_total[data_total[,7]=="oligo",3])[1],table(data_total[data_total[,7]=="astro",3])[1])
Median_followup=c(round(summary(data_total[data_total[,7]=="oligo",4])[3],1),round(summary(data_total[data_total[,7]=="astro",4])[3],1))
oligo_astro_table=rbind(Age,Female,Male,Event,No_event,Median_followup)
colnames(oligo_astro_table)=c("oligo (n=87)","astro (n=68)")
rownames(oligo_astro_table)=c("Age (mean)","Female","Male","Event","No event", "Median follow up (years)")
```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE}
library(kableExtra)

 kable_classic(kbl(x=oligo_astro_table,caption = "Oligo vs Astro patients"),full_width = F, html_font = "Cambria")

```


We can look at the Kaplan Meier curves: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE, fig.cap="Kaplan Meier curves, astro vs oligo, without G4 patients"}
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro, data = data_total_sans)


 p_oa_total=ggsurvplot(fit, data = data_total_sans,
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Oligo vs Astro",
            legend.labs = c("Astro", "Oligo"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            #palette = c("#E7B800", "#2E9FDF"),
            
            title="O vs A, Total",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 p_oa_total
```



## Recap { .tabset .tabset-pills}

When we look at the total data, the difference between cold and diffuse seems to be linked with survival, but not the astro vs oligo. We can also look at the data patients based on the RNA or proteomics data being measured to be sure that we have similar patients. 




We can look at the Kaplan Meier curves: 


### With G4
```{r,echo=F,message=FALSE,warning=FALSE, fig.cap="Kaplan Meier curves comparison"}
data_total=data_total_avec
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total[data_total[,8]=="1",])


 p_cd_rna=ggsurvplot(fit, data = data_total[data_total[,8]=="1",],
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Cold vs Diffuse",
            legend.labs = c("Cold", "Diffuse"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            palette = c("#E7B800", "#2E9FDF"),
            
            title="C vs D, RNA",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total[data_total[,8]=="0",])


 p_cd_nrna=ggsurvplot(fit, data = data_total[data_total[,8]=="0",],
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Cold vs Diffuse",
            legend.labs = c("Cold", "Diffuse"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            palette = c("#E7B800", "#2E9FDF"),
            
            title="C vs D, not on RNA",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 


fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro, data = data_total[data_total[,8]=="1",])


 p_oa_rna=ggsurvplot(fit, data = data_total[data_total[,8]=="1",],
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Oligo vs Astro",
            legend.labs = c("Astro", "Oligo"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            #palette = c("#E7B800", "#2E9FDF"),
            
            title="O vs A, RNA",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 
 fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total[data_total[,9]=="1",])


 p_cd_proteo=ggsurvplot(fit, data = data_total[data_total[,9]=="1",],
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Cold vs Diffuse",
            legend.labs = c("Cold", "Diffuse"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            palette = c("#E7B800", "#2E9FDF"),
            
            title="C vs D, Protéo",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 


fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro, data = data_total[data_total[,9]=="1",])


 p_oa_proteo=ggsurvplot(fit, data = data_total[data_total[,9]=="1",],
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Oligo vs Astro",
            legend.labs = c("Astro", "Oligo"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            #palette = c("#E7B800", "#2E9FDF"),
            
            title="O vs A, Proteo",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 
 fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro, data = data_total_avec)


 p_oa_total=ggsurvplot(fit, data = data_total_avec,
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Oligo vs Astro",
            legend.labs = c("Astro", "Oligo"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            #palette = c("#E7B800", "#2E9FDF"),
            
            title="O vs A, Total",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 
 
 fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total_avec)


p_cd_total= ggsurvplot(fit, data = data_total_avec,
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Cold vs Diffuse",
            legend.labs = c("Cold", "Diffuse"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            palette = c("#E7B800", "#2E9FDF"),
            
            title="C vs D, Total",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 
 library(patchwork)
(p_cd_rna$plot+p_cd_proteo$plot+p_cd_total$plot)/(p_oa_rna$plot+p_oa_proteo$plot+p_oa_total$plot)/guide_area()+plot_layout(guides="collect")
 
# (p_cd_rna$plot+p_oa_rna$plot)/(p_cd_proteo$plot+p_oa_proteo$plot)/(p_cd_total$plot+p_oa_total$plot)/guide_area()+plot_layout(guides="collect")
```


As we can see, the RNA patients seems to present no differences between cold and diffuse, while it is not the case with the proteomics and total data. We can compare the pvalues associated with different test (Cox, but also other test with no proportional risk hypothesis such as log-rank):

```{r,echo=F,message=FALSE,warning=FALSE}
#total
data_total=data_total_avec
modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=data_total)
modele_univarie= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse,data=data_total)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total)
cd_total_pval=round(c(summary(modele_base)$coefficients[1,5],summary(modele_univarie)$coefficients[5],surv_pvalue(fit,data=data_total,method="1")$pval,surv_pvalue(fit,data=data_total,method="sqrtN")$pval,surv_pvalue(fit,data=data_total,method="S1")$pval,surv_pvalue(fit,data=data_total,method="S2")$pval),3)

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=data_total)
modele_univarie= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro,data=data_total)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro, data = data_total)
oa_total_pval=round(c(summary(modele_base)$coefficients[4,5],summary(modele_univarie)$coefficients[5],surv_pvalue(fit,data=data_total,method="1")$pval,surv_pvalue(fit,data=data_total,method="sqrtN")$pval,surv_pvalue(fit,data=data_total,method="S1")$pval,surv_pvalue(fit,data=data_total,method="S2")$pval),3)


modele_univarie= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_oligo,data=data_total)


#RNA

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=data_total[data_total[,8]=="1",])
modele_univarie= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse,data=data_total[data_total[,8]=="1",])
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total[data_total[,8]=="1",])
cd_rna_pval=round(c(summary(modele_base)$coefficients[1,5],summary(modele_univarie)$coefficients[5],surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="1")$pval,surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="sqrtN")$pval,surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="S1")$pval,surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="S2")$pval),3)

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=data_total[data_total[,8]=="1",])
modele_univarie= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro,data=data_total[data_total[,8]=="1",])
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro, data = data_total[data_total[,8]=="1",])
oa_rna_pval=round(c(summary(modele_base)$coefficients[4,5],summary(modele_univarie)$coefficients[5],surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="1")$pval,surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="sqrtN")$pval,surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="S1")$pval,surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="S2")$pval),3)

#proteo

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=data_total[data_total[,9]=="1",])
modele_univarie= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse,data=data_total[data_total[,9]=="1",])
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total[data_total[,9]=="1",])
cd_proteo_pval=round(c(summary(modele_base)$coefficients[1,5],summary(modele_univarie)$coefficients[5],surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="1")$pval,surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="sqrtN")$pval,surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="S1")$pval,surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="S2")$pval),3)

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=data_total[data_total[,9]=="1",])
modele_univarie= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro,data=data_total[data_total[,9]=="1",])
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro, data = data_total[data_total[,9]=="1",])
oa_proteo_pval=round(c(summary(modele_base)$coefficients[4,5],summary(modele_univarie)$coefficients[5],surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="1")$pval,surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="sqrtN")$pval,surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="S1")$pval,surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="S2")$pval),3)


table_pvals=rbind(cd_rna_pval,oa_rna_pval,cd_proteo_pval,oa_proteo_pval,cd_total_pval,oa_total_pval)
colnames(table_pvals)=c("Cox multivarié","Cox univarié","Log-rank","Tarone-Ware","Peto-Peto","Modified Peto-Peto")
rownames(table_pvals)=c("RNA: C vs D ","RNA: O vs A ","Protéo: C vs D ","Protéo: O vs A ","total: C vs D ","total: O vs A ")

 kable_classic(kbl(x=table_pvals,caption = "Statistical test"),full_width = F, html_font = "Cambria")

```
The cox model is significant for Cold vs Diffuse, with proteomics data and total data

### Without G4 

```{r,echo=F,message=FALSE,warning=FALSE, fig.cap="Kaplan Meier curves comparison, without G4 patients"}
data_total=data_total_sans
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total[data_total[,8]=="1",])


 p_cd_rna=ggsurvplot(fit, data = data_total[data_total[,8]=="1",],
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Cold vs Diffuse",
            legend.labs = c("Cold", "Diffuse"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            palette = c("#E7B800", "#2E9FDF"),
            
            title="C vs D, RNA",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 


fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro, data = data_total[data_total[,8]=="1",])


 p_oa_rna=ggsurvplot(fit, data = data_total[data_total[,8]=="1",],
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Oligo vs Astro",
            legend.labs = c("Astro", "Oligo"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            #palette = c("#E7B800", "#2E9FDF"),
            
            title="O vs A, RNA",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 
 fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total[data_total[,9]=="1",])


 p_cd_proteo=ggsurvplot(fit, data = data_total[data_total[,9]=="1",],
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Cold vs Diffuse",
            legend.labs = c("Cold", "Diffuse"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            palette = c("#E7B800", "#2E9FDF"),
            
            title="C vs D, Protéo",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 


fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro, data = data_total[data_total[,9]=="1",])


 p_oa_proteo=ggsurvplot(fit, data = data_total[data_total[,9]=="1",],
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Oligo vs Astro",
            legend.labs = c("Astro", "Oligo"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            #palette = c("#E7B800", "#2E9FDF"),
            
            title="O vs A, Proteo",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 
 fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro, data = data_total_sans)


 p_oa_total=ggsurvplot(fit, data = data_total_sans,
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Oligo vs Astro",
            legend.labs = c("Astro", "Oligo"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            #palette = c("#E7B800", "#2E9FDF"),
            
            title="O vs A, Total",
            ggtheme = theme_bw() # Change ggplot2 theme
 )

 fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total_sans)


p_cd_total= ggsurvplot(fit, data = data_total_sans,
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Cold vs Diffuse",
            legend.labs = c("Cold", "Diffuse"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            palette = c("#E7B800", "#2E9FDF"),
            
            title="C vs D, Total",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 library(patchwork)
(p_cd_rna$plot+p_cd_proteo$plot+p_cd_total$plot)/(p_oa_rna$plot+p_oa_proteo$plot+p_oa_total$plot)/guide_area()+plot_layout(guides="collect")
 
# (p_cd_rna$plot+p_oa_rna$plot)/(p_cd_proteo$plot+p_oa_proteo$plot)/(p_cd_total$plot+p_oa_total$plot)/guide_area()+plot_layout(guides="collect")
```


As we can see, the RNA patients seems to present no differences between cold and diffuse, while it is not the case with the proteomics and total data. We can compare the pvalues associated with different test (Cox, but also other test with no proportional risk hypothesis such as log-rank):

```{r,echo=F,message=FALSE,warning=FALSE}
#total
data_total=data_total_sans
modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=data_total)
modele_univarie= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse,data=data_total)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total)
cd_total_pval=round(c(summary(modele_base)$coefficients[1,5],summary(modele_univarie)$coefficients[5],surv_pvalue(fit,data=data_total,method="1")$pval,surv_pvalue(fit,data=data_total,method="sqrtN")$pval,surv_pvalue(fit,data=data_total,method="S1")$pval,surv_pvalue(fit,data=data_total,method="S2")$pval),3)

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=data_total)
modele_univarie= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro,data=data_total)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro, data = data_total)
oa_total_pval=round(c(summary(modele_base)$coefficients[4,5],summary(modele_univarie)$coefficients[5],surv_pvalue(fit,data=data_total,method="1")$pval,surv_pvalue(fit,data=data_total,method="sqrtN")$pval,surv_pvalue(fit,data=data_total,method="S1")$pval,surv_pvalue(fit,data=data_total,method="S2")$pval),3)

#RNA

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=data_total[data_total[,8]=="1",])
modele_univarie= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse,data=data_total[data_total[,8]=="1",])
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total[data_total[,8]=="1",])
cd_rna_pval=round(c(summary(modele_base)$coefficients[1,5],summary(modele_univarie)$coefficients[5],surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="1")$pval,surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="sqrtN")$pval,surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="S1")$pval,surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="S2")$pval),3)

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=data_total[data_total[,8]=="1",])
modele_univarie= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro,data=data_total[data_total[,8]=="1",])
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro, data = data_total[data_total[,8]=="1",])
oa_rna_pval=round(c(summary(modele_base)$coefficients[4,5],summary(modele_univarie)$coefficients[5],surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="1")$pval,surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="sqrtN")$pval,surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="S1")$pval,surv_pvalue(fit,data=data_total[data_total[,8]=="1",],method="S2")$pval),3)

#proteo

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=data_total[data_total[,9]=="1",])
modele_univarie= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse,data=data_total[data_total[,9]=="1",])
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total[data_total[,9]=="1",])
cd_proteo_pval=round(c(summary(modele_base)$coefficients[1,5],summary(modele_univarie)$coefficients[5],surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="1")$pval,surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="sqrtN")$pval,surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="S1")$pval,surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="S2")$pval),3)

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=data_total[data_total[,9]=="1",])
modele_univarie= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro,data=data_total[data_total[,9]=="1",])
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ oligo_astro, data = data_total[data_total[,9]=="1",])
oa_proteo_pval=round(c(summary(modele_base)$coefficients[4,5],summary(modele_univarie)$coefficients[5],surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="1")$pval,surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="sqrtN")$pval,surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="S1")$pval,surv_pvalue(fit,data=data_total[data_total[,9]=="1",],method="S2")$pval),3)


table_pvals=rbind(cd_rna_pval,oa_rna_pval,cd_proteo_pval,oa_proteo_pval,cd_total_pval,oa_total_pval)
colnames(table_pvals)=c("Cox multivarié","Cox univarié","Log-rank","Tarone-Ware","Peto-Peto","Modified Peto-Peto")
rownames(table_pvals)=c("RNA: C vs D ","RNA: O vs A ","Protéo: C vs D ","Protéo: O vs A ","total: C vs D ","total: O vs A ")

 kable_classic(kbl(x=table_pvals,caption = "Statistical test"),full_width = F, html_font = "Cambria")

```
The cold vs diffuse status is significan for the cox model with proteomics and total data

## Patient included in RNA 

We can observe that the patients of RNA might be different from the rest of the patients: the cold vs diffuse status is not significant, and the proportion of event is not the same than the rest of the patients. It could be simply due to randomness, but we can describe more this difference. The following table compares the characteristics of the patients that were included in the RNAseq, and the characteristics of the patients that were not. 

```{r, eval=T,echo=FALSE,message=FALSE,warning=FALSE}
data_total=data_total_avec[data_total_avec[,8]=="1",]
Age=c(round(mean(data_total[data_total[,2]=="COLD"&!is.na(data_total[,6]),6]),1),round(mean(data_total[data_total[,2]=="DIFFUSE",6]),1))
Female=c(table(data_total[data_total[,2]=="COLD",5])[1],table(data_total[data_total[,2]=="DIFFUSE",5])[1])
Male=c(table(data_total[data_total[,2]=="COLD",5])[2],table(data_total[data_total[,2]=="DIFFUSE",5])[2])
Event=c(table(data_total[data_total[,2]=="COLD",3])[2],table(data_total[data_total[,2]=="DIFFUSE",3])[2])
No_event=c(table(data_total[data_total[,2]=="COLD",3])[1],table(data_total[data_total[,2]=="DIFFUSE",3])[1])
Somme_event=Event+No_event
percentage_event=100*round(Event/Somme_event,2)
percentage_noevent=100*round(No_event/Somme_event,2)

Event=paste(Event," (",percentage_event,"%)",sep="")
No_event=paste(No_event," (",percentage_noevent,"%)",sep="")

Median_followup=c(round(summary(data_total[data_total[,2]=="COLD",4])[3],1),round(summary(data_total[data_total[,2]=="DIFFUSE",4])[3],1))
cold_diffuse_table=rbind(Age,Female,Male,Event,No_event,Median_followup)
colnames(cold_diffuse_table)=c("Cold (n=24)","Diffuse (n=35)")
rownames(cold_diffuse_table)=c("Age (mean)","Female","Male","Event","No event", "Median follow up (years)")
cold_diffuse_table_RNA=cold_diffuse_table
```

```{r, eval=T,echo=FALSE,message=FALSE,warning=FALSE}
data_total=data_total_avec[data_total_avec[,8]=="0",]
Age=c(round(mean(data_total[data_total[,2]=="COLD"&!is.na(data_total[,6]),6]),1),round(mean(data_total[data_total[,2]=="DIFFUSE",6]),1))
Female=c(table(data_total[data_total[,2]=="COLD",5])[1],table(data_total[data_total[,2]=="DIFFUSE",5])[1])
Male=c(table(data_total[data_total[,2]=="COLD",5])[2],table(data_total[data_total[,2]=="DIFFUSE",5])[2])
Event=c(table(data_total[data_total[,2]=="COLD",3])[2],table(data_total[data_total[,2]=="DIFFUSE",3])[2])
No_event=c(table(data_total[data_total[,2]=="COLD",3])[1],table(data_total[data_total[,2]=="DIFFUSE",3])[1])
Somme_event=Event+No_event
percentage_event=100*round(Event/Somme_event,2)
percentage_noevent=100*round(No_event/Somme_event,2)

Event=paste(Event," (",percentage_event,"%)",sep="")
No_event=paste(No_event," (",percentage_noevent,"%)",sep="")

Median_followup=c(round(summary(data_total[data_total[,2]=="COLD",4])[3],1),round(summary(data_total[data_total[,2]=="DIFFUSE",4])[3],1))
cold_diffuse_table=rbind(Age,Female,Male,Event,No_event,Median_followup)
colnames(cold_diffuse_table)=c("Cold (n=44)","Diffuse (n=60)")
rownames(cold_diffuse_table)=c("Age (mean)","Female","Male","Event","No event", "Median follow up (years)")
cold_diffuse_table_nRNA=cold_diffuse_table
```

`



```{r, echo = FALSE}
t1=cold_diffuse_table_RNA
t2=cold_diffuse_table_nRNA
kable(t1,caption = c("Patients in RNA")) %>%
  kable_styling(full_width = FALSE, position = "float_left")
kable(t2,caption = c("Patients not in RNA")) %>%
  kable_styling(full_width = FALSE, position = "left")
```

As we can seen, the age and sex variables are similar between the included and excluded, but there is less event for the diffuse for the patients included than for the excluded patients. We can look further at the Kaplan Meier curves: 

```{r,echo=F,message=FALSE,warning=FALSE, fig.cap="Kaplan Meier curves comparison"}
data_total=data_total_avec
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total[data_total[,8]=="1",])


 p_cd_rna=ggsurvplot(fit, data = data_total[data_total[,8]=="1",],
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Cold vs Diffuse",
            legend.labs = c("Cold", "Diffuse"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            palette = c("#E7B800", "#2E9FDF"),
            
            title="C vs D, RNA",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse, data = data_total[data_total[,8]=="0",])


 p_cd_nrna=ggsurvplot(fit, data = data_total[data_total[,8]=="0",],
            surv.median.line = "hv", # Add medians survival
            
            # Change legends: title & labels
            legend.title = "Cold vs Diffuse",
            legend.labs = c("Cold", "Diffuse"),
            
            conf.int = TRUE,
            # Add risk table
            risk.table = TRUE,
            tables.height = 0.2,
            tables.theme = theme_cleantable(),
            
            # Color palettes. Use custom color: c("#E7B800", "#2E9FDF"),
            # or brewer color (e.g.: "Dark2"), or ggsci color (e.g.: "jco")
            palette = c("#E7B800", "#2E9FDF"),
            
            title="C vs D, not on RNA",
            ggtheme = theme_bw() # Change ggplot2 theme
 )
 
(p_cd_rna$plot+p_cd_nrna$plot)/guide_area()+plot_layout(guides="collect")
```
The curves for the cold and diffuse on the RNA patients are similar, while the cold patients seems to have a better survival for the excluded patients, event if the interval of confidence overlaps.



## Description of G4 patients

We can look more at the 5 G4 patients. The following table shows the different characteristics of these patients 


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
data_total=data_total_avec
table_G4=data_total[data_total[,11]==1,]
table_G4=table_G4[,c(1,5,6,2,7,4,3)]
rownames(table_G4)=NULL
```


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=table_G4,caption = "Characteristics of G4 patients"),full_width = F, html_font = "Cambria")

```
We can also compare these characteristics with the rest of the patients: 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
data_total=data_total_avec
Age=c(round(mean(data_total[data_total[,11]==1&!is.na(data_total[,6]),6]),1),round(mean(data_total[data_total[,11]==0,6]),1))
Female=c(table(data_total[data_total[,11]==1,5])[1],table(data_total[data_total[,11]==0,5])[1])
Male=c(table(data_total[data_total[,11]==1,5])[2],table(data_total[data_total[,11]==0,5])[2])
Event=c(table(data_total[data_total[,11]==1,3])[2],table(data_total[data_total[,11]==0,3])[2])
No_event=c(table(data_total[data_total[,11]==1,3])[1],table(data_total[data_total[,11]==0,3])[1])
Median_followup=c(round(summary(data_total[data_total[,11]==1,4])[3],1),round(summary(data_total[data_total[,11]==0,4])[3],1))
G4_table=rbind(Age,Female,Male,Event,No_event,Median_followup)
colnames(G4_table)=c("G4 (n=5)","not G4 (n=158)")
rownames(G4_table)=c("Age (mean)","Female","Male","Event","No event", "Median follow up (years)")
```


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=G4_table,caption = "G4 and not G4 patients"),full_width = F, html_font = "Cambria")

```

# RNAseq data 
We have gene expression data (RNAseq) for 59 patients, 4 patients being G4. The following figure presents the boxplot of the normalized counts per patients (normalization log CPM). All data have been normalized according to the log CPM, a transformation that allows us to compare the samples, by make less important the biological variations between them. The log2 transformation means that when the counts after normalization increase of one units, the raw counts have doubled. 

```{r, eval=T,echo=FALSE, message=FALSE,warning=FALSE, fig.cap="normalized counts per patients, depending on the status"}

counts=read.table("data/counts_final.txt")


datas=data_total_avec[data_total_avec[,8]==1,-c(8,9)]
names=datas[,1]
names_table_split=strsplit(as.character(names),split="_")
names_table=numeric(length(datas[,1]))
for (i in 1:length(names_table)){
  names_table[i]=names_table_split[[i]][1]
}
datas[,1]=names_table
datas=datas[!duplicated(datas[,1]),]


counts=counts[rownames(counts)%in%datas[,1],]
liste_vide=c()
liste_variance_nulle=c()
for (i in 1:length(counts[1,])){
  if (sum(as.vector(counts[,i])==rep(0,length(counts[,1])))==length(counts[,1])){
    liste_vide=c(liste_vide,i)
  }
  if (sd(counts[,i])==0){
    liste_variance_nulle=c(liste_variance_nulle,i)
  }
}

counts=counts[,-liste_vide]
norm_counts <- apply(counts, MARGIN = 2, function(v) {
  log2((v + 0.5)/(sum(v) + 1) * 10^6)
})
norm_counts <- as.data.frame(norm_counts)
raw_counts=counts
counts=norm_counts

rownames(datas)=datas[,1]
datas=datas[rownames(counts),]

datas_avec=datas
datas_sans=datas[datas[,9]==0,]
counts_sans=counts[rownames(counts)%in%datas_sans[,1],]
norm_counts_sans=counts[rownames(counts)%in%datas_sans[,1],]
raw_counts_sans=raw_counts[rownames(raw_counts)%in%datas_sans[,1],]


data_histo=cbind(datas[,1],datas[,9],datas[,2],datas[,7],counts)
colnames(data_histo)=c("ID","G4","Cold_diffuse","Astro_oligo",colnames(counts))
data_histo=data_histo[order(data_histo[,2],data_histo[,3]),]
data_histo[,1]=factor(data_histo[,1],levels=data_histo[,1])
df.m <- melt(data_histo[,-c(3,4)], id=c("ID","G4"))
plot1=ggplot(data = df.m, aes(x=ID, y=value)) + geom_boxplot(aes(fill=G4),outlier.shape = NA)+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ 
  scale_y_continuous(limits = quantile(df.m$value, c(0.1, 0.9)))+ scale_fill_viridis_d()+xlab("patients")+ylab("counts")



data_histo=cbind(datas[,1],datas[,9],datas[,2],datas[,7],counts)
colnames(data_histo)=c("ID","G4","Cold_diffuse","Astro_oligo",colnames(counts))
data_histo=data_histo[order(data_histo[,2],data_histo[,3]),]
data_histo[,1]=factor(data_histo[,1],levels=data_histo[,1])
df.m <- melt(data_histo[,-c(2,4)], id=c("ID","Cold_diffuse"))
plot2=ggplot(data = df.m, aes(x=ID, y=value)) + geom_boxplot(aes(fill=Cold_diffuse),outlier.shape = NA)+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ 
  scale_y_continuous(limits = quantile(df.m$value, c(0.1, 0.9)))+xlab("patients")+ylab("counts")+scale_fill_manual(values= c("#E7B800", "#2E9FDF"))


data_histo=cbind(datas[,1],datas[,9],datas[,2],datas[,7],counts)
colnames(data_histo)=c("ID","G4","Cold_diffuse","Astro_oligo",colnames(counts))
data_histo=data_histo[order(data_histo[,2],data_histo[,3]),]
data_histo[,1]=factor(data_histo[,1],levels=data_histo[,1])
df.m <- melt(data_histo[,-c(2,3)], id=c("ID","Astro_oligo"))
plot3=ggplot(data = df.m, aes(x=ID, y=value)) + geom_boxplot(aes(fill=Astro_oligo),outlier.shape = NA)+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ 
  scale_y_continuous(limits = quantile(df.m$value, c(0.1, 0.9)))+xlab("patients")+ylab("counts")
#+scale_fill_manual(values= c("#E7B800", "#2E9FDF"))

plot1/plot2/plot3
```


## Cold versus Diffuse {.tabset .tabset-pills}
We have gene expression data (RNAseq) for 59 patients, with 24 cold patients (2 being G4) and 35 diffuse patients (2 being G4). The following table shows the different characteristics of these patients. The previous figure shows the differences in therm of gene expression between G4 and not G4, but also Cold and Diffuse


We performed a PCA with all counts, with a separation depending on the cold or diffuse status. The PCA was done after a VST transformation on the raw counts, followed by a scaling (centered and reduced), as all PCA were performed

### With G4

```{r, eval=T,echo=FALSE,message=FALSE,warning=FALSE}
datas=datas_avec
Age=c(round(mean(datas[datas[,2]=="COLD"&!is.na(datas[,6]),6]),1),round(mean(datas[datas[,2]=="DIFFUSE",6]),1))
Female=c(table(datas[datas[,2]=="COLD",5])[1],table(datas[datas[,2]=="DIFFUSE",5])[1])
Male=c(table(datas[datas[,2]=="COLD",5])[2],table(datas[datas[,2]=="DIFFUSE",5])[2])
Event=c(table(datas[datas[,2]=="COLD",3])[2],table(datas[datas[,2]=="DIFFUSE",3])[2])
No_event=c(table(datas[datas[,2]=="COLD",3])[1],table(datas[datas[,2]=="DIFFUSE",3])[1])
Median_followup=c(round(summary(datas[datas[,2]=="COLD",4])[3],1),round(summary(datas[datas[,2]=="DIFFUSE",4])[3],1))
cold_diffuse_table=rbind(Age,Female,Male,Event,No_event,Median_followup)
colnames(cold_diffuse_table)=c("Cold (n=24)","Diffuse (n=35)")
rownames(cold_diffuse_table)=c("Age (mean)","Female","Male","Event","No event", "Median follow up (years)")
```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=cold_diffuse_table,caption = "Cold vs Diffuse patients"),full_width = F, html_font = "Cambria")

```



```{r,eval=T,echo=F,message=FALSE,warning=FALSE,results=FALSE, fig.show='hide'}
datas=datas_avec
counts_pca <- raw_counts
counts_pca <- round((counts_pca+1),0)
counts_pca <- vst(t(counts_pca))

counts_pca_scaled <- base::scale(t(counts_pca), center=TRUE, scale=TRUE)

res.pca=PCA(counts_pca_scaled , graph=FALSE)
#plot.PCA(res.pca, axes=c(1, 2), choix="ind")
Cold_or_diffuse=as.factor(datas[,2])
G4=as.factor(datas[,9])
pca_data <- cbind.data.frame(res.pca$ind$coord[, 1:4],as.factor(datas[,2]),rownames(res.pca$ind$coord))
colnames(pca_data)=c("Dim.1","Dim.2","Dim.3","Dim.4","Cold_diffuse","ID")
table_coord=res.pca$ind$coord[,c(1,2)]
particuliers=table_coord[table_coord[,1]>quantile(table_coord[,1],0.97)|table_coord[,1]<quantile(table_coord[,1],0.03)|table_coord[,2]>quantile(table_coord[,2],0.98)|table_coord[,2]<quantile(table_coord[,2],0.02),]

for (i in 1:length(pca_data[,1])){
  if (!pca_data[i,6]%in%rownames(particuliers)){
    pca_data[i,6]=""
    }
}
p <- ggplot(pca_data, aes(x=Dim.1, y=Dim.2, colour = as.factor(datas[,2]),shape=G4,label=ID)) +
 # viridis::scale_color_viridis("Cold_diffuse", discrete = TRUE,end = 0.8)
    scale_color_manual(values= c("#E7B800", "#2E9FDF"))+
  geom_text_repel() +
  theme_bw() +
  geom_point(size=2, alpha=1) +
  guides(color=guide_legend(override.aes = list(alpha=1),title="Cold or Diffuse")) +
  xlab(label=paste0("PC1 (", round(res.pca$eig[1,2], 2),"% of total variance explained)")) +
  ylab(label=paste0("PC2 (", round(res.pca$eig[2,2], 2), "% of total variance explained)")) +
  ggtitle("", subtitle = "First factorial plan")


```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE, fig.cap="PCA of the RNA data, with G4 patients (vst transformation and scaled data)"}
p
```
Some patients seems to be away from the majority of the patients, here are their characteristics: 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
datas=datas_avec
table_particulier=datas[datas[,1]%in%rownames(particuliers),]
table_particulier=table_particulier[,c(1,5,6,2,7,4,3)]
table_particulier[,6]=round(table_particulier[,6],2)
rownames(table_particulier)=NULL
```


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=table_particulier,caption = "Characteristics patients outsite of PCA majority"),full_width = F, html_font = "Cambria")

```

### Without G4

```{r, eval=T,echo=FALSE,message=FALSE,warning=FALSE}
datas=datas_sans
Age=c(round(mean(datas[datas[,2]=="COLD"&!is.na(datas[,6]),6]),1),round(mean(datas[datas[,2]=="DIFFUSE",6]),1))
Female=c(table(datas[datas[,2]=="COLD",5])[1],table(datas[datas[,2]=="DIFFUSE",5])[1])
Male=c(table(datas[datas[,2]=="COLD",5])[2],table(datas[datas[,2]=="DIFFUSE",5])[2])
Event=c(table(datas[datas[,2]=="COLD",3])[2],table(datas[datas[,2]=="DIFFUSE",3])[2])
No_event=c(table(datas[datas[,2]=="COLD",3])[1],table(datas[datas[,2]=="DIFFUSE",3])[1])
Median_followup=c(round(summary(datas[datas[,2]=="COLD",4])[3],1),round(summary(datas[datas[,2]=="DIFFUSE",4])[3],1))
cold_diffuse_table=rbind(Age,Female,Male,Event,No_event,Median_followup)
colnames(cold_diffuse_table)=c("Cold (n=22)","Diffuse (n=33)")
rownames(cold_diffuse_table)=c("Age (mean)","Female","Male","Event","No event", "Median follow up (years)")
```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=cold_diffuse_table,caption = "Cold vs Diffuse patients"),full_width = F, html_font = "Cambria")

```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,results=FALSE, fig.show='hide'}
datas=datas_sans

counts_pca <- raw_counts_sans
counts_pca <- round((counts_pca+1),0)
counts_pca <- vst(t(counts_pca))

counts_pca_scaled <- base::scale(t(counts_pca), center=TRUE, scale=TRUE)

res.pca=PCA(counts_pca_scaled , graph=FALSE)
#plot.PCA(res.pca, axes=c(1, 2), choix="ind")
Cold_or_diffuse=as.factor(datas[,2])
pca_data <- cbind.data.frame(res.pca$ind$coord[, 1:4],as.factor(datas[,2]),rownames(res.pca$ind$coord))
colnames(pca_data)=c("Dim.1","Dim.2","Dim.3","Dim.4","Cold_diffuse","ID")
table_coord=res.pca$ind$coord[,c(1,2)]
particuliers=table_coord[table_coord[,1]>quantile(table_coord[,1],0.97)|table_coord[,1]<quantile(table_coord[,1],0.03)|table_coord[,2]>quantile(table_coord[,2],0.98)|table_coord[,2]<quantile(table_coord[,2],0.02),]

for (i in 1:length(pca_data[,1])){
  if (!pca_data[i,6]%in%rownames(particuliers)){
    pca_data[i,6]=""
    }
}
p <- ggplot(pca_data, aes(x=Dim.1, y=Dim.2, colour = as.factor(datas[,2]),label=ID)) +
 # viridis::scale_color_viridis("Cold_diffuse", discrete = TRUE,end = 0.8)
    scale_color_manual(values= c("#E7B800", "#2E9FDF"))+
  geom_text_repel() +
  theme_bw() +
  geom_point(size=2, alpha=1) +  guides(color=guide_legend(override.aes = list(alpha=1),title="Cold or Diffuse")) +
  xlab(label=paste0("PC1 (", round(res.pca$eig[1,2], 2),"% of total variance explained)")) +
  ylab(label=paste0("PC2 (", round(res.pca$eig[2,2], 2), "% of total variance explained)")) +
  ggtitle("", subtitle = "First factorial plan")

```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE, fig.cap="PCA of the RNA data, without G4 patients (vst transformation and scaled data)"}
p
```
```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
datas=datas_sans
table_particulier=datas[datas[,1]%in%rownames(particuliers),]
table_particulier=table_particulier[,c(1,5,6,2,7,4,3)]
table_particulier[,6]=round(table_particulier[,6],2)
rownames(table_particulier)=NULL
```


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=table_particulier,caption = "Characteristics patients outsite of PCA majority"),full_width = F, html_font = "Cambria")

```

## {- .unlisted .unnumbered }
There seem to be no particular group inside the PCA. We will then look gene per gene if there is a difference between cold and diffuse. For that, we use the dearseq package.

## {- .tabset .tabset-pills .unlisted .unnumbered}

### With G4

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
datas=datas_avec
output=ifelse(datas[,2]=="COLD",0,1)
Y=matrix(output,ncol=1,nrow=length(output))

scores=dear_seq(exprmat=t(as.matrix(raw_counts)),variables2test = Y,preprocessed = FALSE,which_test = "asymptotic")


genewise_dearseq_cold=as.data.frame(cbind(rownames(scores$pvals),scores$pvals[,2],rank(scores$pvals[,2]),scores$pvals[,1]))
genewise_dearseq_cold[,1]=as.character(genewise_dearseq_cold[,1])
genewise_dearseq_cold[,2]=as.numeric(as.character(genewise_dearseq_cold[,2]))
genewise_dearseq_cold[,3]=as.numeric(as.character(genewise_dearseq_cold[,3]))
genewise_dearseq_cold[,4]=as.numeric(as.character(genewise_dearseq_cold[,4]))
genewise_dearseq_cold=genewise_dearseq_cold[order(genewise_dearseq_cold[,2],decreasing = FALSE),]
colnames(genewise_dearseq_cold)=c("ID","pvalue","rank_pvalue","raw_pvalue")
genewise_dearseq_cold_avec=genewise_dearseq_cold



patients_cold=counts[output==0,]
patients_diffuse=counts[output==1,]
means_diffuse=numeric(length(patients_cold[1,]))
means_cold=numeric(length(patients_cold[1,]))
for (i in 1:length(patients_cold[1,])){
  means_diffuse[i]=mean(patients_diffuse[,i])
  means_cold[i]=mean(patients_cold[,i])
}

fold_change=data.frame(colnames(patients_cold),means_diffuse-means_cold)
colnames(fold_change)=c("ID","log2FoldChange")
total_cold=merge(genewise_dearseq_cold_avec,fold_change,by="ID")
total_cold=total_cold[order(total_cold[,4],total_cold[,5]),]

write.xlsx(total_cold[total_cold[,2]<0.05,], file = "dearseq_RNA_C_D_wg4.xlsx",col.names = TRUE)


total_cold$diffexpressed <- "NO"
# if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP" 
total_cold$diffexpressed[total_cold$log2FoldChange > 0.6 & total_cold$pvalue < 0.05] <- "UP"
# if log2Foldchange <  (-0.6) and pvalue < 0.05, set as "DOWN"
total_cold$diffexpressed[total_cold$log2FoldChange <  (-0.6) & total_cold$pvalue < 0.05] <- "DOWN"

total_cold$delabel <- NA
total_cold$delabel[total_cold$diffexpressed != "NO"] <- total_cold$ID[total_cold$diffexpressed != "NO"]


plot1=ggplot(data=total_cold, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed, label=delabel)) + 
    geom_point() + 
    theme_minimal() +
  geom_text_repel()  + scale_color_manual(values= c("#440154FF", "#1F968BFF", "#DCE319FF")) +ggtitle("Log2 Fold change in function of pvalues")

plot2=ggplot(data=total_cold,aes(x=log2FoldChange))+geom_density()+xlim(c(-2,2))+ theme_minimal()+ggtitle("Distribution of Log2FoldChange") 
```

dearseq identifies `r dim(genewise_dearseq_cold[genewise_dearseq_cold[,2]<0.05,])[1]` genes that are differentially expressed between cold and diffuse. Here are the first genes in term of p-value:

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
rownames(total_cold)=NULL
 kable_classic(kbl(x=total_cold[1:10,c(1,2,3,4,5)],caption = "Cold vs Diffuse genes"),full_width = F, html_font = "Cambria")

```

We can look at the vulcano plot and also the distribution of the log2 Fold change, in the following figure. The vulcano plot shows the log10(pvalue) in function of the log2 Fold change, which allows to see if the significant genes are up regulated or down regulated (cold as base). A positive log2 Fold change indicates that the gene is, on average, more expressed among the diffuse patients, while a negative log2 Fold change indicates that the gene is more expressed among the cold patients. 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,fig.cap="Vulcano plot and distribution of log2 Fold change",cache=FALSE}
plot1+plot2
```

We can observe the brier score associated with these genes. The brier score is an indicator of the prediction aibility, between 1 and 0, 0 being a perfect predictor and 1 a poor predictor: 



```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=TRUE}


Y <- output
X <- counts[,colnames(counts)%in%genewise_dearseq_cold[genewise_dearseq_cold[,2]<0.05,1]]
temp=matrix(nrow =dim(X)[1],ncol=dim(X)[2] )
for (i in 1:dim(X)[1]){ # leave one out
  for (j in 1:dim(X)[2]){
    fit <- glm(Y~X,data = data.frame(Y=as.factor(Y[-i]),X = X[-i,j]),family=binomial()) # On apprend sans la ieme observation
    prob <- predict(fit,newdata=data.frame(X=X[i,j]), type="response") # On predit la ieme observation
    temp[i,j] <- prob # vecteur proba prÃ©dite d'appartenir Ã une condition
  }
}

BS <- NULL
for (k in 1:dim(X)[2]){ # gene k
  BS[k] <- brier(Y, temp[,k])$bs # somme de la diffÃ©rence au carrÃ© entre la proba pour chaque obs et la condition
}
BS=cbind(colnames(X),BS)
colnames(BS)=c("ID","BS_v2")
dearseq_brier=merge(genewise_dearseq_cold,BS,by="ID")
dearseq_brier=cbind(dearseq_brier,rank(dearseq_brier[,5]))
dearseq_brier=as.data.frame(dearseq_brier)
colnames(dearseq_brier)=c("ID","pvalue","rank_pvalue","raw_pvalue","BS","rank_BS")
```
```{r,echo=F,eval=T,message=FALSE,warning=FALSE,cache=FALSE, fig.cap="Brier score for the genes identified by dearseq for the cold vs diffuse status (with G4)"}
BS_dearseq_cold=as.numeric(as.character(dearseq_brier[,5]))
df <- data.frame(Brier.score = BS_dearseq_cold,
                 rank=rank(BS_dearseq_cold,ties.method = "random"),
                 method=as.factor(rep("dearseq_asym",length(BS_dearseq_cold))))

g_dearseq_cold <- ggplot(df[df[,3]=="dearseq_asym",], aes(x=rank, y=Brier.score)) +
  theme_minimal() +
  # scale_color_manual( values = c("#CCCCCC", grDevices::heat.colors(n=2)[3])) +
  geom_point(size=1.5,alpha=0.6,color="#FF9900") +
  theme(axis.title = element_text(size=16), axis.text = element_text(size=12), legend.title = element_text(size=16), legend.text = element_text(size=14)) +
  #  guides(color = guide_legend(override.aes = list(alpha=1))) +
  geom_rug(alpha=0.5,color="#FF9900")  + 
  ylab("Brier Score")
g_dearseq_cold
```

### Without G4

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
datas=datas_sans
output_sans=ifelse(datas_sans[,2]=="COLD",0,1)
Y_sans=matrix(output_sans,ncol=1,nrow=length(output_sans))

scores_sans=dear_seq(exprmat=t(as.matrix(raw_counts_sans)),variables2test = Y_sans,preprocessed = FALSE,which_test = "asymptotic")

genewise_dearseq_cold_sans=as.data.frame(cbind(rownames(scores_sans$pvals),scores_sans$pvals[,2],rank(scores_sans$pvals[,2]),scores_sans$pvals[,1]))
genewise_dearseq_cold_sans[,1]=as.character(genewise_dearseq_cold_sans[,1])
genewise_dearseq_cold_sans[,2]=as.numeric(as.character(genewise_dearseq_cold_sans[,2]))
genewise_dearseq_cold_sans[,3]=as.numeric(as.character(genewise_dearseq_cold_sans[,3]))
genewise_dearseq_cold_sans[,4]=as.numeric(as.character(genewise_dearseq_cold_sans[,4]))
genewise_dearseq_cold_sans=genewise_dearseq_cold_sans[order(genewise_dearseq_cold_sans[,2],decreasing = FALSE),]
colnames(genewise_dearseq_cold_sans)=c("ID","pvalue","rank_pvalue","raw_pvalue")




patients_cold=counts_sans[output_sans==0,]
patients_diffuse=counts_sans[output_sans==1,]
means_diffuse=numeric(length(patients_cold[1,]))
means_cold=numeric(length(patients_cold[1,]))
for (i in 1:length(patients_cold[1,])){
  means_diffuse[i]=mean(patients_diffuse[,i])
  means_cold[i]=mean(patients_cold[,i])
}

fold_change=data.frame(colnames(patients_cold),means_diffuse-means_cold)
colnames(fold_change)=c("ID","log2FoldChange")
total_cold_sans=merge(genewise_dearseq_cold_sans,fold_change,by="ID")
total_cold_sans=total_cold_sans[order(total_cold_sans[,4],total_cold_sans[,5]),]

write.xlsx(total_cold_sans[total_cold_sans[,2]<0.05,], file = "dearseq_RNA_C_D_ng4.xlsx",col.names = TRUE)


total_cold_sans$diffexpressed <- "NO"
# if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP" 
total_cold_sans$diffexpressed[total_cold_sans$log2FoldChange > 0.6 & total_cold_sans$pvalue < 0.05] <- "UP"
# if log2Foldchange <  (-0.6) and pvalue < 0.05, set as "DOWN"
total_cold_sans$diffexpressed[total_cold_sans$log2FoldChange <  (-0.6) & total_cold_sans$pvalue < 0.05] <- "DOWN"

total_cold_sans$delabel <- NA
total_cold_sans$delabel[total_cold_sans$diffexpressed != "NO"] <- total_cold_sans$ID[total_cold_sans$diffexpressed != "NO"]


plot1=ggplot(data=total_cold_sans, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed, label=delabel)) + 
    geom_point() + 
    theme_minimal() +
  geom_text_repel()  + scale_color_manual(values= c("#440154FF", "#1F968BFF", "#DCE319FF")) +ggtitle("Log2 Fold change in function of pvalues")

plot2=ggplot(data=total_cold_sans,aes(x=log2FoldChange))+geom_density()+xlim(c(-2,2))+ theme_minimal()+ggtitle("Distribution of Log2FoldChange") 

```

dearseq identifies `r dim(genewise_dearseq_cold[genewise_dearseq_cold_sans[,2]<0.05,])[1]` genes that are differentially expressed between cold and diffuse. Here are the first genes in term of p-value:

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
rownames(total_cold_sans)=NULL
 kable_classic(kbl(x=total_cold_sans[1:10,c(1,2,3,4,5)],caption = "Cold vs Diffuse genes"),full_width = F, html_font = "Cambria")

```

We can look at the vulcano plot and also the distribution of the log2 Fold change, in the following figure. The vulcano plot shows the log10(pvalue) in function of the log2 Fold change, which allows to see if the significant genes are up regulated or down regulated (cold as base). A positive log2 Fold change indicates that the gene is, on average, more expressed among the diffuse patients, while a negative log2 Fold change indicates that the gene is more expressed among the cold patients. 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,fig.cap="Vulcano plot and distribution of log2 Fold change",cache=FALSE}
plot1+plot2
```

We can observe the brier score associated with these genes. The brier score is an indicator of the prediction ability, between 1 and 0, 0 being a perfect predictor and 1 a poor predictor: 



```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=TRUE}


Y <- output_sans
X <- counts_sans[,colnames(counts_sans)%in%genewise_dearseq_cold_sans[genewise_dearseq_cold_sans[,2]<0.05,1]]
temp=matrix(nrow =dim(X)[1],ncol=dim(X)[2] )
for (i in 1:dim(X)[1]){ # leave one out
  for (j in 1:dim(X)[2]){
    fit <- glm(Y~X,data = data.frame(Y=as.factor(Y[-i]),X = X[-i,j]),family=binomial()) # On apprend sans la ieme observation
    prob <- predict(fit,newdata=data.frame(X=X[i,j]), type="response") # On predit la ieme observation
    temp[i,j] <- prob # vecteur proba prÃ©dite d'appartenir Ã une condition
  }
}

BS_sans <- NULL
for (k in 1:dim(X)[2]){ # gene k
  BS_sans[k] <- brier(Y, temp[,k])$bs # somme de la diffÃ©rence au carrÃ© entre la proba pour chaque obs et la condition
}
BS_sans=cbind(colnames(X),BS_sans)
colnames(BS_sans)=c("ID","BS_v2")
dearseq_brier_sans=merge(genewise_dearseq_cold_sans,BS_sans,by="ID")
dearseq_brier_sans=cbind(dearseq_brier_sans,rank(dearseq_brier_sans[,5]))
dearseq_brier_sans=as.data.frame(dearseq_brier_sans)
colnames(dearseq_brier_sans)=c("ID","pvalue","rank_pvalue","raw_pvalue","BS","rank_BS")
```
```{r,echo=F,eval=T,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Brier score for the genes identified by dearseq for the cold vs diffuse status (without G4)"}
BS_dearseq_cold_sans=as.numeric(as.character(dearseq_brier_sans[,5]))
df <- data.frame(Brier.score = BS_dearseq_cold_sans,
                 rank=rank(BS_dearseq_cold_sans,ties.method = "random"),
                 method=as.factor(rep("dearseq_asym",length(BS_dearseq_cold_sans))))

g_dearseq_cold_sans <- ggplot(df[df[,3]=="dearseq_asym",], aes(x=rank, y=Brier.score)) +
  theme_minimal() +
  # scale_color_manual( values = c("#CCCCCC", grDevices::heat.colors(n=2)[3])) +
  geom_point(size=1.5,alpha=0.6,color="#FF9900") +
  theme(axis.title = element_text(size=16), axis.text = element_text(size=12), legend.title = element_text(size=16), legend.text = element_text(size=14)) +
  #  guides(color = guide_legend(override.aes = list(alpha=1))) +
  geom_rug(alpha=0.5,color="#FF9900")  + 
  ylab("Brier Score")
g_dearseq_cold_sans
```


## {- .unlisted .unnumbered }

We can compare the pvalues of dearseq with or without G4, with a first figure that shows the pvalues, and then a bland altman plot, and a bland altman plot with log scale
```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Comparison of pvalues with or without G4 patients"}

scat_comparaison <- function(data_1,data_2,name_1,name_2){
  data_1 <- as.data.frame(data_1)
  data_1[,2]=data_1[,2]+rep(0.001,length(data_1[,2]))
  data_2 <- as.data.frame(data_2)
  data_2[,2]=data_2[,2]+rep(0.001,length(data_2[,2]))

  data <- merge(data_1,data_2,by="ID")
  colnames(data) <- c("gene","avec","sans")

  b <- ggplot(data, aes(x = avec, y = sans))
  plot <- b + geom_point(size = 0.5, alpha=0.1) + geom_abline(intercept = 0, slope = 1, color="blue") + scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x), labels = trans_format("log10", math_format(10^.x))) + scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x), labels = trans_format("log10", math_format(10^.x))) + annotation_logticks() + xlab(name_1) + ylab(name_2)
  return(plot)
}

scat_comparaison(genewise_dearseq_cold[,c(1,2)],genewise_dearseq_cold_sans[,c(1,2)],"with G4","without G4")

comparaison=merge(genewise_dearseq_cold[,c(1,4)],genewise_dearseq_cold_sans[,c(1,4)],by="ID")
# blandr.draw(comparaison[,2],comparaison[,3], ciDisplay = FALSE , ciShading = FALSE )
# log_avec=log(comparaison[,2]+rep(0.001,length(comparaison[,2])))
# log_sans=log(comparaison[,3]+rep(0.001,length(comparaison[,2])))
# blandr.draw(log_avec,log_sans, ciDisplay = FALSE , ciShading = FALSE )




dfEval <- data.frame(Dupont = comparaison[,2], Dupond = comparaison[,3])

plotbland1=dfEval %>%
  mutate(diff = Dupont - Dupond,
         mean = (Dupont + Dupond)/2) %>%
  mutate(mean_diff = mean(diff, na.rm = T),
         sd_diff = sd(diff, na.rm = T),
         sdsup = mean_diff + 1.96 * sd_diff,
         sdinf = mean_diff - 1.96 * sd_diff) %>%
  ggplot(mapping = aes(x = mean, y = diff)) +
  geom_point(alpha=0.05) +
  geom_line(aes(y = mean_diff, lty = "mean")) +
  geom_line(aes(y = 0), color = "red") +
  geom_line(aes(y = sdsup, lty = "sd")) +
  geom_line(aes(y = sdinf, lty = "sd")) +
  scale_linetype_manual("", c("Mean of differences", "Mean +/- 1.96 sd"), values = c(1,2)) +
  theme_bw() +
  theme(legend.position = "bottom") +
  labs(x = "Mean", y = "Difference")+ggtitle("Bland Altman")


plotbland2=dfEval %>%
  mutate(diff = abs(Dupont - Dupond),
         mean = (Dupont + Dupond)/2) %>%
  mutate(mean_diff = mean(diff, na.rm = T),
         sd_diff = sd(diff, na.rm = T),
         sdsup = mean_diff + 1.96 * sd_diff,
         sdinf = mean_diff - 1.96 * sd_diff) %>%
  ggplot(mapping = aes(x = mean, y = diff)) +
  geom_point(alpha=0.05) +
  geom_line(aes(y = mean_diff, lty = "mean")) +
  geom_line(aes(y = 0), color = "red") +
  geom_line(aes(y = sdsup, lty = "sd")) +
  geom_line(aes(y = sdinf, lty = "sd")) +
  scale_linetype_manual("", c("Mean of differences", "Mean +/- 1.96 sd"), values = c(1,2)) +
  theme_bw() +
  theme(legend.position = "bottom") +
  labs(x = "Mean", y = "|Difference|")+ggtitle("Bland Altman (log scale)")+ scale_x_continuous(trans = 'log10') +
  scale_y_continuous(trans = 'log10')

plotbland1+plotbland2

```
The presence or absence of G4 patients does not changes the pvalues deeply, particulary for the low pvalues.

## Astro vs Oligo
We also have the Astro or Oligo status of the patients, with a total of 24 Astro patients (4 being G4) and 35 Oligo patients. The following table presents the different characteristics: 

## {- .tabset .tabset-pills .unlisted .unnumbered}

### With G4
```{r, eval=T,echo=FALSE,message=FALSE,warning=FALSE}
datas=datas_avec
Age=c(round(mean(datas[datas[,7]=="oligo"&!is.na(datas[,6]),6]),1),round(mean(datas[datas[,7]=="astro"&!is.na(datas[,6]),6]),1))
Female=c(table(datas[datas[,7]=="oligo",5])[1],table(datas[datas[,7]=="astro",5])[1])
Male=c(table(datas[datas[,7]=="oligo",5])[2],table(datas[datas[,7]=="astro",5])[2])
Event=c(table(datas[datas[,7]=="oligo",3])[2],table(datas[datas[,7]=="astro",3])[2])
No_event=c(table(datas[datas[,7]=="oligo",3])[1],table(datas[datas[,7]=="astro",3])[1])
Median_followup=c(round(summary(datas[datas[,7]=="oligo",4])[3],1),round(summary(datas[datas[,7]=="astro",4])[3],1))
oligo_astro_table=rbind(Age,Female,Male,Event,No_event,Median_followup)
colnames(oligo_astro_table)=c("oligo (n=35)","astro (n=24)")
rownames(oligo_astro_table)=c("Age (mean)","Female","Male","Event","No event", "Median follow up (years)")
```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)

 kable_classic(kbl(x=oligo_astro_table,caption = "Oligo vs Astro patients"),full_width = F, html_font = "Cambria")

```
The same PCA analysis can be done, this time by highlighting the oligo vs astro status : 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,results=FALSE, fig.show='hide'}
datas=datas_avec
counts_pca <- raw_counts
counts_pca <- round((counts_pca+1),0)
counts_pca <- vst(t(counts_pca))

counts_pca_scaled <- base::scale(t(counts_pca), center=TRUE, scale=TRUE)

res.pca=PCA(counts_pca_scaled , graph=FALSE)
#plot.PCA(res.pca, axes=c(1, 2), choix="ind")
Astro_oligo=as.factor(datas[,7])
G4=as.factor(datas[,9])
pca_data <- cbind.data.frame(res.pca$ind$coord[, 1:4],as.factor(datas[,7]),rownames(res.pca$ind$coord))
colnames(pca_data)=c("Dim.1","Dim.2","Dim.3","Dim.4","Astro_oligo","ID")
table_coord=res.pca$ind$coord[,c(1,2)]
particuliers=table_coord[table_coord[,1]>quantile(table_coord[,1],0.97)|table_coord[,1]<quantile(table_coord[,1],0.03)|table_coord[,2]>quantile(table_coord[,2],0.98)|table_coord[,2]<quantile(table_coord[,2],0.02),]

for (i in 1:length(pca_data[,1])){
  if (!pca_data[i,6]%in%rownames(particuliers)){
    pca_data[i,6]=""
    }
}
p <- ggplot(pca_data, aes(x=Dim.1, y=Dim.2, colour = as.factor(datas[,7]),shape=G4,label=ID)) +
 # viridis::scale_color_viridis("Cold_diffuse", discrete = TRUE,end = 0.8)
  #  scale_color_manual(values= c("#E7B800", "#2E9FDF"))+
  geom_text_repel() +
  theme_bw() +
  geom_point(size=2, alpha=1) +  guides(color=guide_legend(override.aes = list(alpha=1),title="Oligo vs Astro")) +
  xlab(label=paste0("PC1 (", round(res.pca$eig[1,2], 2),"% of total variance explained)")) +
  ylab(label=paste0("PC2 (", round(res.pca$eig[2,2], 2), "% of total variance explained)")) +
  ggtitle("", subtitle = "First factorial plan")

```


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE, fig.cap="PCA of the RNA data, with G4 patients (vst transformation and scaled data)"}
p
```

We can look at the characteristics of the patients outside of the majority of the patients:

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
datas=datas_avec
table_particulier=datas[datas[,1]%in%rownames(particuliers),]
table_particulier=table_particulier[,c(1,5,6,2,7,4,3)]
table_particulier[,6]=round(table_particulier[,6],2)
rownames(table_particulier)=NULL
```


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=table_particulier,caption = "Characteristics patients outsite of PCA majority"),full_width = F, html_font = "Cambria")

```

### Without G4
```{r, eval=T,echo=FALSE,message=FALSE,warning=FALSE,cache=FALSE}
datas=datas_sans
Age=c(round(mean(datas[datas[,7]=="oligo"&!is.na(datas[,6]),6]),1),round(mean(datas[datas[,7]=="astro"&!is.na(datas[,6]),6]),1))
Female=c(table(datas[datas[,7]=="oligo",5])[1],table(datas[datas[,7]=="astro",5])[1])
Male=c(table(datas[datas[,7]=="oligo",5])[2],table(datas[datas[,7]=="astro",5])[2])
Event=c(table(datas[datas[,7]=="oligo",3])[2],table(datas[datas[,7]=="astro",3])[2])
No_event=c(table(datas[datas[,7]=="oligo",3])[1],table(datas[datas[,7]=="astro",3])[1])
Median_followup=c(round(summary(datas[datas[,7]=="oligo",4])[3],1),round(summary(datas[datas[,7]=="astro",4])[3],1))
oligo_astro_table=rbind(Age,Female,Male,Event,No_event,Median_followup)
colnames(oligo_astro_table)=c("oligo (n=35)","astro (n=20)")
rownames(oligo_astro_table)=c("Age (mean)","Female","Male","Event","No event", "Median follow up (years)")
```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)

 kable_classic(kbl(x=oligo_astro_table,caption = "Oligo vs Astro patients"),full_width = F, html_font = "Cambria")

```
The same PCA analysis can be done, this time by highlighting the oligo vs astro status : 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,results=FALSE, fig.show='hide'}
datas=datas_sans
counts_pca <- raw_counts_sans
counts_pca <- round((counts_pca+1),0)
counts_pca <- vst(t(counts_pca))

counts_pca_scaled <- base::scale(t(counts_pca), center=TRUE, scale=TRUE)

res.pca=PCA(counts_pca_scaled , graph=FALSE)

#plot.PCA(res.pca, axes=c(1, 2), choix="ind")
Astro_oligo=as.factor(datas[,7])
G4=as.factor(datas[,9])
pca_data <- cbind.data.frame(res.pca$ind$coord[, 1:4],as.factor(datas[,7]),rownames(res.pca$ind$coord))
colnames(pca_data)=c("Dim.1","Dim.2","Dim.3","Dim.4","Astro_oligo","ID")
table_coord=res.pca$ind$coord[,c(1,2)]
particuliers=table_coord[table_coord[,1]>quantile(table_coord[,1],0.97)|table_coord[,1]<quantile(table_coord[,1],0.03)|table_coord[,2]>quantile(table_coord[,2],0.98)|table_coord[,2]<quantile(table_coord[,2],0.02),]

for (i in 1:length(pca_data[,1])){
  if (!pca_data[i,6]%in%rownames(particuliers)){
    pca_data[i,6]=""
    }
}
p <- ggplot(pca_data, aes(x=Dim.1, y=Dim.2, colour = as.factor(datas[,7]),label=ID)) +
 # viridis::scale_color_viridis("Cold_diffuse", discrete = TRUE,end = 0.8)
  #  scale_color_manual(values= c("#E7B800", "#2E9FDF"))+
  geom_text_repel() +
  theme_bw() +
  geom_point(size=2, alpha=1) +
  guides(color=guide_legend(override.aes = list(alpha=1),title="Oligo vs Astro")) +
  xlab(label=paste0("PC1 (", round(res.pca$eig[1,2], 2),"% of total variance explained)")) +
  ylab(label=paste0("PC2 (", round(res.pca$eig[2,2], 2), "% of total variance explained)")) +
  ggtitle("", subtitle = "First factorial plan")

```


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE, fig.cap="PCA of the RNA data, without G4 patients (vst transformation and scaled data)"}
p
```

We can look at the characteristics of the patients outside of the majority of the patients:


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
datas=datas_avec
table_particulier=datas[datas[,1]%in%rownames(particuliers),]
table_particulier=table_particulier[,c(1,5,6,2,7,4,3)]
table_particulier[,6]=round(table_particulier[,6],2)
rownames(table_particulier)=NULL
```


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=table_particulier,caption = "Characteristics patients outsite of PCA majority"),full_width = F, html_font = "Cambria")

```

## {- .unlisted .unnumbered }

There seem to be no particular group inside the PCA. We will then look gene per gene if there is a difference between oligo and astro. For that, we use the dearseq package.

## {- .tabset .tabset-pills .unlisted .unnumbered}

### With G4


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
datas=datas_avec
output_oligo=ifelse(datas[,7]=="oligo",0,1)
Y=matrix(output_oligo,ncol=1,nrow=length(output_oligo))

scores=dear_seq(exprmat=t(as.matrix(raw_counts)),variables2test = Y,preprocessed = FALSE,which_test = "asymptotic")


genewise_dearseq_astro=as.data.frame(cbind(rownames(scores$pvals),scores$pvals[,2],rank(scores$pvals[,2]),scores$pvals[,1]))
genewise_dearseq_astro[,1]=as.character(genewise_dearseq_astro[,1])
genewise_dearseq_astro[,2]=as.numeric(as.character(genewise_dearseq_astro[,2]))
genewise_dearseq_astro[,3]=as.numeric(as.character(genewise_dearseq_astro[,3]))
genewise_dearseq_astro[,4]=as.numeric(as.character(genewise_dearseq_astro[,4]))
genewise_dearseq_astro=genewise_dearseq_astro[order(genewise_dearseq_astro[,2],decreasing = FALSE),]
colnames(genewise_dearseq_astro)=c("ID","pvalue","rank_pvalue","raw_pvalue")



patients_astro=counts[output_oligo==0,]
patients_oligo=counts[output_oligo==1,]
means_oligo=numeric(length(patients_oligo[1,]))
means_astro=numeric(length(patients_astro[1,]))
for (i in 1:length(patients_cold[1,])){
  means_oligo[i]=mean(patients_oligo[,i])
  means_astro[i]=mean(patients_astro[,i])
}

fold_change=data.frame(colnames(patients_astro),means_oligo-means_astro)
colnames(fold_change)=c("ID","log2FoldChange")
total_astro=merge(genewise_dearseq_astro,fold_change,by="ID")
total_astro=total_astro[order(total_astro[,4],total_astro[,5]),]

write.xlsx(total_astro[total_astro[,2]<0.05,], file = "dearseq_RNA_O_A_wg4.xlsx",col.names = TRUE)


total_astro$diffexpressed <- "NO"
# if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP" 
total_astro$diffexpressed[total_astro$log2FoldChange > 0.6 & total_astro$pvalue < 0.05] <- "UP"
# if log2Foldchange <  (-0.6) and pvalue < 0.05, set as "DOWN"
total_astro$diffexpressed[total_astro$log2FoldChange <  (-0.6) & total_astro$pvalue < 0.05] <- "DOWN"

total_astro$delabel <- NA
total_astro$delabel[total_astro$diffexpressed != "NO"] <- total_astro$ID[total_astro$diffexpressed != "NO"]


plot1=ggplot(data=total_astro, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed, label=delabel)) + 
    geom_point() + 
    theme_minimal() +
  geom_text_repel()  + scale_color_manual(values= c("#440154FF", "#1F968BFF", "#DCE319FF")) +ggtitle("Log2 Fold change in function of pvalues")

plot2=ggplot(data=total_astro,aes(x=log2FoldChange))+geom_density()+xlim(c(-2,2))+ theme_minimal()+ggtitle("Distribution of Log2FoldChange") 

# Label points with the textxy function from the calibrate plot
# library(calibrate)
# with(subset(total_astro, pvalue<.05 & abs(log2(fold_change))>1), textxy(log2(fold_change), -log10(raw_pvalue), labs=Gene, cex=.8))

```
To identify which genes are differentialy expressed between the oligo and astro groups, we perform dearseq, which finds `r dim(genewise_dearseq_astro[genewise_dearseq_astro[,2]<0.05,])[1]` genes. Here are the first genes in term of p-value:

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
rownames(total_astro)=NULL
 kable_classic(kbl(x=total_astro[1:10,c(1,2,3,4,5)],caption = "Oligo vs Astro genes"),full_width = F, html_font = "Cambria")


```



We can look at the vulcano plot and also the distribution of the log2 Fold change, in the following figure. The vulcano plot shows the log10(pvalue) in function of the log2 Fold change, which allows to see if the significant genes are up regulated or down regulated (astro as base). A positive log2 Fold change indicates that the gene is, on average, more expressed among the oligo patients, while a negative log2 Fold change indicates that the gene is more expressed among the astro patients. 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,fig.cap="Vulcano plot and distribution of log2 Fold change",cache=FALSE}
plot1+plot2
```

We can observe the brier score associated with these genes. The brier score is an indicator of the prediction ability, between 1 and 0, 0 being a perfect predictor and 1 a poor predictor: 
```{r,eval=T,echo=F, message=FALSE,warning=FALSE,cache=TRUE}

datas=datas_avec
Y <- output_oligo
X <- counts[,colnames(counts)%in%genewise_dearseq_astro[genewise_dearseq_astro[,2]<0.05,1]]
temp=matrix(nrow =dim(X)[1],ncol=dim(X)[2] )
for (i in 1:dim(X)[1]){ # leave one out
  for (j in 1:dim(X)[2]){
    fit <- glm(Y~X,data = data.frame(Y=as.factor(Y[-i]),X = X[-i,j]),family=binomial()) # On apprend sans la ieme observation
    prob <- predict(fit,newdata=data.frame(X=X[i,j]), type="response") # On predit la ieme observation
    temp[i,j] <- prob # vecteur proba prÃ©dite d'appartenir Ã une condition
  }
}

BS <- NULL
for (k in 1:dim(X)[2]){ # gene k
  BS[k] <- brier(Y, temp[,k])$bs # somme de la diffÃ©rence au carrÃ© entre la proba pour chaque obs et la condition
}
BS=cbind(colnames(X),BS)
colnames(BS)=c("ID","BS_v2")
dearseq_brier=merge(genewise_dearseq_astro,BS,by="ID")
dearseq_brier=cbind(dearseq_brier,rank(dearseq_brier[,5]))
dearseq_brier=as.data.frame(dearseq_brier)
colnames(dearseq_brier)=c("ID","pvalue","rank_pvalue","raw_pvalue","BS","rank_BS")
```
```{r, message=FALSE,warning=FALSE,echo=FALSE,cache=FALSE, fig.cap="Brier score for the genes identified by dearseq for the astro vs oligo status (with G4)"}
BS_dearseq_astro=as.numeric(as.character(dearseq_brier[,5]))
df <- data.frame(Brier.score = BS_dearseq_astro,
                 rank=rank(BS_dearseq_astro,ties.method = "random"),
                 method=as.factor(rep("dearseq_asym",length(BS_dearseq_astro))))

g_dearseq_astro <- ggplot(df[df[,3]=="dearseq_asym",], aes(x=rank, y=Brier.score)) +
  theme_minimal() +
  # scale_color_manual( values = c("#CCCCCC", grDevices::heat.colors(n=2)[3])) +
  geom_point(size=1.5,alpha=0.6,color="#FF9900") +
  theme(axis.title = element_text(size=16), axis.text = element_text(size=12), legend.title = element_text(size=16), legend.text = element_text(size=14)) +
  #  guides(color = guide_legend(override.aes = list(alpha=1))) +
  geom_rug(alpha=0.5,color="#FF9900")  + 
  ylab("Brier Score")
g_dearseq_astro
```



### Without G4


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
datas=datas_sans
output_oligo_sans=ifelse(datas_sans[,7]=="oligo",0,1)
Y=matrix(output_oligo_sans,ncol=1,nrow=length(output_oligo_sans))

scores_sans=dear_seq(exprmat=t(as.matrix(raw_counts_sans)),variables2test = Y,preprocessed = FALSE,which_test = "asymptotic")


genewise_dearseq_astro_sans=as.data.frame(cbind(rownames(scores_sans$pvals),scores_sans$pvals[,2],rank(scores_sans$pvals[,2]),scores_sans$pvals[,1]))
genewise_dearseq_astro_sans[,1]=as.character(genewise_dearseq_astro_sans[,1])
genewise_dearseq_astro_sans[,2]=as.numeric(as.character(genewise_dearseq_astro_sans[,2]))
genewise_dearseq_astro_sans[,3]=as.numeric(as.character(genewise_dearseq_astro_sans[,3]))
genewise_dearseq_astro_sans[,4]=as.numeric(as.character(genewise_dearseq_astro_sans[,4]))
genewise_dearseq_astro_sans=genewise_dearseq_astro_sans[order(genewise_dearseq_astro_sans[,2],decreasing = FALSE),]
colnames(genewise_dearseq_astro_sans)=c("ID","pvalue","rank_pvalue","raw_pvalue")




patients_astro=counts_sans[output_oligo_sans==0,]
patients_oligo=counts_sans[output_oligo_sans==1,]
means_oligo=numeric(length(patients_oligo[1,]))
means_astro=numeric(length(patients_astro[1,]))
for (i in 1:length(patients_cold[1,])){
  means_oligo[i]=mean(patients_oligo[,i])
  means_astro[i]=mean(patients_astro[,i])
}

fold_change=data.frame(colnames(patients_astro),means_oligo-means_astro)
colnames(fold_change)=c("ID","log2FoldChange")
total_astro_sans=merge(genewise_dearseq_astro_sans,fold_change,by="ID")
total_astro_sans=total_astro_sans[order(total_astro_sans[,4],total_astro_sans[,5]),]

write.xlsx(total_astro_sans[total_astro_sans[,2]<0.05,], file = "dearseq_RNA_O_A_ng4.xlsx",col.names = TRUE)


total_astro_sans$diffexpressed <- "NO"
# if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP" 
total_astro_sans$diffexpressed[total_astro_sans$log2FoldChange > 0.6 & total_astro_sans$pvalue < 0.05] <- "UP"
# if log2Foldchange <  (-0.6) and pvalue < 0.05, set as "DOWN"
total_astro_sans$diffexpressed[total_astro_sans$log2FoldChange <  (-0.6) & total_astro_sans$pvalue < 0.05] <- "DOWN"

total_astro_sans$delabel <- NA
total_astro_sans$delabel[total_astro_sans$diffexpressed != "NO"] <- total_astro_sans$ID[total_astro_sans$diffexpressed != "NO"]


plot1=ggplot(data=total_astro_sans, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed, label=delabel)) + 
    geom_point() + 
    theme_minimal() +
  geom_text_repel()  + scale_color_manual(values= c("#440154FF", "#1F968BFF", "#DCE319FF")) +ggtitle("Log2 Fold change in function of pvalues")

plot2=ggplot(data=total_astro_sans,aes(x=log2FoldChange))+geom_density()+xlim(c(-2,2))+ theme_minimal()+ggtitle("Distribution of Log2FoldChange") 

# Label points with the textxy function from the calibrate plot
# library(calibrate)
# with(subset(total_astro_sans, pvalue<.05 & abs(log2(fold_change))>1), textxy(log2(fold_change), -log10(raw_pvalue), labs=Gene, cex=.8))

```
To identify which genes are differentialy expressed between the oligo and astro groups, we perform dearseq, which finds `r dim(genewise_dearseq_astro_sans[genewise_dearseq_astro_sans[,2]<0.05,])[1]` genes. Here are the first genes in term of p-value:

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
rownames(total_astro_sans)=NULL
 kable_classic(kbl(x=total_astro_sans[1:10,c(1,2,3,4,5)],caption = "Oligo vs Astro genes"),full_width = F, html_font = "Cambria")


```



We can look at the vulcano plot and also the distribution of the log2 Fold change, in the following figure. The vulcano plot shows the log10(pvalue) in function of the log2 Fold change, which allows to see if the significant genes are up regulated or down regulated (astro as base). A positive log2 Fold change indicates that the gene is, on average, more expressed among the oligo patients, while a negative log2 Fold change indicates that the gene is more expressed among the astro patients. 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,fig.cap="Vulcano plot and distribution of log2 Fold change",cache=FALSE}
plot1+plot2
```

We can observe the brier score associated with these genes. The brier score is an indicator of the prediction ability, between 1 and 0, 0 being a perfect predictor and 1 a poor predictor: 
```{r,eval=T,echo=F, message=FALSE,warning=FALSE,cache=TRUE }
datas=datas_sans

Y <- output_oligo_sans
X <- counts_sans[,colnames(counts_sans)%in%genewise_dearseq_astro_sans[genewise_dearseq_astro_sans[,2]<0.05,1]]
temp=matrix(nrow =dim(X)[1],ncol=dim(X)[2] )
for (i in 1:dim(X)[1]){ # leave one out
  for (j in 1:dim(X)[2]){
    fit <- glm(Y~X,data = data.frame(Y=as.factor(Y[-i]),X = X[-i,j]),family=binomial()) # On apprend sans la ieme observation
    prob <- predict(fit,newdata=data.frame(X=X[i,j]), type="response") # On predit la ieme observation
    temp[i,j] <- prob # vecteur proba prÃ©dite d'appartenir Ã une condition
  }
}

BS_sans <- NULL
for (k in 1:dim(X)[2]){ # gene k
  BS_sans[k] <- brier(Y, temp[,k])$bs # somme de la diffÃ©rence au carrÃ© entre la proba pour chaque obs et la condition
}
BS_sans=cbind(colnames(X),BS_sans)
colnames(BS_sans)=c("ID","BS_v2")
dearseq_brier_sans=merge(genewise_dearseq_astro_sans,BS_sans,by="ID")
dearseq_brier_sans=cbind(dearseq_brier_sans,rank(dearseq_brier_sans[,5]))
dearseq_brier_sans=as.data.frame(dearseq_brier_sans)

colnames(dearseq_brier_sans)=c("ID","pvalue","rank_pvalue","raw_pvalue","BS","rank_BS")
```
```{r, message=FALSE,warning=FALSE,echo=FALSE,cache=FALSE, fig.cap="Brier score for the genes identified by dearseq for the astro vs oligo status (without G4)"}
BS_dearseq_astro_sans=as.numeric(as.character(dearseq_brier_sans[,5]))
df <- data.frame(Brier.score = BS_dearseq_astro_sans,
                 rank=rank(BS_dearseq_astro_sans,ties.method = "random"),
                 method=as.factor(rep("dearseq_asym",length(BS_dearseq_astro_sans))))

g_dearseq_astro_sans <- ggplot(df[df[,3]=="dearseq_asym",], aes(x=rank, y=Brier.score)) +
  theme_minimal() +
  # scale_color_manual( values = c("#CCCCCC", grDevices::heat.colors(n=2)[3])) +
  geom_point(size=1.5,alpha=0.6,color="#FF9900") +
  theme(axis.title = element_text(size=16), axis.text = element_text(size=12), legend.title = element_text(size=16), legend.text = element_text(size=14)) +
  #  guides(color = guide_legend(override.aes = list(alpha=1))) +
  geom_rug(alpha=0.5,color="#FF9900")  + ylab("Brier Score")

g_dearseq_astro_sans
```


## {- .unlisted .unnumbered }

We can compare the pvalues of dearseq with or without G4, with a first figure that shows the pvalues, and then a bland altman plot, and a bland altman plot with log scale
```{r,eval=T,echo=F,message=FALSE,warning=FALSE,fig.cap="comparison of pvalues with or without G4 patients"}


scat_comparaison <- function(data_1,data_2,name_1,name_2){
  data_1 <- as.data.frame(data_1)
  data_1[,2]=data_1[,2]+rep(0.001,length(data_1[,2]))
  data_2 <- as.data.frame(data_2)
  data_2[,2]=data_2[,2]+rep(0.001,length(data_2[,2]))

  data <- merge(data_1,data_2,by="ID")
  colnames(data) <- c("gene","avec","sans")

  b <- ggplot(data, aes(x = avec, y = sans))
  plot <- b + geom_point(size = 0.5, alpha=0.1) + geom_abline(intercept = 0, slope = 1, color="blue") + scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x), labels = trans_format("log10", math_format(10^.x))) + scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x), labels = trans_format("log10", math_format(10^.x))) + annotation_logticks() + xlab(name_1) + ylab(name_2)
  return(plot)
}

scat_comparaison(genewise_dearseq_astro[,c(1,2)],genewise_dearseq_astro_sans[,c(1,2)],"with G4","without G4")+ggtitle("comparison of adjusted pvalues")
comparaison=merge(genewise_dearseq_astro[,c(1,4)],genewise_dearseq_astro_sans[,c(1,4)],by="ID")
# blandr.draw(comparaison[,2],comparaison[,3], ciDisplay = FALSE , ciShading = FALSE )
# log_avec=log(comparaison[,2]+rep(0.001,length(comparaison[,2])))
# log_sans=log(comparaison[,3]+rep(0.001,length(comparaison[,2])))
# blandr.draw(log_avec,log_sans, ciDisplay = FALSE , ciShading = FALSE )




dfEval <- data.frame(Dupont = comparaison[,2], Dupond = comparaison[,3])

plotbland1=dfEval %>%
  mutate(diff = Dupont - Dupond,
         mean = (Dupont + Dupond)/2) %>%
  mutate(mean_diff = mean(diff, na.rm = T),
         sd_diff = sd(diff, na.rm = T),
         sdsup = mean_diff + 1.96 * sd_diff,
         sdinf = mean_diff - 1.96 * sd_diff) %>%
  ggplot(mapping = aes(x = mean, y = diff)) +
  geom_point(alpha=0.05) +
  geom_line(aes(y = mean_diff, lty = "mean")) +
  geom_line(aes(y = 0), color = "red") +
  geom_line(aes(y = sdsup, lty = "sd")) +
  geom_line(aes(y = sdinf, lty = "sd")) +
  scale_linetype_manual("", c("Mean of differences", "Mean +/- 1.96 sd"), values = c(1,2)) +
  theme_bw() +
  theme(legend.position = "bottom") +
  labs(x = "Mean", y = "Difference")+ggtitle("Bland Altman")


plotbland2=dfEval %>%
  mutate(diff = abs(Dupont - Dupond),
         mean = (Dupont + Dupond)/2) %>%
  mutate(mean_diff = mean(diff, na.rm = T),
         sd_diff = sd(diff, na.rm = T),
         sdsup = mean_diff + 1.96 * sd_diff,
         sdinf = mean_diff - 1.96 * sd_diff) %>%
  ggplot(mapping = aes(x = mean, y = diff)) +
  geom_point(alpha=0.05) +
  geom_line(aes(y = mean_diff, lty = "mean")) +
  geom_line(aes(y = 0), color = "red") +
  geom_line(aes(y = sdsup, lty = "sd")) +
  geom_line(aes(y = sdinf, lty = "sd")) +
  scale_linetype_manual("", c("Mean of differences", "Mean +/- 1.96 sd"), values = c(1,2)) +
  theme_bw() +
  theme(legend.position = "bottom") +
  labs(x = "Mean", y = "|Difference|")+ggtitle("Bland Altman (log scale)")+ scale_x_continuous(trans = 'log10') +
  scale_y_continuous(trans = 'log10')

plotbland1+plotbland2

```

## Survival analysis {.tabset .tabset-pills}
When can first perform a simple survival analysis by using a cox model on the base covariates: 

### With G4

```{r, message=FALSE,warning=FALSE}
datas=datas_avec
modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=datas)
summary(modele_base)

```

The p-value not significant for any variables, and the AIC of the model is `r extractAIC(modele_base)[2]`.


### Without G4 

```{r, message=FALSE,warning=FALSE}
datas=datas_sans
modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=datas)
summary(modele_base)

```

The p-value is is significant only for the age variable, and the AIC of the model is `r extractAIC(modele_base)[2]`.

## {- .tabset .tabset-pills .unlisted .unnumbered}


We are now interested to identify the genes that are linked with the risk of recurrence. For that, we use random forest methods. 

### With G4

```{r,eval=T,echo=F, message=FALSE,warning=FALSE,cache=TRUE,fig.cap="First genes in term of importances for the random forests (With G4)"}
datas=datas_avec
 v.obj_baseline<- rfsrc(Surv(time=as.numeric(Time), event=as.numeric(Reccurence))~.,data=datas[,-c(1,8,9)],ntree=2000,block.size=50,mtry=dim(datas[,1])[2])
   importance_baseline=vimp.rfsrc(v.obj_baseline)


   datas_rf=cbind(datas[,-c(1,8,9)],norm_counts)
   v.obj<- rfsrc(Surv(time=as.numeric(Time), event=as.numeric(Reccurence))~.,data=datas_rf,ntree=2000,block.size=50,mtry=dim(datas_rf)[2])
   importance=vimp.rfsrc(v.obj)
   importances=importance$importance
   selected=names(importances[importances>=0.0003])
   table_rf=as.data.frame(cbind(selected,importances[importances>=0.0003]))

   colnames(table_rf)=c("ID","importance")
   write.xlsx(table_rf, file = "survival_RNA_wg4.xlsx",col.names = TRUE)


total_importances=as.data.frame(cbind(names(importances),importances))

 dearseq=genewise_dearseq_cold
 results_rf=merge(table_rf,dearseq)
 results_rf=results_rf[,c(1,2,3,5)]
 results_rf[,2]=round(as.numeric(results_rf[,2]),digits = 5)



 results_rf=results_rf[order(results_rf[,2],decreasing = TRUE),]
 results_rf[,1]=factor(results_rf[,1],levels=c(results_rf[1:dim(results_rf)[1],1]))

 ggplot(results_rf,aes(x=ID,y=importance))+
   geom_point()+ scale_colour_viridis_d() + xlab("selected genes with survival random forest")+ theme(axis.text.x = element_blank())
```

We have `r length(selected)` genes with an importance>0.0003 selected by the random forest method.


### Without G4

```{r,eval=T,echo=F, message=FALSE,warning=FALSE,cache=TRUE,fig.cap="First genes in term of importances for the random forests (without G4)"}
datas=datas_sans
 v.obj_baseline_sans<- rfsrc(Surv(time=as.numeric(Time), event=as.numeric(Reccurence))~.,data=datas_sans[,-c(1,8,9)],ntree=2000,block.size=50,mtry=dim(datas_sans[,1])[2])
  importance_baseline=vimp.rfsrc(v.obj_baseline_sans)
  
  
 





   datas_rf_sans=cbind(datas[,-c(1,8,9)],norm_counts_sans)
   v.obj_sans<- rfsrc(Surv(time=as.numeric(Time), event=as.numeric(Reccurence))~.,data=datas_rf_sans,ntree=2000,block.size=50,mtry=dim(datas_rf_sans)[2])
   importance_sans=vimp.rfsrc(v.obj_sans)
   importances_sans=importance_sans$importance
   selected_sans=names(importances_sans[importances_sans>=0.0003])
   table_rf_sans=as.data.frame(cbind(selected_sans,importances_sans[importances_sans>=0.0003]))

   colnames(table_rf_sans)=c("ID","importance")
   write.xlsx(table_rf_sans, file = "survival_RNA_ng4.xlsx",col.names = TRUE)


total_importances_sans=as.data.frame(cbind(names(importances_sans),importances_sans))

 dearseq_sans=genewise_dearseq_cold_sans
 results_rf_sans=merge(table_rf_sans,dearseq_sans)
 results_rf_sans=results_rf_sans[,c(1,2,3,5)]
 results_rf_sans[,2]=round(as.numeric(results_rf_sans[,2]),digits = 5)



 results_rf_sans=results_rf_sans[order(results_rf_sans[,2],decreasing = TRUE),]
 results_rf_sans[,1]=factor(results_rf_sans[,1],levels=c(results_rf_sans[1:dim(results_rf_sans)[1],1]))

 ggplot(results_rf_sans,aes(x=ID,y=importance))+
   geom_point()+ scale_colour_viridis_d() + xlab("selected genes with survival random forest")+ theme(axis.text.x = element_blank())

```

We have `r length(selected_sans)` genes with an importance>0.0003 selected by the random forest method.

## Link between recurrence and cold vs diffuse {.tabset .tabset-pills}
We can explore if the genes identified by the random forest are also linked with the recurrence risk. For that, we compute three indicators for each gene identify by the random forest: the dearseq pvalue, the fold change (cold as base, on log CPM), and the brier score, linked with the cold vs diffuse. 

### With G4

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="First genes in term of importance for the survival, and their different criterion liked with cold and diffuse status (dearseq pvalue, fold change on log CPM, brier score)"}
datas=datas_avec
dearseq=genewise_dearseq_cold
results_rf=merge(table_rf,dearseq)
results_rf=results_rf[,c(1,2,3,5)]
results_rf[,2]=round(as.numeric(results_rf[,2]),digits = 5)



patients_cold=counts[output==0,colnames(counts)%in%results_rf[,1]]
patients_diffuse=counts[output==1,colnames(counts)%in%results_rf[,1]]
means_diffuse=numeric(length(patients_cold[1,]))
means_cold=numeric(length(patients_cold[1,]))
for (i in 1:length(patients_cold[1,])){
  means_diffuse[i]=mean(patients_diffuse[,i])
  means_cold[i]=mean(patients_cold[,i])
}

fold_change=data.frame(colnames(patients_cold),means_diffuse-means_cold)
colnames(fold_change)=c("ID","fold_change")

results_rf=merge(results_rf,fold_change,by="ID")
results_rf[,2]=round(as.numeric(results_rf[,2]),digits = 5)
Y <- output
X <- counts[,colnames(counts)%in%results_rf[,1]]
temp=matrix(nrow =dim(X)[1],ncol=dim(X)[2] )
for (i in 1:dim(X)[1]){ # leave one out
  for (j in 1:dim(X)[2]){
    fit <- glm(Y~X,data = data.frame(Y=as.factor(Y[-i]),X = X[-i,j]),family=binomial()) # On apprend sans la ieme observation
    prob <- predict(fit,newdata=data.frame(X=X[i,j]), type="response") # On predit la ieme observation
    temp[i,j] <- prob # vecteur proba prÃ©dite d'appartenir Ã une condition
  }
}

BS <- NULL
for (k in 1:dim(X)[2]){ # gene k
  BS[k] <- brier(Y, temp[,k])$bs # somme de la diffÃ©rence au carrÃ© entre la proba pour chaque obs et la condition
}
BS=data.frame(colnames(X),BS)
colnames(BS)=c("ID","BS")

results_rf=merge(results_rf,BS,by="ID")
results_rf=results_rf[order(results_rf[,2],decreasing=TRUE),]
results_rf=results_rf[!is.na(results_rf[,1]),]
colnames(results_rf)=c("ID","importance","dearseq","dearseq_raw","fold_change","BS")
results_rf[,1]=factor(results_rf[,1],levels=c(results_rf[1:dim(results_rf)[1],1]))



datas_total=melt(results_rf,id.var="ID")

ggplot(datas_total[datas_total[,2]!="dearseq",],aes(x=ID,y=value))+
  geom_point()+
  facet_grid(variable~.,scale="free")+ scale_colour_viridis_d() + xlab("selected genes with survival random forest")+ theme(axis.text.x = element_blank())

```

We can identify some genes that are selected by the random forest and also linked with the cold vs diffuse status. For that, we look at the genes with a high (or low) fold change, and also those who are significant for dearseq. 

```{r,echo=F,message=FALSE,warning=FALSE,fig.cap="Genes selected by the random forest (survival), and linked with cold/diffuse status (in red)"}
liste=results_rf[results_rf[,3]<0.05|results_rf[,5]>0.6|results_rf[,5]< (-0.6),1]
color=numeric(length(datas_total[,1]))
color=ifelse(datas_total[,1]%in%liste,"red","black")
datas_highlight=as.data.frame(cbind(datas_total,color))
ggplot(datas_highlight[datas_highlight[,2]!="dearseq",],aes(x=ID,y=value))+
  geom_point(color=color[datas_highlight[,2]!="dearseq"])+
  facet_grid(variable~.,scale="free")+ scale_colour_viridis_d() + xlab("selected genes with survival random forest")+ theme(axis.text.x = element_blank())

```

A total of `r length(liste)` genes are selected to have a link with both survival and cold vs diffuse status: the genes `r as.character(liste)`

### Without G4

```{r,echo=F,message=FALSE,warning=FALSE,cache=TRUE,fig.cap="First genes in term of importance for the survival, and their different criterion liked with cold and diffuse status (dearseq pvalue, fold change on log CPM, brier score)"}
datas=datas_sans
dearseq_sans=genewise_dearseq_cold_sans
results_rf_sans=merge(table_rf_sans,dearseq_sans)
results_rf_sans=results_rf_sans[,c(1,2,3,5)]
results_rf_sans[,2]=round(as.numeric(results_rf_sans[,2]),digits = 5)



patients_cold_sans=counts_sans[output_sans==0,colnames(counts_sans)%in%results_rf_sans[,1]]
patients_diffuse_sans=counts_sans[output_sans==1,colnames(counts_sans)%in%results_rf_sans[,1]]
means_diffuse_sans=numeric(length(patients_cold_sans[1,]))
means_cold_sans=numeric(length(patients_cold_sans[1,]))
for (i in 1:length(patients_cold_sans[1,])){
  means_diffuse_sans[i]=mean(patients_diffuse_sans[,i])
  means_cold_sans[i]=mean(patients_cold_sans[,i])
}

fold_change_sans=data.frame(colnames(patients_cold_sans),means_diffuse_sans-means_cold_sans)
colnames(fold_change_sans)=c("ID","fold_change")

results_rf_sans=merge(results_rf_sans,fold_change_sans,by="ID")
results_rf_sans[,2]=round(as.numeric(results_rf_sans[,2]),digits = 5)
Y <- output_sans
X <- counts_sans[,colnames(counts_sans)%in%results_rf_sans[,1]]
temp=matrix(nrow =dim(X)[1],ncol=dim(X)[2] )
for (i in 1:dim(X)[1]){ # leave one out
  for (j in 1:dim(X)[2]){
    fit <- glm(Y~X,data = data.frame(Y=as.factor(Y[-i]),X = X[-i,j]),family=binomial()) # On apprend sans la ieme observation
    prob <- predict(fit,newdata=data.frame(X=X[i,j]), type="response") # On predit la ieme observation
    temp[i,j] <- prob # vecteur proba prÃ©dite d'appartenir Ã une condition
  }
}

BS_sans <- NULL
for (k in 1:dim(X)[2]){ # gene k
  BS_sans[k] <- brier(Y, temp[,k])$bs # somme de la diffÃ©rence au carrÃ© entre la proba pour chaque obs et la condition
}
BS_sans=data.frame(colnames(X),BS_sans)
colnames(BS_sans)=c("ID","BS")

results_rf_sans=merge(results_rf_sans,BS_sans,by="ID")
results_rf_sans=results_rf_sans[order(results_rf_sans[,2],decreasing=TRUE),]
results_rf_sans=results_rf_sans[!is.na(results_rf_sans[,1]),]
colnames(results_rf_sans)=c("ID","importance","dearseq","dearseq_raw","fold_change","BS")
results_rf_sans[,1]=factor(results_rf_sans[,1],levels=c(results_rf_sans[1:dim(results_rf_sans)[1],1]))



datas_total_sans=melt(results_rf_sans,id.var="ID")

ggplot(datas_total_sans[datas_total_sans[,2]!="dearseq",],aes(x=ID,y=value))+
  geom_point()+
  facet_grid(variable~.,scale="free")+ scale_colour_viridis_d() + xlab("selected genes with survival random forest")+ theme(axis.text.x = element_blank())

```

We can identify some genes that are selected by the random forest and also linked with the cold vs diffuse status. For that, we look at the genes with a high (or low) fold change, and also those who are significant for dearseq. 

```{r,echo=F,message=FALSE,warning=FALSE,fig.cap="Genes selected by the random forest (survival), and linked with cold/diffuse status (in red)"}
liste_sans=results_rf_sans[results_rf_sans[,3]<0.05|results_rf_sans[,5]>0.6|results_rf_sans[,5]< (-0.6),1]
color_sans=numeric(length(datas_total_sans[,1]))
color_sans=ifelse(datas_total_sans[,1]%in%liste_sans,"red","black")
datas_highlight_sans=as.data.frame(cbind(datas_total_sans,color_sans))
ggplot(datas_highlight_sans[datas_highlight_sans[,2]!="dearseq",],aes(x=ID,y=value))+
  geom_point(color=color_sans[datas_highlight_sans[,2]!="dearseq"])+
  facet_grid(variable~.,scale="free")+ scale_colour_viridis_d() + xlab("selected genes with survival random forest")+ theme(axis.text.x = element_blank())

```

A total of `r length(liste_sans)` genes are selected to have a link with both survival and cold vs diffuse status: the genes `r as.character(liste_sans)`

## {- .unlisted .unnumbered .tabset .tabset-pills}

As the random forest are, random, the results can change from one time to another. We computed multiple times (7) the random forest and looked at the genes that were similar from one iteration to the other 

### With G4

```{r,echo=F,message=FALSE,warning=FALSE,cache=TRUE}
results_tot=read.table("data/results_rf.txt")
liste_selected=as.list(NULL)
plots=as.list(NULL)
plots_selected=as.list(NULL)
results=results_tot[results_tot[,3]==1,]
results=results[!duplicated(results[,1]),]
results=results[,c(1,2)]
colnames(results)=c("ID","importance")
liste_merged=results
for (i in 2:8){
  results=results_tot[results_tot[,3]==i,]
  results=results[!duplicated(results[,1]),]
  results=results[,c(1,2)]
  colnames(results)=c("ID","importance")
  liste_merged=merge(liste_merged,results,by="ID")
}
liste_merged_tot=liste_merged
liste_merged=liste_merged[,1]
for (test in 1:8){
  results=results_tot[results_tot[,3]==test,]
  results=results[!duplicated(results[,1]),]
  results=results[,c(1,2)]
  colnames(results)=c("ID","importance")
  dearseq=genewise_dearseq_cold
  results_rf=merge(results,dearseq)
  results_rf=results_rf[,c(1,2,3,5)]
  patients_cold=norm_counts[output==0,colnames(counts)%in%results_rf[,1]]
  patients_diffuse=norm_counts[output==1,colnames(counts)%in%results_rf[,1]]
  means_diffuse=numeric(length(patients_cold[1,]))
  means_cold=numeric(length(patients_cold[1,]))
  for (i in 1:length(patients_cold[1,])){
    means_diffuse[i]=mean(patients_diffuse[,i])
    means_cold[i]=mean(patients_cold[,i])
  }
  fold_change=means_diffuse-means_cold
  results_rf=cbind(results_rf,fold_change)
  results_rf[,2]=round(as.numeric(results_rf[,2]),digits = 5)
  summary(results_rf[,3])
  #### BRIER SCORE : capacitÃ© predictive d'un gÃ¨ne
  # proche de 0 => bon predicteur
  # proche de 1 => mauvais predicteur
  Y <- output
  X <- counts[,colnames(counts)%in%results_rf[,1]]
  temp=matrix(nrow =dim(X)[1],ncol=dim(X)[2] )
  for (i in 1:dim(X)[1]){ # leave one out
    for (j in 1:dim(X)[2]){
      fit <- glm(Y~X,data = data.frame(Y=as.factor(Y[-i]),X = X[-i,j]),family=binomial()) # On apprend sans la ieme observation
      prob <- predict(fit,newdata=data.frame(X=X[i,j]), type="response") # On predit la ieme observation
      temp[i,j] <- prob # vecteur proba prÃ©dite d'appartenir Ã une condition
    }
  }
  BS <- NULL
  for (k in 1:dim(X)[2]){ # gene k
    BS[k] <- brier(Y, temp[,k])$bs # somme de la diffÃ©rence au carrÃ© entre la proba pour chaque obs et la condition
  }
  BS=cbind(colnames(X),BS)
  colnames(BS)=c("ID","BS_v2")
  results_rf=cbind(results_rf,as.numeric(BS[,2]))
  results_rf=results_rf[order(results_rf[,2],decreasing=TRUE),]
  results_rf=results_rf[!is.na(results_rf[,1]),]
  colnames(results_rf)=c("ID","importance","dearseq","dearseq_raw","fold_change","BS")
  datas_total=melt(results_rf,id.var="ID")
  datas_total[,1]=factor(datas_total[,1],levels=c(datas_total[1:dim(results_rf)[1],1]))
  liste=results_rf[results_rf[,3]<0.05|results_rf[,5]>0.6|results_rf[,5]<(-0.6),1]
  liste_selected[[test]]=liste
  color=numeric(length(datas_total[,1]))
  #color=ifelse(datas_total[,1]%in%liste,"red","black")
  color=ifelse(datas_total[,1]%in%liste_merged,"blue","black")
  datas_highlight=as.data.frame(cbind(datas_total,color))
  plots[[test]]=ggplot(datas_highlight[datas_highlight[,2]!="dearseq",],aes(x=ID,y=value))+
    geom_point(color=color[datas_highlight[,2]!="dearseq"])+
    facet_grid(variable~.,scale="free")+ scale_colour_viridis_d() + xlab("selected genes with survival random forest")+ theme(axis.text.x = element_blank())
 
  
  liste=results_rf[results_rf[,3]<0.05|results_rf[,5]>0.6|results_rf[,5]<(-0.6),1]
  color=numeric(length(datas_total[,1]))
  color=ifelse(datas_total[,1]%in%liste,"red","black")
  datas_highlight=as.data.frame(cbind(datas_total,color))
  plots_selected[[test]]=ggplot(datas_highlight[datas_highlight[,2]!="dearseq",],aes(x=ID,y=value))+
    geom_point(color=color[datas_highlight[,2]!="dearseq"])+
    facet_grid(variable~.,scale="free")+ scale_colour_viridis_d() + xlab("selected genes with survival random forest")+ theme(axis.text.x = element_blank())
  
  
  }
plot1=(plots[[1]]+plots[[2]])/(plots[[3]]+plots[[4]])
plot2=(plots_selected[[1]]+plots_selected[[2]])/(plots_selected[[3]]+plots_selected[[4]])
```

We can first see which genes are always found as important genes (here in blue)
```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Repetition of the random forest, with the genes always found (in blue)"}
plot1=(plots[[1]]+plots[[2]])/(plots[[3]]+plots[[4]])

plot1
```

Here is a table that show with genes that are always identified and their mean importance, followed by a graph showing how often a gene is identified among all tries
```{r,echo=F,message=FALSE,warning=FALSE,fig.cap="How often each gene is found in the total repetitions"}
means_tot=numeric(length(liste_merged_tot[,1]))
for (i in 1:length(liste_merged_tot[,1])){
  means_tot[i]=mean(as.numeric(liste_merged_tot[i,-1]))
}
liste_found_avec=as.data.frame(cbind(liste_merged_tot[,1],means_tot))
colnames(liste_found_avec)=c("gene","mean importance")
kable_classic(kbl(x=liste_found_avec,caption = "Genes always found and their mean importance"),full_width = F, html_font = "Cambria")


results=results_tot[results_tot[,3]==1,]
results=results[!duplicated(results[,1]),]
results=results[,c(1,2)]
colnames(results)=c("ID","importance")
liste_merged2=results
for (i in 2:7){
  results=results_tot[results_tot[,3]==i,]
  results=results[!duplicated(results[,1]),]
  results=results[,c(1,2)]
  colnames(results)=c("ID","importance")
  liste_merged2=merge(liste_merged2,results,all=TRUE)
}

variables=unique(liste_merged2[,1])
percent=numeric(length(variables))
mean_imp=numeric(length(variables))
for (i in 1:length(variables)){
  results_variable=liste_merged2[liste_merged2[,1]==variables[i],]
  percent[i]=100*round(dim(results_variable)[1]/7,2)
  mean_imp[i]=mean(results_variable[,2])
}

table_tot=as.data.frame(cbind(variables,percent,round(mean_imp,6)))

colnames(table_tot)=c("ID","percent","importance")
table_tot=merge(table_tot,total_cold,by="ID")
table_tot=table_tot[,c(1,2,3,4,7,8)]
table_tot=merge(table_tot,total_astro,by="ID")
table_tot=table_tot[,c(1,2,3,4,5,6,7,10,11)]
colnames(table_tot)=c("ID","percent","RF_importance","dearseq_cold_diffuse","Log2FC_cold_diffuse","diff_expr_cold_diffuse","dearseq_astro_oligo","Log2FC_astro_oligo","diff_expr_astro_oligo")
table_tot[,2]=as.numeric(table_tot[,2])
 
table_tot=table_tot[order(table_tot[,2],table_tot[,3],decreasing = TRUE),]

write.xlsx(table_tot,"survival_total_RNA.xlsx",row.names = FALSE,col.names = TRUE)


table_nombres=table(liste_merged2[,1])
table_nombres=as.data.frame(cbind(rep(1,length(table_nombres)),table_nombres))
colnames(table_nombres)=c("number of identifications","occurences")
ggplot(as.data.frame(table_nombres),aes(x=occurences)) + 
  geom_histogram()
```


But can also see which genes are identified as linked with cold/diffuse status (here in red)
```{r,echo=F,message=FALSE,warning=FALSE,fig.cap="Repetition of the random forest, with the genes identified as linked with cold and diffuse status (in red)"}
plot2
```


### Without G4

```{r,echo=F,message=FALSE,warning=FALSE,cache=TRUE}

results_tot=read.table("data/results_rf_sans.txt")
liste_selected=as.list(NULL)
plots=as.list(NULL)
results=results_tot[results_tot[,3]==1,]
results=results[!duplicated(results[,1]),]
results=results[,c(1,2)]
colnames(results)=c("ID","importance")
liste_merged=results
for (i in 2:7){
  results=results_tot[results_tot[,3]==i,]
  results=results[!duplicated(results[,1]),]
  results=results[,c(1,2)]
  colnames(results)=c("ID","importance")
  liste_merged=merge(liste_merged,results,by="ID")
}
liste_merged_tot=liste_merged

liste_merged=liste_merged[,1]
for (test in 1:7){
  results=results_tot[results_tot[,3]==test,]
  results=results[!duplicated(results[,1]),]
  results=results[,c(1,2)]
  colnames(results)=c("ID","importance")
  dearseq=genewise_dearseq_cold_sans
  results_rf=merge(results,dearseq)
  results_rf=results_rf[,c(1,2,3,5)]
  patients_cold_sans=norm_counts_sans[output_sans==0,colnames(counts_sans)%in%results_rf[,1]]
  patients_diffuse_sans=norm_counts_sans[output_sans==1,colnames(counts_sans)%in%results_rf[,1]]
  means_diffuse_sans=numeric(length(patients_cold_sans[1,]))
  means_cold_sans=numeric(length(patients_cold_sans[1,]))
  for (i in 1:length(patients_cold_sans[1,])){
    means_diffuse_sans[i]=mean(patients_diffuse_sans[,i])
    means_cold_sans[i]=mean(patients_cold_sans[,i])
  }
  fold_change=means_diffuse_sans-means_cold_sans
  results_rf=cbind(results_rf,fold_change)
  results_rf[,2]=round(as.numeric(results_rf[,2]),digits = 5)
  summary(results_rf[,3])
  #### BRIER SCORE : capacitÃ© predictive d'un gÃ¨ne
  # proche de 0 => bon predicteur
  # proche de 1 => mauvais predicteur
  Y <- output_sans
  X <- counts_sans[,colnames(counts_sans)%in%results_rf[,1]]
  temp=matrix(nrow =dim(X)[1],ncol=dim(X)[2] )
  for (i in 1:dim(X)[1]){ # leave one out
    for (j in 1:dim(X)[2]){
      fit <- glm(Y~X,data = data.frame(Y=as.factor(Y[-i]),X = X[-i,j]),family=binomial()) # On apprend sans la ieme observation
      prob <- predict(fit,newdata=data.frame(X=X[i,j]), type="response") # On predit la ieme observation
      temp[i,j] <- prob # vecteur proba prÃ©dite d'appartenir Ã une condition
    }
  }
  BS <- NULL
  for (k in 1:dim(X)[2]){ # gene k
    BS[k] <- brier(Y, temp[,k])$bs # somme de la diffÃ©rence au carrÃ© entre la proba pour chaque obs et la condition
  }
  BS=cbind(colnames(X),BS)
  colnames(BS)=c("ID","BS_v2")
  results_rf=cbind(results_rf,as.numeric(BS[,2]))
  results_rf=results_rf[order(results_rf[,2],decreasing=TRUE),]
  results_rf=results_rf[!is.na(results_rf[,1]),]
  colnames(results_rf)=c("ID","importance","dearseq","dearseq_raw","fold_change","BS")
  datas_total=melt(results_rf,id.var="ID")
  datas_total[,1]=factor(datas_total[,1],levels=c(datas_total[1:dim(results_rf)[1],1]))
  liste=results_rf[results_rf[,3]<0.05|results_rf[,5]>0.6|results_rf[,5]<(-0.6),1]
  liste_selected[[test]]=liste
  color=numeric(length(datas_total[,1]))
  #color=ifelse(datas_total[,1]%in%liste,"red","black")
  color=ifelse(datas_total[,1]%in%liste_merged,"blue","black")
  datas_highlight=as.data.frame(cbind(datas_total,color))
  plots[[test]]=ggplot(datas_highlight[datas_highlight[,2]!="dearseq",],aes(x=ID,y=value))+
    geom_point(color=color[datas_highlight[,2]!="dearseq"])+
    facet_grid(variable~.,scale="free")+ scale_colour_viridis_d() + xlab("selected genes with survival random forest")+ theme(axis.text.x = element_blank())
  liste=results_rf[results_rf[,3]<0.05|results_rf[,5]>0.6|results_rf[,5]<(-0.6),1]
  
  color=numeric(length(datas_total[,1]))
  color=ifelse(datas_total[,1]%in%liste,"red","black")
  datas_highlight=as.data.frame(cbind(datas_total,color))
  plots_selected[[test]]=ggplot(datas_highlight[datas_highlight[,2]!="dearseq",],aes(x=ID,y=value))+
    geom_point(color=color[datas_highlight[,2]!="dearseq"])+
    facet_grid(variable~.,scale="free")+ scale_colour_viridis_d() + xlab("selected genes with survival random forest")+ theme(axis.text.x = element_blank())
  
  
  }
plot1=(plots[[1]]+plots[[2]])/(plots[[3]]+plots[[4]])
plot2=(plots_selected[[1]]+plots_selected[[2]])/(plots_selected[[3]]+plots_selected[[4]])

```

We can first see wich genes are always found as important genes (here in blue)
```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Repetition of the random forest, with the genes always found (in blue)"}
plot1=(plots[[1]]+plots[[2]])/(plots[[3]]+plots[[4]])

plot1
```

Here is a table that show with genes that are always identified and their mean importance, followed by a graph showing how often a gene is identified among all tries
```{r,echo=F,message=FALSE,warning=FALSE,fig.cap="How often each gene is found in the total repetitions"}
means_tot=numeric(length(liste_merged_tot[,1]))
for (i in 1:length(liste_merged_tot[,1])){
  means_tot[i]=mean(as.numeric(liste_merged_tot[i,-1]))
}
liste_found_sans=as.data.frame(cbind(liste_merged_tot[,1],means_tot))
colnames(liste_found_sans)=c("gene","mean importance")
kable_classic(kbl(x=liste_found_sans,caption = "Genes always found and their mean importance"),full_width = F, html_font = "Cambria")


results=results_tot[results_tot[,3]==1,]
results=results[!duplicated(results[,1]),]
results=results[,c(1,2)]
colnames(results)=c("ID","importance")
liste_merged2=results
for (i in 2:7){
  results=results_tot[results_tot[,3]==i,]
  results=results[!duplicated(results[,1]),]
  results=results[,c(1,2)]
  colnames(results)=c("ID","importance")
  liste_merged2=merge(liste_merged2,results,all=TRUE)
}

table_nombres=table(liste_merged2[,1])
table_nombres=as.data.frame(cbind(rep(1,length(table_nombres)),table_nombres))
colnames(table_nombres)=c("number of identifications","occurences")
ggplot(as.data.frame(table_nombres),aes(x=occurences)) + 
  geom_histogram()
```

But can also see which genes are identified as linked with cold/diffuse status (here in red)
```{r,echo=F,message=FALSE,warning=FALSE,fig.cap="Repetition of the random forest, with the genes identified as linked with cold and diffuse status (in red)"}
plot2
```

## {- .unliste .unnumbered}

We can also explore the difference between with or without G4 in term of importance, with a comparison of the importances and a bland altman plot, followed by a bland altman plot in log scale

```{r,echo=F,message=FALSE,warning=FALSE,fig.cap="comparison of importances with and without G4 patients"}
scat_comparaison <- function(data_1,data_2,name_1,name_2){
  data_1 <- as.data.frame(data_1)
  data_1[,2]=data_1[,2]
  data_2 <- as.data.frame(data_2)
  data_2[,2]=data_2[,2]

  data <- merge(data_1,data_2,by="variable")
  colnames(data) <- c("protein","avec","sans")

  b <- ggplot(data, aes(x = avec, y = sans))
  plot <- b + geom_point(size = 0.5, alpha=0.1) + geom_abline(intercept = 0, slope = 1, color="blue")+xlab(name_1) + ylab(name_2)+ggtitle("Comparison of the importances for each genes, natural scale")
  return(plot)
}

colnames(total_importances_sans)=c("variable","importance")
colnames(total_importances)=c("variable","importance")
total_importances_sans[,2]=round(as.numeric(total_importances_sans[,2]),digits = 5)
total_importances[,2]=round(as.numeric(total_importances[,2]),digits = 5)

scat_comparaison(total_importances,total_importances_sans,"with G4","without G4")



scat_comparaison <- function(data_1,data_2,name_1,name_2){
  data_1 <- as.data.frame(data_1)
  data_1[,2]=data_1[,2]+rep(0.001,length(data_1[,2]))
  data_2 <- as.data.frame(data_2)
  data_2[,2]=data_2[,2]+rep(0.001,length(data_2[,2]))

  data <- merge(data_1,data_2,by="variable")
  colnames(data) <- c("protein","avec","sans")

  b <- ggplot(data, aes(x = avec, y = sans))
  plot <- b + geom_point(size = 0.5, alpha=0.1) + geom_abline(intercept = 0, slope = 1, color="blue") + scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x), labels = trans_format("log10", math_format(10^.x))) + scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x), labels = trans_format("log10", math_format(10^.x))) + annotation_logticks() + xlab(name_1) + ylab(name_2)+ggtitle("Comparison of the importances for each genes, log scale")
  return(plot)
}

scat_comparaison(total_importances[total_importances[,2]>0,],total_importances_sans[total_importances[,2]>0,],"with G4","without G4")



dfEval <- data.frame(Dupont = total_importances[,2], Dupond = total_importances_sans[,2])



plotbland1=dfEval %>%
  mutate(diff = Dupont - Dupond,
         mean = (Dupont + Dupond)/2) %>%
  mutate(mean_diff = mean(diff, na.rm = T),
         sd_diff = sd(diff, na.rm = T),
         sdsup = mean_diff + 1.96 * sd_diff,
         sdinf = mean_diff - 1.96 * sd_diff) %>%
  ggplot(mapping = aes(x = mean, y = diff)) +
  geom_point(alpha=0.05) +
  geom_line(aes(y = mean_diff, lty = "mean")) +
  geom_line(aes(y = 0), color = "red") +
  geom_line(aes(y = sdsup, lty = "sd")) +
  geom_line(aes(y = sdinf, lty = "sd")) +
  scale_linetype_manual("", c("Mean of differences", "Mean +/- 1.96 sd"), values = c(1,2)) +
  theme_bw() +
  theme(legend.position = "bottom") +
  labs(x = "Mean", y = "Difference")+ggtitle("Bland Altman")


plotbland2=dfEval %>%
  mutate(diff = abs(Dupont - Dupond),
         mean = (Dupont + Dupond)/2) %>%
  mutate(mean_diff = mean(diff, na.rm = T),
         sd_diff = sd(diff, na.rm = T),
         sdsup = mean_diff + 1.96 * sd_diff,
         sdinf = mean_diff - 1.96 * sd_diff) %>%
  ggplot(mapping = aes(x = mean, y = diff)) +
  geom_point(alpha=0.05) +
  geom_line(aes(y = mean_diff, lty = "mean")) +
  geom_line(aes(y = 0), color = "red") +
  geom_line(aes(y = sdsup, lty = "sd")) +
  geom_line(aes(y = sdinf, lty = "sd")) +
  scale_linetype_manual("", c("Mean of differences", "Mean +/- 1.96 sd"), values = c(1,2)) +
  theme_bw() +
  theme(legend.position = "bottom") +
  labs(x = "Mean", y = "|Difference|")+ggtitle("Bland Altman (log scale)")+ scale_x_continuous(trans = 'log10') +
  scale_y_continuous(trans = 'log10')

plotbland1+plotbland2


```

For the survival, the presence or absence of G4 patients seems to change the results.

## Genes of interest{ .tabset .tabset-pills}
The gene SLC17A8 was always found as most important gene with the G4 patients, and also had a link with cold and diffuse status, we can explore more the gene. Moreover, we can look and the genes that are always found with or without G4, as presented in the following table: 

```{r,echo=F,message=FALSE,warning=FALSE}
liste_both=merge(liste_found_avec,liste_found_sans,by="gene")
colnames(liste_both)=c("gene","With_g4","Without g4")
liste_both=liste_both[order(liste_both[,2],decreasing = TRUE),]
rownames(liste_both)=NULL
write.xlsx(liste_both, file = "survival_RNA_both.xlsx",col.names = TRUE)
kable_classic(kbl(x=liste_both,caption = "Genes always with and without G4"),full_width = F, html_font = "Cambria")
```

For each proteins of interest, the cox model is computed, followed by the Kaplan Meier curves, with 2 categories (patients with expression bellow the median, and above the median), and the counts per categories (cold/diffuse and astro/oligo)

### SLC17A8 {.tabset .tabset-fade .tabset-pills}

We can explore the SLC17A8 gene, that is selected with G4 patients (importance `r total_importances[total_importances[,1]=="SLC17A8",2]` ), but not without them (importance `r total_importances_sans[total_importances_sans[,1]=="SLC17A8",2]` )

#### With G4 

A multivariate and univariate cox model is computed to see if the gene is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=datas_avec 


modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+SLC17A8,data=cbind(datas,norm_counts))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~SLC17A8,data=cbind(datas,norm_counts))
summary(modele_base2)

gene="SLC17A8"
values=counts[,colnames(counts)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")

```

Here are the Kaplan meier curves with a separation of the patients according to the genes counts. This allows to see if there is a differences between the patients with genes counts superior or inferior to the median in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the gene counts for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}

gene=norm_counts[,colnames(norm_counts)=="SLC17A8"]
datas_gene=as.data.frame(cbind(datas[,c(1,2,4,3)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")
ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for SLC17A8") #+theme_classic(axis.text.x=element_blank())


```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts[,colnames(norm_counts)=="SLC17A8"]
datas_gene=as.data.frame(cbind(datas[,c(1,7,4,3)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","Time","Event","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for SLC17A8") #+theme_classic(axis.text.x=element_blank())
```





#### Without G4 

A multivariate and univariate cox model is computed to see if the gene is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=datas_sans

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+SLC17A8,data=cbind(datas,norm_counts_sans))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~SLC17A8,data=cbind(datas,norm_counts_sans))
summary(modele_base2)

gene="SLC17A8"
values=counts_sans[,colnames(counts_sans)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")

```

Here are the Kaplan meier curves with a separation of the patients according to the genes counts. This allows to see if there is a differences between the patients with genes counts superior or inferior to the median in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the gene counts for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}


gene=norm_counts_sans[,colnames(norm_counts_sans)=="SLC17A8"]
datas_gene=as.data.frame(cbind(datas[,c(1,2,4,3)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")
ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for SLC17A8") #+theme_classic(axis.text.x=element_blank())

gene=norm_counts_sans[,colnames(norm_counts_sans)=="SLC17A8"]
datas_gene=as.data.frame(cbind(datas[,c(1,7,4,3)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","Time","Event","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()


```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for SLC17A8") #+theme_classic(axis.text.x=element_blank())
```


### CABIN1 {.tabset .tabset-fade .tabset-pills}

We can explore the CABIN1 gene, that is selected with G4 patients (importance `r total_importances[total_importances[,1]=="CABIN1",2]` ), and with G4 (importance `r total_importances_sans[total_importances_sans[,1]=="CABIN1",2]` )

#### With G4

A multivariate and univariate cox model is computed to see if the gene is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=datas_avec 


modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+CABIN1,data=cbind(datas,norm_counts))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~CABIN1,data=cbind(datas,norm_counts))
summary(modele_base2)

gene="CABIN1"
values=counts[,colnames(counts)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]



datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")


```

Here are the Kaplan meier curves with a separation of the patients according to the genes counts. This allows to see if there is a differences between the patients with genes counts superior or inferior to the median in term of risk of recurrence. 

The cox model is not significant, and the Kaplan meier curvers show no differences. To see if there is a link not proportional between the counts of CABIN1 and survival, we show the gene expression in function of the time to event: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Counts of CABIN1 linked with the time to event"}

gene=norm_counts[,colnames(norm_counts)=="CABIN1"]
datas_gene=as.data.frame(cbind(datas[,c(1,2,4,3)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")

datas_genes_graph=datas_gene[order(datas_gene[,4],datas_gene[,3]),]
datas_genes_graph[,4]=as.factor(datas_genes_graph[,4])
ggplot(datas_genes_graph)+geom_point(aes(x=Time,y=counts,color=Event))+ facet_grid(. ~ Event)+ggtitle("Counts of CABIN1 in function of the time to event, with or without recurrence")


```

We observe then the differences between the cold and diffuse expression, with first a boxplot of the gene counts for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}
ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for CABIN1") #+theme_classic(axis.text.x=element_blank())

gene=norm_counts[,colnames(norm_counts)=="CABIN1"]
datas_gene=as.data.frame(cbind(datas[,c(1,7,4,3)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","Time","Event","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()
```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for CABIN1") #+theme_classic(axis.text.x=element_blank())
```



#### Without G4 

A multivariate and univariate cox model is computed to see if the gene is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=datas_sans

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+CABIN1,data=cbind(datas,norm_counts_sans))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~CABIN1,data=cbind(datas,norm_counts_sans))
summary(modele_base2)

gene="CABIN1"
values=counts_sans[,colnames(counts_sans)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]



datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")

```

Here are the Kaplan meier curves with a separation of the patients according to the genes counts. This allows to see if there is a differences between the patients with genes counts superior or inferior to the median in term of risk of recurrence. 

The cox model is not significant, and the Kaplan meier curvers show no differences. To see if there is a link not proportional between the counts of CABIN1 and survival, we show the gene expression in function of the time to event: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Counts of CABIN1 linked with the time to event"}


gene=norm_counts_sans[,colnames(norm_counts_sans)=="CABIN1"]
datas_gene=as.data.frame(cbind(datas[,c(1,2,4,3)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")

datas_genes_graph=datas_gene[order(datas_gene[,4],datas_gene[,3]),]
datas_genes_graph[,4]=as.factor(datas_genes_graph[,4])
ggplot(datas_genes_graph)+geom_point(aes(x=Time,y=counts,color=Event))+ facet_grid(. ~ Event)+ggtitle("Counts of CABIN1 in function of the time to event, with or without recurrence")

```

We observe then the differences between the cold and diffuse expression, with first a boxplot of the gene counts for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}

ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for CABIN1") #+theme_classic(axis.text.x=element_blank())

gene=norm_counts_sans[,colnames(norm_counts_sans)=="CABIN1"]
datas_gene=as.data.frame(cbind(datas[,c(1,7,4,3)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","Time","Event","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for CABIN1") #+theme_classic(axis.text.x=element_blank())
```


### SLC22A11 {.tabset .tabset-fade .tabset-pills}

We can explore the SLC22A11 gene, that is selected with G4 patients (importance `r total_importances[total_importances[,1]=="SLC22A11",2]` ), but not without them (importance `r total_importances_sans[total_importances_sans[,1]=="SLC22A11",2]` )

#### With G4 

A multivariate and univariate cox model is computed to see if the gene is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=datas_avec 


modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+SLC22A11,data=cbind(datas,norm_counts))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~SLC22A11,data=cbind(datas,norm_counts))
summary(modele_base2)

gene="SLC22A11"
values=counts[,colnames(counts)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")

```

Here are the Kaplan meier curves with a separation of the patients according to the genes counts. This allows to see if there is a differences between the patients with genes counts superior or inferior to the median in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the gene counts for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}

gene=norm_counts[,colnames(norm_counts)=="SLC22A11"]
datas_gene=as.data.frame(cbind(datas[,c(1,2,4,3)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")
ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for SLC22A11") #+theme_classic(axis.text.x=element_blank())


```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts[,colnames(norm_counts)=="SLC22A11"]
datas_gene=as.data.frame(cbind(datas[,c(1,7,4,3)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","Time","Event","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for SLC22A11") #+theme_classic(axis.text.x=element_blank())
```





#### Without G4 

A multivariate and univariate cox model is computed to see if the gene is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=datas_sans

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+SLC22A11,data=cbind(datas,norm_counts_sans))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~SLC22A11,data=cbind(datas,norm_counts_sans))
summary(modele_base2)

gene="SLC22A11"
values=counts_sans[,colnames(counts_sans)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")

```

Here are the Kaplan meier curves with a separation of the patients according to the genes counts. This allows to see if there is a differences between the patients with genes counts superior or inferior to the median in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the gene counts for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}


gene=norm_counts_sans[,colnames(norm_counts_sans)=="SLC22A11"]
datas_gene=as.data.frame(cbind(datas[,c(1,2,4,3)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")
ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for SLC22A11") #+theme_classic(axis.text.x=element_blank())

gene=norm_counts_sans[,colnames(norm_counts_sans)=="SLC22A11"]
datas_gene=as.data.frame(cbind(datas[,c(1,7,4,3)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","Time","Event","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()


```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for SLC22A11") #+theme_classic(axis.text.x=element_blank())
```



### RUNX3 {.tabset .tabset-fade .tabset-pills}

We can explore the RUNX3 gene, that is selected with G4 patients (importance `r total_importances[total_importances[,1]=="RUNX3",2]` ), and with G4 (importance `r total_importances_sans[total_importances_sans[,1]=="RUNX3",2]` )


#### With G4 

A multivariate and univariate cox model is computed to see if the gene is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=datas_avec 


modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+RUNX3,data=cbind(datas,norm_counts))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~RUNX3,data=cbind(datas,norm_counts))
summary(modele_base2)

gene="RUNX3"
values=counts[,colnames(counts)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")

```

Here are the Kaplan meier curves with a separation of the patients according to the genes counts. This allows to see if there is a differences between the patients with genes counts superior or inferior to the median in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the gene counts for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}

gene=norm_counts[,colnames(norm_counts)=="RUNX3"]
datas_gene=as.data.frame(cbind(datas[,c(1,2,4,3)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")
ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for RUNX3") #+theme_classic(axis.text.x=element_blank())


```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts[,colnames(norm_counts)=="RUNX3"]
datas_gene=as.data.frame(cbind(datas[,c(1,7,4,3)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","Time","Event","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for RUNX3") #+theme_classic(axis.text.x=element_blank())
```





#### Without G4 

A multivariate and univariate cox model is computed to see if the gene is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=datas_sans

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+RUNX3,data=cbind(datas,norm_counts_sans))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~RUNX3,data=cbind(datas,norm_counts_sans))
summary(modele_base2)

gene="RUNX3"
values=counts_sans[,colnames(counts_sans)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")

```

Here are the Kaplan meier curves with a separation of the patients according to the genes counts. This allows to see if there is a differences between the patients with genes counts superior or inferior to the median in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the gene counts for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}


gene=norm_counts_sans[,colnames(norm_counts_sans)=="RUNX3"]
datas_gene=as.data.frame(cbind(datas[,c(1,2,4,3)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")
ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for RUNX3") #+theme_classic(axis.text.x=element_blank())

gene=norm_counts_sans[,colnames(norm_counts_sans)=="RUNX3"]
datas_gene=as.data.frame(cbind(datas[,c(1,7,4,3)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","Time","Event","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()


```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for RUNX3") #+theme_classic(axis.text.x=element_blank())
```




### RPS23P1 {.tabset .tabset-fade .tabset-pills}

We can explore the RPS23P1 gene, that is selected with G4 patients (importance `r total_importances[total_importances[,1]=="RPS23P1",2]` ), and with G4 (importance `r total_importances_sans[total_importances_sans[,1]=="RPS23P1",2]` )


#### With G4 

A multivariate and univariate cox model is computed to see if the gene is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=datas_avec 


modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+RPS23P1,data=cbind(datas,norm_counts))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~RPS23P1,data=cbind(datas,norm_counts))
summary(modele_base2)

gene="RPS23P1"
values=counts[,colnames(counts)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")

```

Here are the Kaplan meier curves with a separation of the patients according to the genes counts. This allows to see if there is a differences between the patients with genes counts superior or inferior to the median in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the gene counts for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}

gene=norm_counts[,colnames(norm_counts)=="RPS23P1"]
datas_gene=as.data.frame(cbind(datas[,c(1,2,4,3)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")
ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for RPS23P1") #+theme_classic(axis.text.x=element_blank())


```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts[,colnames(norm_counts)=="RPS23P1"]
datas_gene=as.data.frame(cbind(datas[,c(1,7,4,3)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","Time","Event","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for RPS23P1") #+theme_classic(axis.text.x=element_blank())
```





#### Without G4 

A multivariate and univariate cox model is computed to see if the gene is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=datas_sans

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+RPS23P1,data=cbind(datas,norm_counts_sans))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~RPS23P1,data=cbind(datas,norm_counts_sans))
summary(modele_base2)

gene="RPS23P1"
values=counts_sans[,colnames(counts_sans)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")

```

Here are the Kaplan meier curves with a separation of the patients according to the genes counts. This allows to see if there is a differences between the patients with genes counts superior or inferior to the median in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the gene counts for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}


gene=norm_counts_sans[,colnames(norm_counts_sans)=="RPS23P1"]
datas_gene=as.data.frame(cbind(datas[,c(1,2,4,3)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")
ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for RPS23P1") #+theme_classic(axis.text.x=element_blank())

gene=norm_counts_sans[,colnames(norm_counts_sans)=="RPS23P1"]
datas_gene=as.data.frame(cbind(datas[,c(1,7,4,3)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","Time","Event","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()


```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for RPS23P1") #+theme_classic(axis.text.x=element_blank())
```




### RPS2P36 {.tabset .tabset-fade .tabset-pills}

We can explore the RPS2P36 gene, that is selected with G4 patients (importance `r total_importances[total_importances[,1]=="RPS2P36",2]` ), and with G4 (importance `r total_importances_sans[total_importances_sans[,1]=="RPS2P36",2]` )



#### With G4 

A multivariate and univariate cox model is computed to see if the gene is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=datas_avec 


modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+RPS2P36,data=cbind(datas,norm_counts))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~RPS2P36,data=cbind(datas,norm_counts))
summary(modele_base2)

gene="RPS2P36"
values=counts[,colnames(counts)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")

```

Here are the Kaplan meier curves with a separation of the patients according to the genes counts. This allows to see if there is a differences between the patients with genes counts superior or inferior to the median in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the gene counts for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}

gene=norm_counts[,colnames(norm_counts)=="RPS2P36"]
datas_gene=as.data.frame(cbind(datas[,c(1,2,4,3)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")
ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for RPS2P36") #+theme_classic(axis.text.x=element_blank())


```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts[,colnames(norm_counts)=="RPS2P36"]
datas_gene=as.data.frame(cbind(datas[,c(1,7,4,3)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","Time","Event","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for RPS2P36") #+theme_classic(axis.text.x=element_blank())
```





#### Without G4 

A multivariate and univariate cox model is computed to see if the gene is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=datas_sans

modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+RPS2P36,data=cbind(datas,norm_counts_sans))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~RPS2P36,data=cbind(datas,norm_counts_sans))
summary(modele_base2)

gene="RPS2P36"
values=counts_sans[,colnames(counts_sans)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")

```

Here are the Kaplan meier curves with a separation of the patients according to the genes counts. This allows to see if there is a differences between the patients with genes counts superior or inferior to the median in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the gene counts for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}


gene=norm_counts_sans[,colnames(norm_counts_sans)=="RPS2P36"]
datas_gene=as.data.frame(cbind(datas[,c(1,2,4,3)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")
ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for RPS2P36") #+theme_classic(axis.text.x=element_blank())

gene=norm_counts_sans[,colnames(norm_counts_sans)=="RPS2P36"]
datas_gene=as.data.frame(cbind(datas[,c(1,7,4,3)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","Time","Event","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()


```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

ID2=rep("",length(datas_gene[,1]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for RPS2P36") #+theme_classic(axis.text.x=element_blank())
```




## {- .unlisted .unnumbered .tabset .tabset-pills}

We can look at the correlation between the different genes of interest and SLC17A8, to see the link between them and to try to understand why a gene always found with the G4 patients is then lost when we remove those patients (with negative importance) 



### With G4

```{r,echo=FALSE,message=FALSE,warning=FALSE,cache=TRUE,fig.cap="heatmap of the correlation between SLC17A8 and the genes always found"}
SLC17A8=norm_counts[,colnames(norm_counts)=="SLC17A8"]
CABIN1=norm_counts[,colnames(norm_counts)=="CABIN1"]
SLC22A11=norm_counts[,colnames(norm_counts)=="SLC22A11"]
RUNX3=norm_counts[,colnames(norm_counts)=="RUNX3"]
RPS23P1=norm_counts[,colnames(norm_counts)=="RPS23P1"]
RPS2P36=norm_counts[,colnames(norm_counts)=="RPS2P36"]

data_corr=cbind(SLC17A8,CABIN1,SLC22A11,RUNX3,RPS23P1,RPS2P36)
correlation=cor(data_corr)

library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=correlation,caption = "Correlations between the genes of interest"),full_width = F, html_font = "Cambria")
 
 data_melt=melt(correlation)

ggplot(data = data_melt, aes(x=Var1, y=Var2, fill=value)) + 
  geom_tile()+scale_fill_viridis()+ggtitle("Correlation between the genes of interest")
```

There is a negative correlation between SLC17A8 and RUNX3

### Without G4

```{r,echo=FALSE,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="heatmap of the correlation between SLC17A8 and the genes always found"}
SLC17A8=norm_counts_sans[,colnames(norm_counts_sans)=="SLC17A8"]
CABIN1=norm_counts_sans[,colnames(norm_counts_sans)=="CABIN1"]
SLC22A11=norm_counts_sans[,colnames(norm_counts_sans)=="SLC22A11"]
RUNX3=norm_counts_sans[,colnames(norm_counts_sans)=="RUNX3"]
RPS23P1=norm_counts_sans[,colnames(norm_counts_sans)=="RPS23P1"]
RPS2P36=norm_counts_sans[,colnames(norm_counts_sans)=="RPS2P36"]

data_corr=cbind(SLC17A8,CABIN1,SLC22A11,RUNX3,RPS23P1,RPS2P36)
correlation=cor(data_corr)

library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=correlation,caption = "Correlations between the genes of interest"),full_width = F, html_font = "Cambria")
 data_melt=melt(correlation)

 
ggplot(data = data_melt, aes(x=Var1, y=Var2, fill=value)) + 
  geom_tile()+scale_fill_viridis()+ggtitle("Correlation between the genes of interest")

```

There is still a negative correlation between SLC17A8 and RUNX3, that seems to have not changed a lot.

# Proteomics analysis 
We are then interested in the Proteomics analysis. We have a total of 98 patients, with data on proteomics, obtained by DIA and the spectronaut software (and a median normalisation). As two patients seems to be outliers (TOR-ROB and BAL-PIE), we removed them from the analysis 
```{r,echo=FALSE,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="boxplots of the counts for all patients"}

  
data_total <- read_excel("data/13_Transcan_All_survival.xlsx",sheet = "All_survie")
data_total=as.data.frame(data_total)
data_total=data_total[,c(4,5,16,18,10,11,6,8,9,3)]
data_total=data_total[-1,]
data_total=data_total[!is.na(data_total[,2]),]
data_total=data_total[!data_total[,2]=="SPOTS",]
data_total=data_total[!data_total[,2]=="SPOT",]

colnames(data_total)=c("ID","cold_diffuse","Reccurence","Time","Age","Sexe","oligo_astro","RNA","proteo","ID_proteo")
data_total=data_total[,c(1,2,3,4,6,5,7,8,9,10)]
for (i in 1:length(data_total[,1])){
  
  if (data_total[i,2]=="diffuse"){
    data_total[i,2]="DIFFUSE"
  }
    if (data_total[i,2]=="cold"){
    data_total[i,2]="COLD"
    }
  if (data_total[i,7]=="."){
    data_total[i,7]=NA
  }else{
    
  
  if (data_total[i,7]=="1"){
    data_total[i,7]="oligo"
  }
    if (data_total[i,7]=="0"){
    data_total[i,7]="astro"
  }}
}

data_total=data_total[!duplicated(data_total[,1]),]


data_total[,6]=as.numeric(data_total[,6])
data_total[,3]=as.numeric(data_total[,3])
data_total[,4]=as.numeric(data_total[,4])/365
data_total[,2]=as.factor(data_total[,2])
data_total[,5]=as.factor(data_total[,5])
data_total[,7]=as.factor(data_total[,7])
data_total[,8]=as.factor(data_total[,8])
data_total[,9]=as.factor(data_total[,9])


oligo=c(as.data.frame( read_excel("data/Transcan_Liste_Astro_Oligo.xlsx",sheet = "Oligo",col_names = FALSE)))$"...1"
astro=as.data.frame( read_excel("data/Transcan_Liste_Astro_Oligo.xlsx",sheet = "Astro",col_names = FALSE))
G4=astro[astro[,2]=="G4",]
G4=ifelse(data_total[,1]%in%G4[,1],"1","0")

data_total=cbind(data_total,G4)

data_total[,11]=as.factor(data_total[,11])
data_total=data_total[!data_total[,1]%in%c("BAL-PIE","TOR-ROB"),]
data_total_avec=data_total
data_total_sans=data_total[data_total[,11]=="0",]





datas=data_total_avec[data_total_avec[,9]=="1",-c(8,9)]
IDs=datas[,1]
IDs_table_split=strsplit(as.character(IDs),split="_")
IDs_table=numeric(length(datas[,1]))
for (i in 1:length(IDs_table)){
  IDs_table[i]=IDs_table_split[[i]][1]
}
datas[,1]=IDs_table
table_proteo=datas[!duplicated(datas[,1]),]


names_table=table_proteo[,8]
names_table_split=strsplit(as.character(names_table),split=" ")
names_table=numeric(length(names_table))
for (i in 1:length(names_table)){
  names_table[i]=names_table_split[[i]][1]
}
table_proteo=table_proteo[,c(2,8,1,3,4,5,6,7,9)]
table_proteo[,2]=names_table
row.names(table_proteo)=names_table
table_proteo=table_proteo[!is.na(table_proteo[,2]),]
table_proteo=table_proteo[,c(2,3,1,4,5,6,7,8,9)]


matrix_proteomic=read.table("data/matrix_proteomics_final.txt")
matrix_proteomic=matrix_proteomic[rownames(matrix_proteomic)%in%table_proteo[,1],]
names_patients=row.names(matrix_proteomic)
table_proteo=table_proteo[names_patients,]

counts=matrix_proteomic

norm_counts <- apply(counts, MARGIN = 2, function(v) {
  log2((v + 0.5)/(sum(v) + 1) * 10^6)
})
norm_counts <- as.data.frame(norm_counts)
raw_counts=counts


table_proteo_avec=table_proteo
table_proteo_sans=table_proteo[table_proteo[,9]=="0",]


norm_counts_sans=norm_counts[rownames(norm_counts)%in%table_proteo_sans[,1],]
raw_counts_sans=raw_counts[rownames(raw_counts)%in%table_proteo_sans[,1],]
library(kableExtra)



data_histo=cbind(table_proteo[,1],table_proteo[,9],table_proteo[,3],table_proteo[,8],counts)
colnames(data_histo)=c("ID","G4","Cold_diffuse","astro_oligo",colnames(counts))
data_histo=data_histo[order(data_histo[,2],data_histo[,3]),]
data_histo[,1]=factor(data_histo[,1],levels=data_histo[,1])
df.m <- melt(data_histo[,-c(3,4)], id=c("ID","G4"))
plot1=ggplot(data = df.m, aes(x=ID, y=value)) + geom_boxplot(aes(fill=G4),outlier.shape = NA)+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ 
  scale_y_continuous(limits = quantile(df.m$value, c(0.1, 0.9)))+ scale_fill_viridis_d()+xlab("patients")+ylab("counts")


data_histo=cbind(table_proteo[,1],table_proteo[,9],table_proteo[,3],table_proteo[,8],counts)
colnames(data_histo)=c("ID","G4","Cold_diffuse","astro_oligo",colnames(counts))
data_histo=data_histo[order(data_histo[,2],data_histo[,3]),]
data_histo[,1]=factor(data_histo[,1],levels=data_histo[,1])
df.m <- melt(data_histo[,-c(2,4)], id=c("ID","Cold_diffuse"))
plot2=ggplot(data = df.m, aes(x=ID, y=value)) + geom_boxplot(aes(fill=Cold_diffuse),outlier.shape = NA)+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ 
  scale_y_continuous(limits = quantile(df.m$value, c(0.1, 0.9)))+xlab("patients")+ylab("counts")+scale_fill_manual(values= c("#E7B800", "#2E9FDF"))


data_histo=cbind(table_proteo[,1],table_proteo[,9],table_proteo[,3],table_proteo[,8],counts)
colnames(data_histo)=c("ID","G4","Cold_diffuse","Astro_oligo",colnames(counts))
data_histo=data_histo[order(data_histo[,2],data_histo[,3]),]
data_histo[,1]=factor(data_histo[,1],levels=data_histo[,1])
df.m <- melt(data_histo[,-c(2,3)], id=c("ID","Astro_oligo"))
plot3=ggplot(data = df.m, aes(x=ID, y=value)) + geom_boxplot(aes(fill=Astro_oligo),outlier.shape = NA)+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ 
  scale_y_continuous(limits = quantile(df.m$value, c(0.1, 0.9)))+xlab("patients")+ylab("counts")
#+scale_fill_manual(values= c("#E7B800", "#2E9FDF"))

plot1/plot2/plot3


```


## Cold vs Diffuse {.tabset .tabset-pills}

We have 37 cold and 59 diffuse patients. Four patients are "G4": two cold and two diffuse, 4 astro patients.

### With G4


```{r, eval=T,echo=FALSE,message=FALSE,warning=FALSE,cache=FALSE}

table_proteo=table_proteo_avec
table_na=table_proteo[!is.na(table_proteo[,4])&!is.na(table_proteo[,6]),]
Age=c(round(mean(table_na[table_na[,3]=="COLD",7]),1),round(mean(table_na[table_na[,3]=="DIFFUSE",7]),1))
Female=c(table(table_na[table_na[,3]=="COLD",6])[1],table(table_na[table_na[,3]=="DIFFUSE",6])[1])
Male=c(table(table_na[table_na[,3]=="COLD",6])[2],table(table_na[table_na[,3]=="DIFFUSE",6])[2])
Event=c(table(table_na[table_na[,3]=="COLD",4])[2],table(table_na[table_na[,3]=="DIFFUSE",4])[2])
No_event=c(table(table_na[table_na[,3]=="COLD",4])[1],table(table_na[table_na[,3]=="DIFFUSE",4])[1])
Median_followup=c(round(summary(table_na[table_na[,3]=="COLD",5])[2],1),round(summary(table_na[table_na[,3]=="DIFFUSE",5])[2],1))
cold_diffuse_table=rbind(Age,Female,Male,Event,No_event,Median_followup)
colnames(cold_diffuse_table)=c("Cold (n=37)","Diffuse (n=59) ")
rownames(cold_diffuse_table)=c("Age (mean)","Female","Male","Event","No event", "Median follow up (years)")

```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=cold_diffuse_table,caption = "Cold vs Diffuse patients"),full_width = F, html_font = "Cambria")

 
```



We performed a PCA with all proteins values, with a separation depending on the cold or diffuse status: 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,results=FALSE, fig.show='hide',cache=FALSE}
table_proteo=table_proteo_avec

counts_pca <- raw_counts
counts_pca <- round((counts_pca+1),0)
counts_pca <- vst(t(counts_pca))

counts_pca_scaled <- base::scale(t(counts_pca), center=TRUE, scale=TRUE)

res.pca=PCA(counts_pca_scaled , graph=FALSE)
#plot.PCA(res.pca, axes=c(1, 2), choix="ind")
Cold_or_diffuse=as.factor(table_proteo[,3])
G4=as.factor(table_proteo_avec[,9])

pca_data <- cbind.data.frame(res.pca$ind$coord[, 1:4],as.factor(table_proteo[,3]),table_proteo[,2])
colnames(pca_data)=c("Dim.1","Dim.2","Dim.3","Dim.4","Cold_diffuse","ID")
table_coord=res.pca$ind$coord[,c(1,2)]
particuliers=table_coord[table_coord[,1]>quantile(table_coord[,1],0.97)|table_coord[,1]<quantile(table_coord[,1],0.03)|table_coord[,2]>quantile(table_coord[,2],0.98)|table_coord[,2]<quantile(table_coord[,2],0.02),]

for (i in 1:length(pca_data[,1])){
  if (!rownames(pca_data)[i]%in%rownames(particuliers)){
    pca_data[i,6]=""
    }
}
p <- ggplot(pca_data, aes(x=Dim.1, y=Dim.2, colour = as.factor(table_proteo[,3]),shape=G4,label=ID)) +
 # viridis::scale_color_viridis("Cold_diffuse", discrete = TRUE,end = 0.8)
    scale_color_manual(values= c("#E7B800", "#2E9FDF"))+
  geom_text_repel() +
  theme_bw() +
  geom_point(size=2, alpha=1) +
  guides(color=guide_legend(override.aes = list(alpha=1),title="Cold or Diffuse")) +
  xlab(label=paste0("PC1 (", round(res.pca$eig[1,2], 2),"% of total variance explained)")) +
  ylab(label=paste0("PC2 (", round(res.pca$eig[2,2], 2), "% of total variance explained)")) +
  ggtitle("", subtitle = "First factorial plan")


```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE, fig.cap="PCA of the proteomics data, with G4 patients (vst transformation and scaled data)"}
p

```



We can look at the characteristics of the patients outside of the majority of the patients:


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
datas=table_proteo_avec
table_particulier=datas[datas[,1]%in%rownames(particuliers),]
table_particulier=table_particulier[,c(2,6,7,3,8,5,4)]
table_particulier[,6]=round(table_particulier[,6],2)
rownames(table_particulier)=NULL
```


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=table_particulier,caption = "Characteristics patients outsite of PCA majority"),full_width = F, html_font = "Cambria")

```

### Without G4


Without the G4 patients, we have 35 cold patients and 57 diffuse patients: 

```{r, eval=T,echo=FALSE,message=FALSE,warning=FALSE,cache=FALSE}
table_proteo=table_proteo_sans
table_na=table_proteo[!is.na(table_proteo[,4])&!is.na(table_proteo[,6]),]
Age=c(round(mean(table_na[table_na[,3]=="COLD",7]),1),round(mean(table_na[table_na[,3]=="DIFFUSE",7]),1))
Female=c(table(table_na[table_na[,3]=="COLD",6])[1],table(table_na[table_na[,3]=="DIFFUSE",6])[1])
Male=c(table(table_na[table_na[,3]=="COLD",6])[2],table(table_na[table_na[,3]=="DIFFUSE",6])[2])
Event=c(table(table_na[table_na[,3]=="COLD",4])[2],table(table_na[table_na[,3]=="DIFFUSE",4])[2])
No_event=c(table(table_na[table_na[,3]=="COLD",4])[1],table(table_na[table_na[,3]=="DIFFUSE",4])[1])
Median_followup=c(round(summary(table_na[table_na[,3]=="COLD",5])[2],1),round(summary(table_na[table_na[,3]=="DIFFUSE",5])[2],1))
cold_diffuse_table=rbind(Age,Female,Male,Event,No_event,Median_followup)
colnames(cold_diffuse_table)=c("Cold (n=35)","Diffuse (n=57)")
rownames(cold_diffuse_table)=c("Age (mean)","Female","Male","Event","No event", "Median follow up (years)")
```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=cold_diffuse_table,caption = "Cold vs Diffuse patients"),full_width = F, html_font = "Cambria")

 
```



We performed a PCA with all proteins values, with a separation depending on the cold or diffuse status: 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,results=FALSE, fig.show='hide',cache=FALSE}
table_proteo=table_proteo_sans
counts_pca <- raw_counts_sans
counts_pca <- round((counts_pca+1),0)
counts_pca <- vst(t(counts_pca))

counts_pca_scaled <- base::scale(t(counts_pca), center=TRUE, scale=TRUE)

res.pca=PCA(counts_pca_scaled , graph=FALSE)
#plot.PCA(res.pca, axes=c(1, 2), choix="ind")
Cold_or_diffuse=as.factor(table_proteo[,3])
#G4=as.factor(table_proteo[,9])
pca_data <- cbind.data.frame(res.pca$ind$coord[, 1:4],as.factor(table_proteo[,3]),table_proteo[,2])
colnames(pca_data)=c("Dim.1","Dim.2","Dim.3","Dim.4","Cold_diffuse","ID")
table_coord=res.pca$ind$coord[,c(1,2)]
particuliers=table_coord[table_coord[,1]>quantile(table_coord[,1],0.97)|table_coord[,1]<quantile(table_coord[,1],0.03)|table_coord[,2]>quantile(table_coord[,2],0.98)|table_coord[,2]<quantile(table_coord[,2],0.02),]

for (i in 1:length(pca_data[,1])){
  if (!rownames(pca_data)[i]%in%rownames(particuliers)){
    pca_data[i,6]=""
    }
}
p <- ggplot(pca_data, aes(x=Dim.1, y=Dim.2, colour = as.factor(table_proteo[,3]),label=ID)) +
 # viridis::scale_color_viridis("Cold_diffuse", discrete = TRUE,end = 0.8)
    scale_color_manual(values= c("#E7B800", "#2E9FDF"))+
  geom_text_repel() +
  theme_bw() +
  geom_point(size=2, alpha=1) +
  guides(color=guide_legend(override.aes = list(alpha=1),title="Cold or Diffuse")) +
  xlab(label=paste0("PC1 (", round(res.pca$eig[1,2], 2),"% of total variance explained)")) +
  ylab(label=paste0("PC2 (", round(res.pca$eig[2,2], 2), "% of total variance explained)")) +
  ggtitle("", subtitle = "First factorial plan")


```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE, fig.cap="PCA of the proteomics data, without G4 patients (vst transformation and scaled data)"}
p

```

Some patients are outside of the majority, we can look at their characteristics: 


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
datas=table_proteo_sans
table_particulier=datas[datas[,1]%in%rownames(particuliers),]
table_particulier=table_particulier[,c(2,6,7,3,8,5,4)]
table_particulier[,6]=round(table_particulier[,6],2)
rownames(table_particulier)=NULL
```


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=table_particulier,caption = "Characteristics patients outsite of PCA majority"),full_width = F, html_font = "Cambria")

```


## {- .tabset  .tabset-pills .unlisted .unnumbered}

We will then look protein per protein if there is a difference between cold and diffuse. For that, we use the dearseq package. 

### With G4

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
table_proteo=table_proteo_avec
output_avec=ifelse(table_proteo[,3]=="COLD",0,1)
output=output_avec
Y=matrix(output,ncol=1,nrow=length(output))

scores=dear_seq(exprmat=t(as.matrix(counts)),variables2test = Y,preprocessed = FALSE,which_test = "asymptotic")


genewise_dearseq_cold=as.data.frame(cbind(rownames(scores$pvals),scores$pvals[,2],rank(scores$pvals[,2]),scores$pvals[,1]))
genewise_dearseq_cold[,1]=as.character(genewise_dearseq_cold[,1])
genewise_dearseq_cold[,2]=as.numeric(as.character(genewise_dearseq_cold[,2]))
genewise_dearseq_cold[,3]=as.numeric(as.character(genewise_dearseq_cold[,3]))
genewise_dearseq_cold[,4]=as.numeric(as.character(genewise_dearseq_cold[,4]))
genewise_dearseq_cold=genewise_dearseq_cold[order(genewise_dearseq_cold[,2],decreasing = FALSE),]
colnames(genewise_dearseq_cold)=c("ID","pvalue","rank_pvalue","raw_pvalue")





patients_cold=norm_counts[output_avec==0,]
patients_diffuse=norm_counts[output_avec==1,]
means_diffuse=numeric(length(patients_cold[1,]))
means_cold=numeric(length(patients_cold[1,]))
for (i in 1:length(patients_cold[1,])){
  means_diffuse[i]=mean(patients_diffuse[,i])
  means_cold[i]=mean(patients_cold[,i])
}

fold_change=data.frame(colnames(patients_cold),means_diffuse-means_cold)
colnames(fold_change)=c("ID","log2FoldChange")
total_proteo_cold=merge(genewise_dearseq_cold,fold_change,by="ID")
total_proteo_cold=total_proteo_cold[order(total_proteo_cold[,4],total_proteo_cold[,5]),]
write.xlsx(total_proteo_cold[total_proteo_cold[,2]<0.05,], file = "dearseq_proteo_C_D_wg4.xlsx",col.names = TRUE)


total_proteo_cold$diffexpressed <- "NO"
# if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP" 
total_proteo_cold$diffexpressed[total_proteo_cold$log2FoldChange > 0.6 & total_proteo_cold$pvalue < 0.05] <- "UP"
# if log2Foldchange < (-0.6) and pvalue < 0.05, set as "DOWN"
total_proteo_cold$diffexpressed[total_proteo_cold$log2FoldChange < (-0.6) & total_proteo_cold$pvalue < 0.05] <- "DOWN"

total_proteo_cold$delabel <- NA
total_proteo_cold$delabel[total_proteo_cold$diffexpressed != "NO"] <- total_proteo_cold$ID[total_proteo_cold$diffexpressed != "NO"]


plot1=ggplot(data=total_proteo_cold, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed, label=delabel)) + 
    geom_point() + 
    theme_minimal() +
  geom_text_repel()  + scale_color_manual(values= c( "#1F968BFF", "#DCE319FF","#440154FF")) +ggtitle("Log2 Fold change in function of pvalues")

plot2=ggplot(data=total_proteo_cold,aes(x=log2FoldChange))+geom_density()+xlim(c(-5,5))+ theme_minimal()+ggtitle("Distribution of Log2FoldChange") 
```

dearseq identifies `r dim(genewise_dearseq_cold[genewise_dearseq_cold[,2]<0.05,])[1]` proteins that are differentially expressed between cold and diffuse. Here are the first genes in term of p-value:

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
rownames(total_proteo_cold)=NULL
 kable_classic(kbl(x=total_proteo_cold[1:10,c(1,2,3,4,5)],caption = "Cold vs Diffuse genes"),full_width = F, html_font = "Cambria")

```

We can look at the vulcano plot and also the distribution of the log2 Fold change, in the following figure. The vulcano plot shows the log10(pvalue) in function of the log2 Fold change, which allows to see if the significant genes are up regulated or down regulated (cold as base). A positive log2 Fold change indicates that the gene is, on average, more expressed among the diffuse patients, while a negative log2 Fold change indicates that the gene is more expressed among the cold patients. 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,fig.cap="Vulcano plot and distribution of log2 Fold change",cache=FALSE}
plot1+plot2
```

We can observe the brier score associated with these genes. The brier score is an indicator of the prediction aibility, between 1 and 0, 0 being a perfect predictor and 1 a poor predictor: 
```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}


Y <- output
X <- norm_counts[,colnames(norm_counts)%in%genewise_dearseq_cold[genewise_dearseq_cold[,2]<0.05,1]]
temp=matrix(nrow =dim(X)[1],ncol=dim(X)[2] )
for (i in 1:dim(X)[1]){ # leave one out
  for (j in 1:dim(X)[2]){
    fit <- glm(Y~X,data = data.frame(Y=as.factor(Y[-i]),X = X[-i,j]),family=binomial()) # On apprend sans la ieme observation
    prob <- predict(fit,newdata=data.frame(X=X[i,j]), type="response") # On predit la ieme observation
    temp[i,j] <- prob # vecteur proba prÃ©dite d'appartenir Ã une condition
  }
}

BS <- NULL
for (k in 1:dim(X)[2]){ # gene k
  BS[k] <- brier(Y, temp[,k])$bs # somme de la diffÃ©rence au carrÃ© entre la proba pour chaque obs et la condition
}
BS=cbind(colnames(X),BS)
colnames(BS)=c("ID","BS_v2")
dearseq_brier=merge(genewise_dearseq_cold,BS,by="ID")
dearseq_brier=cbind(dearseq_brier,rank(dearseq_brier[,5]))
dearseq_brier=as.data.frame(dearseq_brier)
colnames(dearseq_brier)=c("ID","pvalue","rank_pvalue","raw_pvalue","BS","rank_BS")
```
```{r,echo=F,eval=T,message=FALSE,warning=FALSE, fig.cap="Brier score for the genes identified by dearseq for the cold vs diffuse status (with G4)",cache=FALSE}
BS_dearseq_cold=as.numeric(as.character(dearseq_brier[,5]))
df <- data.frame(Brier.score = BS_dearseq_cold,
                 rank=rank(BS_dearseq_cold,ties.method = "random"),
                 method=as.factor(rep("dearseq_asym",length(BS_dearseq_cold))))

g_dearseq_cold <- ggplot(df[df[,3]=="dearseq_asym",], aes(x=rank, y=Brier.score)) +
  theme_minimal() +
  # scale_color_manual( values = c("#CCCCCC", grDevices::heat.colors(n=2)[3])) +
  geom_point(size=1.5,alpha=0.6,color="#FF9900") +
  theme(axis.title = element_text(size=16), axis.text = element_text(size=12), legend.title = element_text(size=16), legend.text = element_text(size=14)) +
  #  guides(color = guide_legend(override.aes = list(alpha=1))) +
  geom_rug(alpha=0.5,color="#FF9900")  + 
  ylab("Brier Score")
g_dearseq_cold
```


### Without G4 

We will then look protein per protein if there is a difference between cold and diffuse. For that, we use the dearseq package. 



```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
table_proteo=table_proteo_sans
output_sans=ifelse(table_proteo[,3]=="COLD",0,1)
output=output_sans
Y=matrix(output,ncol=1,nrow=length(output))

scores=dear_seq(exprmat=t(as.matrix(raw_counts_sans)),variables2test = Y,preprocessed = FALSE,which_test = "asymptotic")


genewise_dearseq_cold_sans=as.data.frame(cbind(rownames(scores$pvals),scores$pvals[,2],rank(scores$pvals[,2]),scores$pvals[,1]))
genewise_dearseq_cold_sans[,1]=as.character(genewise_dearseq_cold_sans[,1])
genewise_dearseq_cold_sans[,2]=as.numeric(as.character(genewise_dearseq_cold_sans[,2]))
genewise_dearseq_cold_sans[,3]=as.numeric(as.character(genewise_dearseq_cold_sans[,3]))
genewise_dearseq_cold_sans[,4]=as.numeric(as.character(genewise_dearseq_cold_sans[,4]))
genewise_dearseq_cold_sans=genewise_dearseq_cold_sans[order(genewise_dearseq_cold_sans[,2],decreasing = FALSE),]
colnames(genewise_dearseq_cold_sans)=c("ID","pvalue","rank_pvalue","raw_pvalue")




patients_cold=norm_counts_sans[output_sans==0,]
patients_diffuse=norm_counts_sans[output_sans==1,]
means_diffuse=numeric(length(patients_cold[1,]))
means_cold=numeric(length(patients_cold[1,]))
for (i in 1:length(patients_cold[1,])){
  means_diffuse[i]=mean(patients_diffuse[,i])
  means_cold[i]=mean(patients_cold[,i])
}

fold_change=data.frame(colnames(patients_cold),means_diffuse-means_cold)
colnames(fold_change)=c("ID","log2FoldChange")
total_proteo_cold_sans=merge(genewise_dearseq_cold_sans,fold_change,by="ID")
total_proteo_cold_sans=total_proteo_cold_sans[order(total_proteo_cold_sans[,4],total_proteo_cold_sans[,5]),]

write.xlsx(total_proteo_cold_sans[total_proteo_cold_sans[,2]<0.05,], file = "dearseq_proteo_C_D_ng4.xlsx",col.names = TRUE)

total_proteo_cold_sans$diffexpressed <- "NO"
# if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP" 
total_proteo_cold_sans$diffexpressed[total_proteo_cold_sans$log2FoldChange > 0.6 & total_proteo_cold_sans$pvalue < 0.05] <- "UP"
# if log2Foldchange < (-0.6) and pvalue < 0.05, set as "DOWN"
total_proteo_cold_sans$diffexpressed[total_proteo_cold_sans$log2FoldChange < (-0.6) & total_proteo_cold_sans$pvalue < 0.05] <- "DOWN"

total_proteo_cold_sans$delabel <- NA
total_proteo_cold_sans$delabel[total_proteo_cold_sans$diffexpressed != "NO"] <- total_proteo_cold_sans$ID[total_proteo_cold_sans$diffexpressed != "NO"]


plot1=ggplot(data=total_proteo_cold_sans, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed, label=delabel)) + 
    geom_point() + 
    theme_minimal() +
  geom_text_repel()  + scale_color_manual(values= c("#440154FF", "#1F968BFF", "#DCE319FF")) +ggtitle("Log2 Fold change in function of pvalues")

plot2=ggplot(data=total_proteo_cold_sans,aes(x=log2FoldChange))+geom_density()+xlim(c(-5,5))+ theme_minimal()+ggtitle("Distribution of Log2FoldChange") 

```

dearseq identifies `r dim(genewise_dearseq_cold_sans[genewise_dearseq_cold_sans[,2]<0.05,])[1]` proteins that are differentially expressed between cold and diffuse. Here are the first genes in term of p-value:

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
rownames(total_proteo_cold_sans)=NULL
 kable_classic(kbl(x=total_proteo_cold_sans[1:10,c(1,2,3,4,5)],caption = "Cold vs Diffuse genes"),full_width = F, html_font = "Cambria")

```


We can look at the vulcano plot and also the distribution of the log2 Fold change, in the following figure. The vulcano plot shows the log10(pvalue) in function of the log2 Fold change, which allows to see if the significant genes are up regulated or down regulated (cold as base). A positive log2 Fold change indicates that the gene is, on average, more expressed among the diffuse patients, while a negative log2 Fold change indicates that the gene is more expressed among the cold patients. 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,fig.cap="Vulcano plot and distribution of log2 Fold change",cache=FALSE}
plot1+plot2
```

We can observe the brier score associated with these genes. The brier score is an indicator of the prediction aibility, between 1 and 0, 0 being a perfect predictor and 1 a poor predictor: 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}


Y <- output_sans
X <- norm_counts_sans[,colnames(norm_counts_sans)%in%genewise_dearseq_cold_sans[genewise_dearseq_cold_sans[,2]<0.05,1]]
temp=matrix(nrow =dim(X)[1],ncol=dim(X)[2] )
for (i in 1:dim(X)[1]){ # leave one out
  for (j in 1:dim(X)[2]){
    fit <- glm(Y~X,data = data.frame(Y=as.factor(Y[-i]),X = X[-i,j]),family=binomial()) # On apprend sans la ieme observation
    prob <- predict(fit,newdata=data.frame(X=X[i,j]), type="response") # On predit la ieme observation
    temp[i,j] <- prob # vecteur proba prÃ©dite d'appartenir Ã une condition
  }
}

BS_sans <- NULL
for (k in 1:dim(X)[2]){ # gene k
  BS_sans[k] <- brier(Y, temp[,k])$bs # somme de la diffÃ©rence au carrÃ© entre la proba pour chaque obs et la condition
}
BS_sans=cbind(colnames(X),BS_sans)
colnames(BS_sans)=c("ID","BS_v2")
dearseq_brier_sans=merge(genewise_dearseq_cold_sans,BS_sans,by="ID")
dearseq_brier_sans=cbind(dearseq_brier_sans,rank(dearseq_brier_sans[,5]))
dearseq_brier_sans=as.data.frame(dearseq_brier_sans)
colnames(dearseq_brier_sans)=c("ID","pvalue","rank_pvalue","raw_pvalue","BS","rank_BS")
```
```{r,echo=F,eval=T,message=FALSE,warning=FALSE, fig.cap="Brier score for the genes identified by dearseq for the cold vs diffuse status (without G4)",cache=FALSE}
BS_dearseq_cold_sans=as.numeric(as.character(dearseq_brier_sans[,5]))
df <- data.frame(Brier.score = BS_dearseq_cold_sans,
                 rank=rank(BS_dearseq_cold_sans,ties.method = "random"),
                 method=as.factor(rep("dearseq_asym",length(BS_dearseq_cold_sans))))

g_dearseq_cold_sans <- ggplot(df[df[,3]=="dearseq_asym",], aes(x=rank, y=Brier.score)) +
  theme_minimal() +
  # scale_color_manual( values = c("#CCCCCC", grDevices::heat.colors(n=2)[3])) +
  geom_point(size=1.5,alpha=0.6,color="#FF9900") +
  theme(axis.title = element_text(size=16), axis.text = element_text(size=12), legend.title = element_text(size=16), legend.text = element_text(size=14)) +
  #  guides(color = guide_legend(override.aes = list(alpha=1))) +
  geom_rug(alpha=0.5,color="#FF9900")  + 
  ylab("Brier Score")
g_dearseq_cold_sans
```

## {- .unlisted .unnumbered}

We can compare the raw pvalues, with or without G4 patients, with a comparison of the pvalues, and two bland altman plots (normal and log scale)

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE, fig.cap="Comparison of the pvalues with and without G4 patients"}

scat_comparaison <- function(data_1,data_2,name_1,name_2){
  data_1 <- as.data.frame(data_1)
  data_1[,2]=data_1[,2]+rep(0.001,length(data_1[,2]))
  data_2 <- as.data.frame(data_2)
  data_2[,2]=data_2[,2]+rep(0.001,length(data_2[,2]))

  data <- merge(data_1,data_2,by="ID")
  colnames(data) <- c("gene","avec","sans")

  b <- ggplot(data, aes(x = avec, y = sans))
  plot <- b + geom_point(size = 0.5, alpha=0.1) + geom_abline(intercept = 0, slope = 1, color="blue") + scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x), labels = trans_format("log10", math_format(10^.x))) + scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x), labels = trans_format("log10", math_format(10^.x))) + annotation_logticks() + xlab(name_1) + ylab(name_2)
  return(plot)
}

scat_comparaison(genewise_dearseq_cold[,c(1,4)],genewise_dearseq_cold_sans[,c(1,4)],"with G4","without G4")

comparaison=merge(genewise_dearseq_cold[,c(1,4)],genewise_dearseq_cold_sans[,c(1,4)],by="ID")
# blandr.draw(comparaison[,2],comparaison[,3], ciDisplay = FALSE , ciShading = FALSE )
# log_avec=log(comparaison[,2]+rep(0.001,length(comparaison[,2])))
# log_sans=log(comparaison[,3]+rep(0.001,length(comparaison[,2])))
# blandr.draw(log_avec,log_sans, ciDisplay = FALSE , ciShading = FALSE )



dfEval <- data.frame(Dupont = comparaison[,2], Dupond = comparaison[,3])



plotbland1=dfEval %>%
  mutate(diff = Dupont - Dupond,
         mean = (Dupont + Dupond)/2) %>%
  mutate(mean_diff = mean(diff, na.rm = T),
         sd_diff = sd(diff, na.rm = T),
         sdsup = mean_diff + 1.96 * sd_diff,
         sdinf = mean_diff - 1.96 * sd_diff) %>%
  ggplot(mapping = aes(x = mean, y = diff)) +
  geom_point(alpha=0.05) +
  geom_line(aes(y = mean_diff, lty = "mean")) +
  geom_line(aes(y = 0), color = "red") +
  geom_line(aes(y = sdsup, lty = "sd")) +
  geom_line(aes(y = sdinf, lty = "sd")) +
  scale_linetype_manual("", c("Mean of differences", "Mean +/- 1.96 sd"), values = c(1,2)) +
  theme_bw() +
  theme(legend.position = "bottom") +
  labs(x = "Mean", y = "Difference")+ggtitle("Bland Altman")


plotbland2=dfEval %>%
  mutate(diff = abs(Dupont - Dupond),
         mean = (Dupont + Dupond)/2) %>%
  mutate(mean_diff = mean(diff, na.rm = T),
         sd_diff = sd(diff, na.rm = T),
         sdsup = mean_diff + 1.96 * sd_diff,
         sdinf = mean_diff - 1.96 * sd_diff) %>%
  ggplot(mapping = aes(x = mean, y = diff)) +
  geom_point(alpha=0.05) +
  geom_line(aes(y = mean_diff, lty = "mean")) +
  geom_line(aes(y = 0), color = "red") +
  geom_line(aes(y = sdsup, lty = "sd")) +
  geom_line(aes(y = sdinf, lty = "sd")) +
  scale_linetype_manual("", c("Mean of differences", "Mean +/- 1.96 sd"), values = c(1,2)) +
  theme_bw() +
  theme(legend.position = "bottom") +
  labs(x = "Mean", y = "|Difference|")+ggtitle("Bland Altman (log scale)")+ scale_x_continuous(trans = 'log10') +
  scale_y_continuous(trans = 'log10')

plotbland1+plotbland2



```
There seems to be similar pvalues with and without G4 patients.

## Astro vs Oligo {.tabset .tabset-pills}
We also have the Astro or Oligo status of the patients, with a total of 37 Astro patients and 56 Oligo patients. The following table presents the different characteristics: 

### With G4

```{r, eval=T,echo=FALSE,message=FALSE,warning=FALSE,cache=FALSE}
table_proteo=table_proteo_avec
table_na=table_proteo_avec[!is.na(table_proteo_avec[,4])&!is.na(table_proteo_avec[,6]),]
Age=c(round(mean(table_na[table_na[,8]=="oligo",7]),1),round(mean(table_na[table_na[,8]=="astro",7]),1))
Female=c(table(table_na[table_na[,8]=="oligo",6])[1],table(table_na[table_na[,8]=="astro",6])[1])
Male=c(table(table_na[table_na[,8]=="oligo",6])[2],table(table_na[table_na[,8]=="astro",6])[2])
Event=c(table(table_na[table_na[,8]=="oligo",4])[2],table(table_na[table_na[,8]=="astro",4])[2])
No_event=c(table(table_na[table_na[,8]=="oligo",4])[1],table(table_na[table_na[,8]=="astro",4])[1])
Median_followup=c(round(summary(table_na[table_na[,8]=="oligo",5])[3],1),round(summary(table_na[table_na[,8]=="astro",5])[3],1))
oligo_astro_table=rbind(Age,Female,Male,Event,No_event,Median_followup)
colnames(oligo_astro_table)=c("Oligo (n=56)","Astro (n=37)")
rownames(oligo_astro_table)=c("Age (mean)","Female","Male","Event","No event", "Median follow up (years)")
```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)

 kable_classic(kbl(x=oligo_astro_table,caption = "Oligo vs Astro patients"),full_width = F, html_font = "Cambria")

```
The same PCA analysis can be done, this time by highlighting the oligo vs astro status : 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,results=FALSE,fig.show='hide',cache=FALSE}
table_proteo=table_proteo_avec
table_temp=table_proteo_avec[!is.na(table_proteo_avec[,8]),]
counts_pca <- raw_counts[!is.na(table_proteo[,8]),]
counts_pca <- round((counts_pca+1),0)
counts_pca <- vst(t(counts_pca))

counts_pca_scaled <- base::scale(t(counts_pca), center=TRUE, scale=TRUE)

res.pca=PCA(counts_pca_scaled , graph=FALSE)

#plot.PCA(res.pca, axes=c(1, 2), choix="ind")
Astro_oligo=as.factor(table_temp[,8])
G4=as.factor(table_temp[,9])
pca_data <- cbind.data.frame(res.pca$ind$coord[, 1:4],as.factor(table_temp[,8]),table_temp[,2])
colnames(pca_data)=c("Dim.1","Dim.2","Dim.3","Dim.4","Astro_oligo","ID")
table_coord=res.pca$ind$coord[,c(1,2)]
particuliers=table_coord[table_coord[,1]>quantile(table_coord[,1],0.97)|table_coord[,1]<quantile(table_coord[,1],0.03)|table_coord[,2]>quantile(table_coord[,2],0.98)|table_coord[,2]<quantile(table_coord[,2],0.02),]

for (i in 1:length(pca_data[,1])){
  if (!rownames(pca_data)[i]%in%rownames(particuliers)){
    pca_data[i,6]=""
    }
}
p <- ggplot(pca_data, aes(x=Dim.1, y=Dim.2, colour = as.factor(table_temp[,8]),shape=G4,label=ID)) +
 # viridis::scale_color_viridis("Cold_diffuse", discrete = TRUE,end = 0.8)
  #  scale_color_manual(values= c("#E7B800", "#2E9FDF"))+
  geom_text_repel() +
   # scale_color_manual(values= c("#E7B800", "#2E9FDF"))+
  theme_bw() +
  geom_point(size=2, alpha=1) +
  guides(color=guide_legend(override.aes = list(alpha=1),title="Oligo vs Astro")) +
  xlab(label=paste0("PC1 (", round(res.pca$eig[1,2], 2),"% of total variance explained)")) +
  ylab(label=paste0("PC2 (", round(res.pca$eig[2,2], 2), "% of total variance explained)")) +
  ggtitle("", subtitle = "First factorial plan")

```


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="PCA of the proteomics data, with G4 patients (vst transformation and scaled data)",cache=FALSE}
p
```


We can look at the characteristics of the patients outside of the majority of the patients:


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE,cache=FALSE}
datas=table_proteo_avec
table_particulier=datas[datas[,1]%in%rownames(particuliers),]
table_particulier=table_particulier[,c(2,6,7,3,8,5,4)]
table_particulier[,6]=round(table_particulier[,6],2)
rownames(table_particulier)=NULL
```


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=table_particulier,caption = "Characteristics patients outsite of PCA majority"),full_width = F, html_font = "Cambria")

```



### Without G4
Without G4 patients, we have 33 Astro patients and 56 Oligo patients

```{r, eval=T,echo=FALSE,message=FALSE,warning=FALSE,cache=FALSE}
table_proteo=table_proteo_sans
table_na=table_proteo[!is.na(table_proteo[,4])&!is.na(table_proteo[,6]),]
Age=c(round(mean(table_na[table_na[,8]=="oligo",7]),1),round(mean(table_na[table_na[,8]=="astro",7]),1))
Female=c(table(table_na[table_na[,8]=="oligo",6])[1],table(table_na[table_na[,8]=="astro",6])[1])
Male=c(table(table_na[table_na[,8]=="oligo",6])[2],table(table_na[table_na[,8]=="astro",6])[2])
Event=c(table(table_na[table_na[,8]=="oligo",4])[2],table(table_na[table_na[,8]=="astro",4])[2])
No_event=c(table(table_na[table_na[,8]=="oligo",4])[1],table(table_na[table_na[,8]=="astro",4])[1])
Median_followup=c(round(summary(table_na[table_na[,8]=="oligo",5])[3],1),round(summary(table_na[table_na[,8]=="astro",5])[3],1))
oligo_astro_table=rbind(Age,Female,Male,Event,No_event,Median_followup)
colnames(oligo_astro_table)=c("Oligo (n=56)","Astro (n=33)")
rownames(oligo_astro_table)=c("Age (mean)","Female","Male","Event","No event", "Median follow up (years)")
```

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)

 kable_classic(kbl(x=oligo_astro_table,caption = "Oligo vs Astro patients"),full_width = F, html_font = "Cambria")

```
The same PCA analysis can be done, this time by highlighting the oligo vs astro status : 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,results=FALSE,fig.show='hide',cache=FALSE}
table_temp=table_proteo_sans[!is.na(table_proteo_sans[,8]),]
counts_pca <- raw_counts_sans[!is.na(table_proteo_sans[,8]),]
counts_pca <- round((counts_pca+1),0)
counts_pca <- vst(t(counts_pca))

counts_pca_scaled <- base::scale(t(counts_pca), center=TRUE, scale=TRUE)
res.pca=PCA(counts_pca_scaled , graph=FALSE)
#plot.PCA(res.pca, axes=c(1, 2), choix="ind")
Astro_oligo=as.factor(table_temp[,8])
G4=as.factor(table_temp[,9])
pca_data <- cbind.data.frame(res.pca$ind$coord[, 1:4],as.factor(table_temp[,8]),table_temp[,2])
colnames(pca_data)=c("Dim.1","Dim.2","Dim.3","Dim.4","Astro_oligo","ID")
table_coord=res.pca$ind$coord[,c(1,2)]
particuliers=table_coord[table_coord[,1]>quantile(table_coord[,1],0.97)|table_coord[,1]<quantile(table_coord[,1],0.03)|table_coord[,2]>quantile(table_coord[,2],0.98)|table_coord[,2]<quantile(table_coord[,2],0.02),]

for (i in 1:length(pca_data[,1])){
  if (!rownames(pca_data)[i]%in%rownames(particuliers)){
    pca_data[i,6]=""
    }
}
p <- ggplot(pca_data, aes(x=Dim.1, y=Dim.2, colour = as.factor(table_temp[,8]),label=ID)) +
 # viridis::scale_color_viridis("Cold_diffuse", discrete = TRUE,end = 0.8)
  #  scale_color_manual(values= c("#E7B800", "#2E9FDF"))+
  geom_text_repel() +
  theme_bw() +
  geom_point(size=2, alpha=1) +
  guides(color=guide_legend(override.aes = list(alpha=1),title="Oligo vs Astro")) +
  xlab(label=paste0("PC1 (", round(res.pca$eig[1,2], 2),"% of total variance explained)")) +
  ylab(label=paste0("PC2 (", round(res.pca$eig[2,2], 2), "% of total variance explained)")) +
  ggtitle("", subtitle = "First factorial plan")

```


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="PCA of the proteomics data, without G4 patients (vst transformation and scaled data)",cache=FALSE}
p
```


We can look at the characteristics of the patients outside of the majority of the patients:


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
datas=table_proteo_sans
table_particulier=datas[datas[,1]%in%rownames(particuliers),]
table_particulier=table_particulier[,c(2,6,7,3,8,5,4)]
table_particulier[,6]=round(table_particulier[,6],2)
rownames(table_particulier)=NULL
```


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
library(kableExtra)
# dt %>%
#   kbl(caption = "Cold vs Diffuse patients") %>%
#   kable_classic(full_width = F, html_font = "Cambria")
 kable_classic(kbl(x=table_particulier,caption = "Characteristics patients outsite of PCA majority"),full_width = F, html_font = "Cambria")

```


## {- .tabset .tabset-pills .unnumbered .unlisted}

We look now at the proteins selected by dearseq

### With G4 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
table_proteo=table_proteo_avec
table_temp=table_proteo_avec[!is.na(table_proteo_avec[,8]),]

output_oligo=ifelse(table_temp[,8]=="oligo",0,1)
Y=matrix(output_oligo,ncol=1,nrow=length(output_oligo))

scores=dear_seq(exprmat=t(as.matrix( counts[table_temp[,1],])),variables2test = Y,preprocessed = FALSE,which_test = "asymptotic")


genewise_dearseq_astro=as.data.frame(cbind(rownames(scores$pvals),scores$pvals[,2],rank(scores$pvals[,2]),scores$pvals[,1]))
genewise_dearseq_astro[,1]=as.character(genewise_dearseq_astro[,1])
genewise_dearseq_astro[,2]=as.numeric(as.character(genewise_dearseq_astro[,2]))
genewise_dearseq_astro[,3]=as.numeric(as.character(genewise_dearseq_astro[,3]))
genewise_dearseq_astro[,4]=as.numeric(as.character(genewise_dearseq_astro[,4]))
genewise_dearseq_astro=genewise_dearseq_astro[order(genewise_dearseq_astro[,2],decreasing = FALSE),]
colnames(genewise_dearseq_astro)=c("ID","pvalue","rank_pvalue","raw_pvalue")
#write.xlsx(genewise_dearseq_astro[genewise_dearseq_astro[,2]<0.05,], file = "dearseq_proteo_O_A_ng4.xlsx",col.names = TRUE)



patients_astro=norm_counts[output_oligo==0,]
patients_oligo=norm_counts[output_oligo==1,]
means_oligo=numeric(length(patients_oligo[1,]))
means_astro=numeric(length(patients_astro[1,]))
for (i in 1:length(patients_cold[1,])){
  means_oligo[i]=mean(patients_oligo[,i])
  means_astro[i]=mean(patients_astro[,i])
}

fold_change=data.frame(colnames(patients_astro),means_oligo-means_astro)
colnames(fold_change)=c("ID","log2FoldChange")
total_proteo_astro=merge(genewise_dearseq_astro,fold_change,by="ID")
total_proteo_astro=total_proteo_astro[order(total_proteo_astro[,4],total_proteo_astro[,5]),]



total_proteo_astro$diffexpressed <- "NO"
# if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP" 
total_proteo_astro$diffexpressed[total_proteo_astro$log2FoldChange > 0.6 & total_proteo_astro$pvalue < 0.05] <- "UP"
# if log2Foldchange < -0.6 and pvalue < 0.05, set as "DOWN"
total_proteo_astro$diffexpressed[total_proteo_astro$log2FoldChange <  (-0.6) & total_proteo_astro$pvalue < 0.05] <- "DOWN"

total_proteo_astro$delabel <- NA
total_proteo_astro$delabel[total_proteo_astro$diffexpressed != "NO"] <- total_proteo_astro$ID[total_proteo_astro$diffexpressed != "NO"]


plot1=ggplot(data=total_proteo_astro, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed, label=delabel)) + 
    geom_point() + 
    theme_minimal() +
  geom_text_repel()  + scale_color_manual(values=  c( "#1F968BFF", "#DCE319FF","#440154FF")) +ggtitle("Log2 Fold change in function of pvalues")

plot2=ggplot(data=total_proteo_astro,aes(x=log2FoldChange))+geom_density()+xlim(c(-2,2))+ theme_minimal()+ggtitle("Distribution of Log2FoldChange") 
```

To identify which genes are differentialy expressed between the oligo and astro groups, we perform dearseq, which finds `r dim(genewise_dearseq_astro[genewise_dearseq_astro[,2]<0.05,])[1]` proteins. Here are the first proteins in term of p-value:

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
rownames(total_proteo_astro)=NULL
 kable_classic(kbl(x=total_proteo_astro[1:10,c(1,2,3,4,5)],caption = "Oligo vs Astro genes"),full_width = F, html_font = "Cambria")

```

We can look at the vulcano plot and also the distribution of the log2 Fold change, in the following figure. The vulcano plot shows the log10(pvalue) in function of the log2 Fold change, which allows to see if the significant genes are up regulated or down regulated (astro as base). A positive log2 Fold change indicates that the gene is, on average, more expressed among the oligo patients, while a negative log2 Fold change indicates that the gene is more expressed among the astro patients. 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,fig.cap="Vulcano plot and distribution of log2 Fold change",cache=FALSE}
plot1+plot2
```

### Without G4


```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
table_proteo=table_proteo_sans
table_temp=table_proteo_sans[!is.na(table_proteo_sans[,8]),]

output_oligo_sans=ifelse(table_temp[,8]=="oligo",0,1)
Y=matrix(output_oligo_sans,ncol=1,nrow=length(output_oligo_sans))

scores=dear_seq(exprmat=t(as.matrix( raw_counts_sans[table_temp[,1],])),variables2test = Y,preprocessed = FALSE,which_test = "asymptotic")


genewise_dearseq_astro_sans=as.data.frame(cbind(rownames(scores$pvals),scores$pvals[,2],rank(scores$pvals[,2]),scores$pvals[,1]))
genewise_dearseq_astro_sans[,1]=as.character(genewise_dearseq_astro_sans[,1])
genewise_dearseq_astro_sans[,2]=as.numeric(as.character(genewise_dearseq_astro_sans[,2]))
genewise_dearseq_astro_sans[,3]=as.numeric(as.character(genewise_dearseq_astro_sans[,3]))
genewise_dearseq_astro_sans[,4]=as.numeric(as.character(genewise_dearseq_astro_sans[,4]))
genewise_dearseq_astro_sans=genewise_dearseq_astro_sans[order(genewise_dearseq_astro_sans[,2],decreasing = FALSE),]
colnames(genewise_dearseq_astro_sans)=c("ID","pvalue","rank_pvalue","raw_pvalue")
#write.xlsx(genewise_dearseq_astro[genewise_dearseq_astro[,2]<0.05,], file = "dearseq_proteo_O_A_ng4.xlsx",col.names = TRUE)




patients_astro=norm_counts_sans[output_oligo_sans==0,]
patients_oligo=norm_counts_sans[output_oligo_sans==1,]
means_oligo=numeric(length(patients_oligo[1,]))
means_astro=numeric(length(patients_astro[1,]))
for (i in 1:length(patients_cold[1,])){
  means_oligo[i]=mean(patients_oligo[,i])
  means_astro[i]=mean(patients_astro[,i])
}

fold_change=data.frame(colnames(patients_astro),means_oligo-means_astro)
colnames(fold_change)=c("ID","log2FoldChange")
total_proteo_astro_sans=merge(genewise_dearseq_astro_sans,fold_change,by="ID")
total_proteo_astro_sans=total_proteo_astro_sans[order(total_proteo_astro_sans[,4],total_proteo_astro_sans[,5]),]



total_proteo_astro_sans$diffexpressed <- "NO"
# if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP" 
total_proteo_astro_sans$diffexpressed[total_proteo_astro_sans$log2FoldChange > 0.6 & total_proteo_astro_sans$pvalue < 0.05] <- "UP"
# if log2Foldchange <  (-0.6) and pvalue < 0.05, set as "DOWN"
total_proteo_astro_sans$diffexpressed[total_proteo_astro_sans$log2FoldChange <  (-0.6) & total_proteo_astro_sans$pvalue < 0.05] <- "DOWN"

total_proteo_astro_sans$delabel <- NA
total_proteo_astro_sans$delabel[total_proteo_astro_sans$diffexpressed != "NO"] <- total_proteo_astro_sans$ID[total_proteo_astro_sans$diffexpressed != "NO"]


plot1=ggplot(data=total_proteo_astro_sans, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed, label=delabel)) + 
    geom_point() + 
    theme_minimal() +
  geom_text_repel()  + scale_color_manual(values=  c( "#1F968BFF", "#DCE319FF","#440154FF")) +ggtitle("Log2 Fold change in function of pvalues")

plot2=ggplot(data=total_proteo_astro_sans,aes(x=log2FoldChange))+geom_density()+xlim(c(-2,2))+ theme_minimal()+ggtitle("Distribution of Log2FoldChange") 
```

To identify which genes are differentialy expressed between the oligo and astro groups, we perform dearseq, which finds `r dim(genewise_dearseq_astro[genewise_dearseq_astro[,2]<0.05,])[1]` proteins. Here are the first proteins in term of p-value:

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
rownames(total_proteo_astro_sans)=NULL
 kable_classic(kbl(x=total_proteo_astro_sans[1:10,c(1,2,3,4,5)],caption = "Oligo vs Astro genes"),full_width = F, html_font = "Cambria")

```

We can look at the vulcano plot and also the distribution of the log2 Fold change, in the following figure. The vulcano plot shows the log10(pvalue) in function of the log2 Fold change, which allows to see if the significant genes are up regulated or down regulated (astro as base). A positive log2 Fold change indicates that the gene is, on average, more expressed among the oligo patients, while a negative log2 Fold change indicates that the gene is more expressed among the astro patients. 

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,fig.cap="Vulcano plot and distribution of log2 Fold change",cache=FALSE}
plot1+plot2
```
## {- .unlisted .unnumbered}

We can compare the raw pvalues with or without G, with a comparison of the pvalues and two bland altman plots (with and without G4 patients)

```{r,eval=T,echo=F,message=FALSE,warning=FALSE,fig.cap="comparison of pvalues, with and without G4",cache=FALSE}


scat_comparaison <- function(data_1,data_2,name_1,name_2){
  data_1 <- as.data.frame(data_1)
  data_1[,2]=data_1[,2]+rep(0.001,length(data_1[,2]))
  data_2 <- as.data.frame(data_2)
  data_2[,2]=data_2[,2]+rep(0.001,length(data_2[,2]))

  data <- merge(data_1,data_2,by="ID")
  colnames(data) <- c("gene","avec","sans")

  b <- ggplot(data, aes(x = avec, y = sans))
  plot <- b + geom_point(size = 0.5, alpha=0.1) + geom_abline(intercept = 0, slope = 1, color="blue") + scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x), labels = trans_format("log10", math_format(10^.x))) + scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x), labels = trans_format("log10", math_format(10^.x))) + annotation_logticks() + xlab(name_1) + ylab(name_2)
  return(plot)
}

scat_comparaison(genewise_dearseq_astro[,c(1,4)],genewise_dearseq_astro_sans[,c(1,4)],"with G4","without G4")
comparaison=merge(genewise_dearseq_astro[,c(1,4)],genewise_dearseq_astro_sans[,c(1,4)],by="ID")
# blandr.draw(comparaison[,2],comparaison[,3], ciDisplay = FALSE , ciShading = FALSE )
# log_avec=log(comparaison[,2]+rep(0.001,length(comparaison[,2])))
# log_sans=log(comparaison[,3]+rep(0.001,length(comparaison[,2])))
# blandr.draw(log_avec,log_sans, ciDisplay = FALSE , ciShading = FALSE )



dfEval <- data.frame(Dupont = comparaison[,2], Dupond = comparaison[,3])



plotbland1=dfEval %>%
  mutate(diff = Dupont - Dupond,
         mean = (Dupont + Dupond)/2) %>%
  mutate(mean_diff = mean(diff, na.rm = T),
         sd_diff = sd(diff, na.rm = T),
         sdsup = mean_diff + 1.96 * sd_diff,
         sdinf = mean_diff - 1.96 * sd_diff) %>%
  ggplot(mapping = aes(x = mean, y = diff)) +
  geom_point(alpha=0.05) +
  geom_line(aes(y = mean_diff, lty = "mean")) +
  geom_line(aes(y = 0), color = "red") +
  geom_line(aes(y = sdsup, lty = "sd")) +
  geom_line(aes(y = sdinf, lty = "sd")) +
  scale_linetype_manual("", c("Mean of differences", "Mean +/- 1.96 sd"), values = c(1,2)) +
  theme_bw() +
  theme(legend.position = "bottom") +
  labs(x = "Mean", y = "Difference")+ggtitle("Bland Altman")


plotbland2=dfEval %>%
  mutate(diff = abs(Dupont - Dupond),
         mean = (Dupont + Dupond)/2) %>%
  mutate(mean_diff = mean(diff, na.rm = T),
         sd_diff = sd(diff, na.rm = T),
         sdsup = mean_diff + 1.96 * sd_diff,
         sdinf = mean_diff - 1.96 * sd_diff) %>%
  ggplot(mapping = aes(x = mean, y = diff)) +
  geom_point(alpha=0.05) +
  geom_line(aes(y = mean_diff, lty = "mean")) +
  geom_line(aes(y = 0), color = "red") +
  geom_line(aes(y = sdsup, lty = "sd")) +
  geom_line(aes(y = sdinf, lty = "sd")) +
  scale_linetype_manual("", c("Mean of differences", "Mean +/- 1.96 sd"), values = c(1,2)) +
  theme_bw() +
  theme(legend.position = "bottom") +
  labs(x = "Mean", y = "|Difference|")+ggtitle("Bland Altman (log scale)")+ scale_x_continuous(trans = 'log10') +
  scale_y_continuous(trans = 'log10')

plotbland1+plotbland2

```


In both case, no proteins is found significant, and the presence or absence of G4 does not changes deeply the pvalues.


## Survival analysis {.tabset .tabset-pills}

### With G4 

When can first perform a simple survival analysis by using a cox model on the base covariates: 

```{r, message=FALSE,warning=FALSE,cache=FALSE}
table_proteo=table_proteo_avec
modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=table_proteo)
summary(modele_base)

```

The p-value is significant for the cold vs diffuse, and the AIC of the model is `r extractAIC(modele_base)[2]`.
We are now interested to identify the genes that are linked with the risk of recurrence. For that, we use random forest methods. 

```{r,eval=T,echo=F, message=FALSE,warning=FALSE,cache=TRUE,fig.cap="First proteins in term of importances for the random forests (with G4)",cache=TRUE}
table_proteo=table_proteo_avec
 v.obj_baseline<- rfsrc(Surv(time=as.numeric(Time), event=as.numeric(Reccurence))~.,data=table_proteo[,-c(1,2)],ntree=2000,block.size=50,mtry=dim(table_proteo[,1])[2])
  importance_baseline=vimp.rfsrc(v.obj_baseline)
  
  
  datas_rf=cbind(table_proteo[,-c(1,2)],norm_counts)
  v.obj<- rfsrc(Surv(time=as.numeric(Time), event=as.numeric(Reccurence))~.,data=datas_rf,ntree=2000,block.size=50,mtry=(dim(datas_rf)[2])/2)
  importance=vimp.rfsrc(v.obj)
  importances=importance$importance
  selected=names(importances[importances>=0.0003])
  table_rf_avec=as.data.frame(cbind(selected,importances[importances>=0.0003]))
    colnames(table_rf_avec)=c("ID","importance")
  write.xlsx(table_rf_avec, file = "survival_proteo_wg4.xlsx",col.names = TRUE)
  total_importances=as.data.frame(cbind(names(importances),importances))

dearseq=genewise_dearseq_cold
results_rf=merge(table_rf_avec,dearseq)
results_rf=results_rf[,c(1,2,3,5)]
results_rf[,2]=round(as.numeric(results_rf[,2]),digits = 5)



results_rf_avec=results_rf[order(results_rf[,2],decreasing = TRUE),]
results_rf_avec[,1]=factor(results_rf_avec[,1],levels=c(results_rf_avec[1:dim(results_rf_avec)[1],1]))

ggplot(results_rf_avec,aes(x=ID,y=importance))+
  geom_point()+ scale_colour_viridis_d() + xlab("selected proteins with survival random forest")+ theme(axis.text.x = element_blank())



```

We have `r length(selected)` proteins with an importance>0.0003 selected by the random forest method.



### Without G4 

When can first perform a simple survival analysis by using a cox model on the base covariates: 

```{r, message=FALSE,warning=FALSE,cache=FALSE}
table_proteo=table_proteo_sans
modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro,data=table_proteo)
summary(modele_base)

```

The p-value is significant for the cold vs diffuse, and the AIC of the model is `r extractAIC(modele_base)[2]`.
We are now interested to identify the genes that are linked with the risk of recurrence. For that, we use random forest methods. 

```{r,eval=T,echo=F, message=FALSE,warning=FALSE,cache=TRUE,fig.cap="First proteins in term of importances for the random forests (without G4)",cache=TRUE}
table_proteo=table_proteo_sans
 v.obj_baseline<- rfsrc(Surv(time=as.numeric(Time), event=as.numeric(Reccurence))~.,data=table_proteo[,-c(1,2)],ntree=2000,block.size=50,mtry=dim(table_proteo[,1])[2])
  importance_baseline=vimp.rfsrc(v.obj_baseline)
  
  
  datas_rf=cbind(table_proteo[,-c(1,2)],norm_counts_sans)
  v.obj<- rfsrc(Surv(time=as.numeric(Time), event=as.numeric(Reccurence))~.,data=datas_rf,ntree=2000,block.size=50,mtry=(dim(datas_rf)[2]/2))
  importance=vimp.rfsrc(v.obj)
  importances=importance$importance
  selected=names(importances[importances>=0.0003])
  table_rf_sans=as.data.frame(cbind(selected,importances[importances>=0.0003]))
    colnames(table_rf_sans)=c("ID","importance")
  write.xlsx(table_rf_sans, file = "survival_proteo_ng4.xlsx",col.names = TRUE)
  total_importances_sans=as.data.frame(cbind(names(importances),importances))

dearseq=genewise_dearseq_cold_sans
results_rf_sans=merge(table_rf_sans,dearseq)
results_rf_sans=results_rf_sans[,c(1,2,3,5)]
results_rf_sans[,2]=round(as.numeric(results_rf_sans[,2]),digits = 5)



results_rf_sans=results_rf_sans[order(results_rf_sans[,2],decreasing = TRUE),]
results_rf_sans[,1]=factor(results_rf_sans[,1],levels=c(results_rf_sans[1:dim(results_rf_sans)[1],1]))

ggplot(results_rf_sans,aes(x=ID,y=importance))+
  geom_point()+ scale_colour_viridis_d() + xlab("selected proteins with survival random forest")+ theme(axis.text.x = element_blank())



```

We have `r length(selected)` proteins with an importance>0.0003 selected by the random forest method.

## {- .unliste .unnumbered .tabset .tabset-pills}

As the random forest are, random, the results can change from one time to another. We computed multiple times (7) the random forest and looked at the genes that were similar from one iteration to the other 

### With G4

```{r,echo=F,message=FALSE,warning=FALSE,cache=TRUE}
results_tot=read.table("data/results_rf_proteo_avec2.txt")
results_tot=results_tot[,-1]
results_tot=results_tot[results_tot[,2]>0.0003,]
liste_selected=as.list(NULL)
plots=as.list(NULL)
plots_selected=as.list(NULL)
results=results_tot[results_tot[,3]==1,]
results=results[!duplicated(results[,1]),]
results=results[,c(1,2)]
colnames(results)=c("ID","importance")
liste_merged=results
for (i in 2:8){
  results=results_tot[results_tot[,3]==i,]
  results=results[!duplicated(results[,1]),]
  results=results[,c(1,2)]
  colnames(results)=c("ID","importance")
  liste_merged=merge(liste_merged,results,by="ID")
}
liste_merged_tot=liste_merged
liste_merged=liste_merged[,1]
for (test in 1:8){
  results=results_tot[results_tot[,3]==test,]
  results=results[!duplicated(results[,1]),]
  results=results[,c(1,2)]
  colnames(results)=c("ID","importance")
  dearseq=genewise_dearseq_cold
  results_rf=merge(results,dearseq)
  results_rf=results_rf[,c(1,2,3,5)]
  patients_cold=norm_counts[output_avec==0,colnames(norm_counts)%in%results_rf[,1]]
  patients_diffuse=norm_counts[output_avec==1,colnames(norm_counts)%in%results_rf[,1]]
  means_diffuse=numeric(length(patients_cold[1,]))
  means_cold=numeric(length(patients_cold[1,]))
  for (i in 1:length(patients_cold[1,])){
    means_diffuse[i]=mean(patients_diffuse[,i])
    means_cold[i]=mean(patients_cold[,i])
  }
  fold_change=means_diffuse-means_cold
  results_rf=cbind(results_rf,fold_change)
  results_rf[,2]=round(as.numeric(results_rf[,2]),digits = 5)
  summary(results_rf[,3])
  #### BRIER SCORE : capacitÃ© predictive d'un gÃ¨ne
  # proche de 0 => bon predicteur
  # proche de 1 => mauvais predicteur
  Y <- output_avec
  X <- norm_counts[,colnames(norm_counts)%in%results_rf[,1]]
  temp=matrix(nrow =dim(X)[1],ncol=dim(X)[2] )
  for (i in 1:dim(X)[1]){ # leave one out
    for (j in 1:dim(X)[2]){
      fit <- glm(Y~X,data = data.frame(Y=as.factor(Y[-i]),X = X[-i,j]),family=binomial()) # On apprend sans la ieme observation
      prob <- predict(fit,newdata=data.frame(X=X[i,j]), type="response") # On predit la ieme observation
      temp[i,j] <- prob # vecteur proba prÃ©dite d'appartenir Ã une condition
    }
  }
  BS <- NULL
  for (k in 1:dim(X)[2]){ # gene k
    BS[k] <- brier(Y, temp[,k])$bs # somme de la diffÃ©rence au carrÃ© entre la proba pour chaque obs et la condition
  }
  BS=cbind(colnames(X),BS)
  colnames(BS)=c("ID","BS_v2")
  results_rf=cbind(results_rf,as.numeric(BS[,2]))
  results_rf=results_rf[order(results_rf[,2],decreasing=TRUE),]
  results_rf=results_rf[!is.na(results_rf[,1]),]
  colnames(results_rf)=c("ID","importance","dearseq","dearseq_raw","fold_change","BS")
  datas_total=melt(results_rf,id.var="ID")
  datas_total[,1]=factor(datas_total[,1],levels=c(datas_total[1:dim(results_rf)[1],1]))
  liste=results_rf[results_rf[,3]<0.05|results_rf[,5]>1|results_rf[,5]< (-1),1]
  liste_selected[[test]]=liste
  color=numeric(length(datas_total[,1]))
  #color=ifelse(datas_total[,1]%in%liste,"red","black")
  color=ifelse(datas_total[,1]%in%liste_merged,"blue","black")
  datas_highlight=as.data.frame(cbind(datas_total,color))
  plots[[test]]=ggplot(datas_highlight[datas_highlight[,2]!="dearseq",],aes(x=ID,y=value))+
    geom_point(color=color[datas_highlight[,2]!="dearseq"])+
    facet_grid(variable~.,scale="free")+ scale_colour_viridis_d() + xlab("selected proteins with survival random forest")+ theme(axis.text.x = element_blank())
  
  
  liste=results_rf[results_rf[,3]<0.05|results_rf[,5]>1|results_rf[,5]< (-1),1]
  color=numeric(length(datas_total[,1]))
  color=ifelse(datas_total[,1]%in%liste,"red","black")
  datas_highlight=as.data.frame(cbind(datas_total,color))
  plots_selected[[test]]=ggplot(datas_highlight[datas_highlight[,2]!="dearseq",],aes(x=ID,y=value))+
    geom_point(color=color[datas_highlight[,2]!="dearseq"])+
    facet_grid(variable~.,scale="free")+ scale_colour_viridis_d() + xlab("selected proteins with survival random forest")+ theme(axis.text.x = element_blank())
  
  
}
plot1=(plots[[1]]+plots[[2]])/(plots[[3]]+plots[[4]])
plot2=(plots_selected[[1]]+plots_selected[[2]])/(plots_selected[[3]]+plots_selected[[4]])
```

We can first see which genes are always found as important genes (here in blue)
```{r,echo=F,message=FALSE,warning=FALSE,fig.cap="Repetition of the random forest, with the genes always found (in blue)",cache=FALSE}
plot1
```

Here is a table that show with genes that are always identified and their mean importance, followed by a graph showing how often a gene is identified among all tries
```{r,echo=F,message=FALSE,warning=FALSE,fig.cap="How often each gene is found in the total repetitions",cache=FALSE}
means_tot=numeric(length(liste_merged_tot[,1]))
for (i in 1:length(liste_merged_tot[,1])){
  means_tot[i]=mean(as.numeric(liste_merged_tot[i,-1]))
}
liste_found_avec=as.data.frame(cbind(liste_merged_tot[,1],means_tot))
colnames(liste_found_avec)=c("protein","mean importance")
kable_classic(kbl(x=liste_found_avec,caption = "Proteins always found and their mean importance"),full_width = F, html_font = "Cambria")


results=results_tot[results_tot[,3]==1,]
results=results[!duplicated(results[,1]),]
results=results[,c(1,2)]
colnames(results)=c("ID","importance")
liste_merged2=results
for (i in 2:7){
  results=results_tot[results_tot[,3]==i,]
  results=results[!duplicated(results[,1]),]
  results=results[,c(1,2)]
  colnames(results)=c("ID","importance")
  liste_merged2=merge(liste_merged2,results,all=TRUE)
}

variables=unique(liste_merged2[,1])
percent=numeric(length(variables))
mean_imp=numeric(length(variables))
for (i in 1:length(variables)){
  results_variable=liste_merged2[liste_merged2[,1]==variables[i],]
  percent[i]=100*round(dim(results_variable)[1]/7,2)
  mean_imp[i]=mean(results_variable[,2])
}

table_tot=as.data.frame(cbind(variables,percent,round(mean_imp,6)))


colnames(table_tot)=c("ID","percent","importance")
table_tot=merge(table_tot,total_proteo_cold,by="ID")
table_tot=table_tot[,c(1,2,3,4,7,8)]
table_tot=merge(table_tot,total_proteo_astro,by="ID")
table_tot=table_tot[,c(1,2,3,4,5,6,7,10,11)]
colnames(table_tot)=c("ID","percent","RF_importance","dearseq_cold_diffuse","Log2FC_cold_diffuse","diff_expr_cold_diffuse","dearseq_astro_oligo","Log2FC_astro_oligo","diff_expr_astro_oligo")
table_tot[,2]=as.numeric(table_tot[,2])
 
table_tot=table_tot[order(table_tot[,2],table_tot[,3],decreasing = TRUE),]


table_tot[,2]=as.numeric(table_tot[,2])
table_tot=table_tot[order(table_tot[,2],table_tot[,3],decreasing = TRUE),]
write.xlsx(table_tot,"survival_total_proteo.xlsx",row.names = FALSE,col.names = TRUE)

table_nombres=table(liste_merged2[,1])
table_nombres=as.data.frame(cbind(rep(1,length(table_nombres)),table_nombres))
colnames(table_nombres)=c("number of identifications","occurences")
ggplot(as.data.frame(table_nombres),aes(x=occurences)) + 
  geom_histogram()
```

But can also see which genes are identified as linked with cold/diffuse status (here in red)
```{r,echo=F,message=FALSE,warning=FALSE,fig.cap="Repetition of the random forest, with the genes linked with cold/diffuse status (in red)",cache=FALSE}
plot2
```

### Without G4

```{r,echo=F,message=FALSE,warning=FALSE,cache=TRUE}
results_tot=read.table("data/results_rf_proteo_sans2.txt")
results_tot=results_tot[,-1]
results_tot=results_tot[results_tot[,2]>0.0003,]

liste_selected=as.list(NULL)
plots=as.list(NULL)
plots_selected=as.list(NULL)
results=results_tot[results_tot[,3]==1,]
results=results[!duplicated(results[,1]),]
results=results[,c(1,2)]
colnames(results)=c("ID","importance")
liste_merged=results
for (i in 2:8){
  results=results_tot[results_tot[,3]==i,]
  results=results[!duplicated(results[,1]),]
  results=results[,c(1,2)]
  colnames(results)=c("ID","importance")
  liste_merged=merge(liste_merged,results,by="ID")
}
liste_merged_tot=liste_merged
liste_merged=liste_merged[,1]
for (test in 1:8){
  results=results_tot[results_tot[,3]==test,]
  results=results[!duplicated(results[,1]),]
  results=results[,c(1,2)]
  colnames(results)=c("ID","importance")
  dearseq=genewise_dearseq_cold_sans
  results_rf=merge(results,dearseq)
  results_rf=results_rf[,c(1,2,3,5)]
  patients_cold=norm_counts[output_sans==0,colnames(norm_counts_sans)%in%results_rf[,1]]
  patients_diffuse=norm_counts[output_sans==1,colnames(norm_counts_sans)%in%results_rf[,1]]
  means_diffuse=numeric(length(patients_cold[1,]))
  means_cold=numeric(length(patients_cold[1,]))
  for (i in 1:length(patients_cold[1,])){
    means_diffuse[i]=mean(patients_diffuse[,i])
    means_cold[i]=mean(patients_cold[,i])
  }
  fold_change=means_diffuse-means_cold
  results_rf=cbind(results_rf,fold_change)
  results_rf[,2]=round(as.numeric(results_rf[,2]),digits = 5)
  summary(results_rf[,3])
  #### BRIER SCORE : capacitÃ© predictive d'un gÃ¨ne
  # proche de 0 => bon predicteur
  # proche de 1 => mauvais predicteur
  Y <- output_sans
  X <- norm_counts_sans[,colnames(norm_counts_sans)%in%results_rf[,1]]
  temp=matrix(nrow =dim(X)[1],ncol=dim(X)[2] )
  for (i in 1:dim(X)[1]){ # leave one out
    for (j in 1:dim(X)[2]){
      fit <- glm(Y~X,data = data.frame(Y=as.factor(Y[-i]),X = X[-i,j]),family=binomial()) # On apprend sans la ieme observation
      prob <- predict(fit,newdata=data.frame(X=X[i,j]), type="response") # On predit la ieme observation
      temp[i,j] <- prob # vecteur proba prÃ©dite d'appartenir Ã une condition
    }
  }
  BS <- NULL
  for (k in 1:dim(X)[2]){ # gene k
    BS[k] <- brier(Y, temp[,k])$bs # somme de la diffÃ©rence au carrÃ© entre la proba pour chaque obs et la condition
  }
  BS=cbind(colnames(X),BS)
  colnames(BS)=c("ID","BS_v2")
  results_rf=cbind(results_rf,as.numeric(BS[,2]))
  results_rf=results_rf[order(results_rf[,2],decreasing=TRUE),]
  results_rf=results_rf[!is.na(results_rf[,1]),]
  colnames(results_rf)=c("ID","importance","dearseq","dearseq_raw","fold_change","BS")
  datas_total=melt(results_rf,id.var="ID")
  datas_total[,1]=factor(datas_total[,1],levels=c(datas_total[1:dim(results_rf)[1],1]))
  liste=results_rf[results_rf[,3]<0.05|results_rf[,5]>1|results_rf[,5]< (-1),1]
  liste_selected[[test]]=liste
  color=numeric(length(datas_total[,1]))
  #color=ifelse(datas_total[,1]%in%liste,"red","black")
  color=ifelse(datas_total[,1]%in%liste_merged,"blue","black")
  datas_highlight=as.data.frame(cbind(datas_total,color))
  plots[[test]]=ggplot(datas_highlight[datas_highlight[,2]!="dearseq",],aes(x=ID,y=value))+
    geom_point(color=color[datas_highlight[,2]!="dearseq"])+
    facet_grid(variable~.,scale="free")+ scale_colour_viridis_d() + xlab("selected proteins with survival random forest")+ theme(axis.text.x = element_blank())
  
  
  liste=results_rf[results_rf[,3]<0.05|results_rf[,5]>1|results_rf[,5]< (-1),1]
  color=numeric(length(datas_total[,1]))
  color=ifelse(datas_total[,1]%in%liste,"red","black")
  datas_highlight=as.data.frame(cbind(datas_total,color))
  plots_selected[[test]]=ggplot(datas_highlight[datas_highlight[,2]!="dearseq",],aes(x=ID,y=value))+
    geom_point(color=color[datas_highlight[,2]!="dearseq"])+
    facet_grid(variable~.,scale="free")+ scale_colour_viridis_d() + xlab("selected proteins with survival random forest")+ theme(axis.text.x = element_blank())
  
  
}
plot1=(plots[[1]]+plots[[2]])/(plots[[3]]+plots[[4]])
plot2=(plots_selected[[1]]+plots_selected[[2]])/(plots_selected[[3]]+plots_selected[[4]])
```

We can first see which genes are always found as important genes (here in blue)
```{r,echo=F,message=FALSE,warning=FALSE,fig.cap="Repetition of the random forest, with the genes always found (in blue)",cache=FALSE}
plot1
```

Here is a table that show with genes that are always identified and their mean importance, followed by a graph showing how often a gene is identified among all tries
```{r,echo=F,message=FALSE,warning=FALSE,fig.cap="How often each gene is found in the total repetitions",cache=FALSE}
means_tot=numeric(length(liste_merged_tot[,1]))
for (i in 1:length(liste_merged_tot[,1])){
  means_tot[i]=mean(as.numeric(liste_merged_tot[i,-1]))
}
liste_found_sans=as.data.frame(cbind(liste_merged_tot[,1],means_tot))
colnames(liste_found_sans)=c("protein","mean importance")
kable_classic(kbl(x=liste_found_sans,caption = "Proteins always found and their mean importance"),full_width = F, html_font = "Cambria")


results=results_tot[results_tot[,3]==1,]
results=results[!duplicated(results[,1]),]
results=results[,c(1,2)]
colnames(results)=c("ID","importance")
liste_merged2=results
for (i in 2:7){
  results=results_tot[results_tot[,3]==i,]
  results=results[!duplicated(results[,1]),]
  results=results[,c(1,2)]
  colnames(results)=c("ID","importance")
  liste_merged2=merge(liste_merged2,results,all=TRUE)
}

table_nombres=table(liste_merged2[,1])
table_nombres=as.data.frame(cbind(rep(1,length(table_nombres)),table_nombres))
colnames(table_nombres)=c("number of identifications","occurences")
ggplot(as.data.frame(table_nombres),aes(x=occurences)) + 
  geom_histogram()
```

But can also see which genes are identified as linked with cold/diffuse status (here in red)

```{r,echo=F,message=FALSE,warning=FALSE,fig.cap="Repetition of the random forest, with the genes linked with cold/diffuse status (in red)",cache=FALSE}
plot2
```




## Proteins of interest {.tabset .tabset-pills}

We can see the proteins that are always found, with and without G4 patients, and explore them more: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE}
liste_both=merge(liste_found_avec,liste_found_sans,by="protein")
colnames(liste_both)=c("protein","With_g4","Without g4")
liste_both=liste_both[order(liste_both[,2],decreasing = TRUE),]
rownames(liste_both)=NULL
write.xlsx(liste_both,"survival_proteo_both.xlsx",col.names = TRUE)
kable_classic(kbl(x=liste_both,caption = "Proteins always with and without G4"),full_width = F, html_font = "Cambria")
```

For the 6 firsts proteins of interest, the cox model is computed, followed by the Kaplan Meier curves, with 2 categories (patients with expression bellow the median, and above the median), and the counts per categories (cold/diffuse and astro/oligo)
 
### Q5HYK3 {.tabset .tabset-fade .tabset-pills}

We can explore the Q5HYK3 protein, that is selected with G4 patients (importance `r total_importances[total_importances[,1]=="Q5HYK3",2]` ), and with G4 (importance `r total_importances_sans[total_importances_sans[,1]=="Q5HYK3",2]` )

#### With G4 

A multivariate and univariate cox model is computed to see if the protein is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=table_proteo_avec


modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+Q5HYK3,data=cbind(datas,norm_counts))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~Q5HYK3,data=cbind(datas,norm_counts))
summary(modele_base2)

gene="Q5HYK3"
values=norm_counts[,colnames(norm_counts)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")



```

Here are the Kaplan meier curves with a separation of the patients according to the proteins levels. This allows to see if there is a differences between the patients with high proteins levels and low proteins levels in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the proteins levels for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}


gene=norm_counts[,colnames(norm_counts)=="Q5HYK3"]
datas_gene=as.data.frame(cbind(datas[,c(2,3,5,4)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")



ID2=rep("",length(datas_gene[,2]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()



datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q5HYK3") #+theme_classic(axis.text.x=element_blank())

```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts[,colnames(norm_counts)=="Q5HYK3"]
datas_gene=as.data.frame(cbind(datas[,c(1,8)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q5HYK3") #+theme_classic(axis.text.x=element_blank())
```



#### Without G4 

A multivariate and univariate cox model is computed to see if the protein is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=table_proteo_sans



modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+Q5HYK3,data=cbind(datas,norm_counts_sans))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~Q5HYK3,data=cbind(datas,norm_counts_sans))
summary(modele_base2)

gene="Q5HYK3"
values=norm_counts_sans[,colnames(norm_counts_sans)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")


```

Here are the Kaplan meier curves with a separation of the patients according to the proteins levels. This allows to see if there is a differences between the patients with high proteins levels and low proteins levels in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the proteins levels for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}



gene=norm_counts_sans[,colnames(norm_counts_sans)=="Q5HYK3"]
datas_gene=as.data.frame(cbind(datas[,c(2,3,5,4)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")



ID2=rep("",length(datas_gene[,2]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()



datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q5HYK3") #+theme_classic(axis.text.x=element_blank())
```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts_sans[,colnames(norm_counts_sans)=="Q5HYK3"]
datas_gene=as.data.frame(cbind(datas[,c(1,8)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q5HYK3") #+theme_classic(axis.text.x=element_blank())
```




### O43639 {.tabset .tabset-fade .tabset-pills}

We can explore the O43639 protein, that is selected with G4 patients (importance `r total_importances[total_importances[,1]=="O43639",2]` ), and with G4 (importance `r total_importances_sans[total_importances_sans[,1]=="O43639",2]` )

#### With G4 

A multivariate and univariate cox model is computed to see if the protein is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=table_proteo_avec


modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+O43639,data=cbind(datas,norm_counts))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~O43639,data=cbind(datas,norm_counts))
summary(modele_base2)

gene="O43639"
values=norm_counts[,colnames(norm_counts)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")



```

Here are the Kaplan meier curves with a separation of the patients according to the proteins levels. This allows to see if there is a differences between the patients with high proteins levels and low proteins levels in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the proteins levels for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}


gene=norm_counts[,colnames(norm_counts)=="O43639"]
datas_gene=as.data.frame(cbind(datas[,c(2,3,5,4)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")



ID2=rep("",length(datas_gene[,2]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()



datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for O43639") #+theme_classic(axis.text.x=element_blank())

```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts[,colnames(norm_counts)=="O43639"]
datas_gene=as.data.frame(cbind(datas[,c(1,8)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for O43639") #+theme_classic(axis.text.x=element_blank())
```



#### Without G4 

A multivariate and univariate cox model is computed to see if the protein is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=table_proteo_sans



modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+O43639,data=cbind(datas,norm_counts_sans))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~O43639,data=cbind(datas,norm_counts_sans))
summary(modele_base2)

gene="O43639"
values=norm_counts_sans[,colnames(norm_counts_sans)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")


```

Here are the Kaplan meier curves with a separation of the patients according to the proteins levels. This allows to see if there is a differences between the patients with high proteins levels and low proteins levels in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the proteins levels for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}



gene=norm_counts_sans[,colnames(norm_counts_sans)=="O43639"]
datas_gene=as.data.frame(cbind(datas[,c(2,3,5,4)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")



ID2=rep("",length(datas_gene[,2]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()



datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for O43639") #+theme_classic(axis.text.x=element_blank())
```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts_sans[,colnames(norm_counts_sans)=="O43639"]
datas_gene=as.data.frame(cbind(datas[,c(1,8)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for O43639") #+theme_classic(axis.text.x=element_blank())
```




### Q8N608 {.tabset .tabset-fade .tabset-pills}

We can explore the Q8N608 protein, that is selected with G4 patients (importance `r total_importances[total_importances[,1]=="Q8N608",2]` ), and with G4 (importance `r total_importances_sans[total_importances_sans[,1]=="Q8N608",2]` )

#### With G4 

A multivariate and univariate cox model is computed to see if the protein is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=table_proteo_avec


modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+Q8N608,data=cbind(datas,norm_counts))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~Q8N608,data=cbind(datas,norm_counts))
summary(modele_base2)

gene="Q8N608"
values=norm_counts[,colnames(norm_counts)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")



```

Here are the Kaplan meier curves with a separation of the patients according to the proteins levels. This allows to see if there is a differences between the patients with high proteins levels and low proteins levels in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the proteins levels for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}


gene=norm_counts[,colnames(norm_counts)=="Q8N608"]
datas_gene=as.data.frame(cbind(datas[,c(2,3,5,4)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")



ID2=rep("",length(datas_gene[,2]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()



datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q8N608") #+theme_classic(axis.text.x=element_blank())

```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts[,colnames(norm_counts)=="Q8N608"]
datas_gene=as.data.frame(cbind(datas[,c(1,8)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q8N608") #+theme_classic(axis.text.x=element_blank())
```



#### Without G4 

A multivariate and univariate cox model is computed to see if the protein is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=table_proteo_sans



modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+Q8N608,data=cbind(datas,norm_counts_sans))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~Q8N608,data=cbind(datas,norm_counts_sans))
summary(modele_base2)

gene="Q8N608"
values=norm_counts_sans[,colnames(norm_counts_sans)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")


```

Here are the Kaplan meier curves with a separation of the patients according to the proteins levels. This allows to see if there is a differences between the patients with high proteins levels and low proteins levels in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the proteins levels for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}



gene=norm_counts_sans[,colnames(norm_counts_sans)=="Q8N608"]
datas_gene=as.data.frame(cbind(datas[,c(2,3,5,4)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")



ID2=rep("",length(datas_gene[,2]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()



datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q8N608") #+theme_classic(axis.text.x=element_blank())
```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts_sans[,colnames(norm_counts_sans)=="Q8N608"]
datas_gene=as.data.frame(cbind(datas[,c(1,8)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q8N608") #+theme_classic(axis.text.x=element_blank())
```




### Q13126 {.tabset .tabset-fade .tabset-pills}

We can explore the Q13126 protein, that is selected with G4 patients (importance `r total_importances[total_importances[,1]=="Q13126",2]` ), and with G4 (importance `r total_importances_sans[total_importances_sans[,1]=="Q13126",2]` )

#### With G4 

A multivariate and univariate cox model is computed to see if the protein is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=table_proteo_avec


modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+Q13126,data=cbind(datas,norm_counts))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~Q13126,data=cbind(datas,norm_counts))
summary(modele_base2)

gene="Q13126"
values=norm_counts[,colnames(norm_counts)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")



```

Here are the Kaplan meier curves with a separation of the patients according to the proteins levels. This allows to see if there is a differences between the patients with high proteins levels and low proteins levels in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the proteins levels for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}


gene=norm_counts[,colnames(norm_counts)=="Q13126"]
datas_gene=as.data.frame(cbind(datas[,c(2,3,5,4)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")



ID2=rep("",length(datas_gene[,2]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()



datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q13126") #+theme_classic(axis.text.x=element_blank())

```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts[,colnames(norm_counts)=="Q13126"]
datas_gene=as.data.frame(cbind(datas[,c(1,8)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q13126") #+theme_classic(axis.text.x=element_blank())
```



#### Without G4 

A multivariate and univariate cox model is computed to see if the protein is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=table_proteo_sans



modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+Q13126,data=cbind(datas,norm_counts_sans))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~Q13126,data=cbind(datas,norm_counts_sans))
summary(modele_base2)

gene="Q13126"
values=norm_counts_sans[,colnames(norm_counts_sans)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")


```

Here are the Kaplan meier curves with a separation of the patients according to the proteins levels. This allows to see if there is a differences between the patients with high proteins levels and low proteins levels in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the proteins levels for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}



gene=norm_counts_sans[,colnames(norm_counts_sans)=="Q13126"]
datas_gene=as.data.frame(cbind(datas[,c(2,3,5,4)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")



ID2=rep("",length(datas_gene[,2]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()



datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q13126") #+theme_classic(axis.text.x=element_blank())
```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts_sans[,colnames(norm_counts_sans)=="Q13126"]
datas_gene=as.data.frame(cbind(datas[,c(1,8)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q13126") #+theme_classic(axis.text.x=element_blank())
```




### Q9HA77 {.tabset .tabset-fade .tabset-pills}

We can explore the Q9HA77 protein, that is selected with G4 patients (importance `r total_importances[total_importances[,1]=="Q9HA77",2]` ), and with G4 (importance `r total_importances_sans[total_importances_sans[,1]=="Q9HA77",2]` )

#### With G4 

A multivariate and univariate cox model is computed to see if the protein is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=table_proteo_avec


modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+Q9HA77,data=cbind(datas,norm_counts))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~Q9HA77,data=cbind(datas,norm_counts))
summary(modele_base2)

gene="Q9HA77"
values=norm_counts[,colnames(norm_counts)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")



```

Here are the Kaplan meier curves with a separation of the patients according to the proteins levels. This allows to see if there is a differences between the patients with high proteins levels and low proteins levels in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the proteins levels for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}


gene=norm_counts[,colnames(norm_counts)=="Q9HA77"]
datas_gene=as.data.frame(cbind(datas[,c(2,3,5,4)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")



ID2=rep("",length(datas_gene[,2]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()



datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q9HA77") #+theme_classic(axis.text.x=element_blank())

```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts[,colnames(norm_counts)=="Q9HA77"]
datas_gene=as.data.frame(cbind(datas[,c(1,8)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q9HA77") #+theme_classic(axis.text.x=element_blank())
```



#### Without G4 

A multivariate and univariate cox model is computed to see if the protein is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=table_proteo_sans



modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+Q9HA77,data=cbind(datas,norm_counts_sans))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~Q9HA77,data=cbind(datas,norm_counts_sans))
summary(modele_base2)

gene="Q9HA77"
values=norm_counts_sans[,colnames(norm_counts_sans)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")


```

Here are the Kaplan meier curves with a separation of the patients according to the proteins levels. This allows to see if there is a differences between the patients with high proteins levels and low proteins levels in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the proteins levels for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}



gene=norm_counts_sans[,colnames(norm_counts_sans)=="Q9HA77"]
datas_gene=as.data.frame(cbind(datas[,c(2,3,5,4)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")



ID2=rep("",length(datas_gene[,2]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()



datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q9HA77") #+theme_classic(axis.text.x=element_blank())
```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts_sans[,colnames(norm_counts_sans)=="Q9HA77"]
datas_gene=as.data.frame(cbind(datas[,c(1,8)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q9HA77") #+theme_classic(axis.text.x=element_blank())
```




### Q9NWS8 {.tabset .tabset-fade .tabset-pills}

We can explore the Q9NWS8 protein, that is selected with G4 patients (importance `r total_importances[total_importances[,1]=="Q9NWS8",2]` ), and with G4 (importance `r total_importances_sans[total_importances_sans[,1]=="Q9NWS8",2]` )

#### With G4 

A multivariate and univariate cox model is computed to see if the protein is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=table_proteo_avec


modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+Q9NWS8,data=cbind(datas,norm_counts))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~Q9NWS8,data=cbind(datas,norm_counts))
summary(modele_base2)

gene="Q9NWS8"
values=norm_counts[,colnames(norm_counts)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")



```

Here are the Kaplan meier curves with a separation of the patients according to the proteins levels. This allows to see if there is a differences between the patients with high proteins levels and low proteins levels in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the proteins levels for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}


gene=norm_counts[,colnames(norm_counts)=="Q9NWS8"]
datas_gene=as.data.frame(cbind(datas[,c(2,3,5,4)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")



ID2=rep("",length(datas_gene[,2]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()



datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q9NWS8") #+theme_classic(axis.text.x=element_blank())

```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts[,colnames(norm_counts)=="Q9NWS8"]
datas_gene=as.data.frame(cbind(datas[,c(1,8)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q9NWS8") #+theme_classic(axis.text.x=element_blank())
```



#### Without G4 

A multivariate and univariate cox model is computed to see if the protein is significant or not: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="Kaplan Meier curve of the patients among levels superior or inferior to the median"}
datas=table_proteo_sans



modele_base= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ cold_diffuse+Sexe+Age+oligo_astro+Q9NWS8,data=cbind(datas,norm_counts_sans))
summary(modele_base)


modele_base2= coxph(formula = Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~Q9NWS8,data=cbind(datas,norm_counts_sans))
summary(modele_base2)

gene="Q9NWS8"
values=norm_counts_sans[,colnames(norm_counts_sans)==gene]
levels=numeric(length(values))
legend=numeric(length(values))

for (j in 1:length(values)){
  if (values[j]<=summary(values)[3]){
    legend[j]="<=Median"
    levels[j]=0
  }
  if (values[j]>summary(values)[3]){
    legend[j]=">Median"
    levels[j]=1
  }
}
legend_total=c("<=Median",">Median")
legend_total=legend_total[legend_total%in%legend]


datas_gene=cbind(datas,levels)
fit <- survfit(Surv(time=as.numeric(Time), event=as.numeric(Reccurence)) ~ levels, data = datas_gene)
ggsurvplot(fit, legend.title = gene,
           legend.labs =c(legend_total),palette = c("#2D708EFF","#FDE725FF"))+xlab("Time (years)")


```

Here are the Kaplan meier curves with a separation of the patients according to the proteins levels. This allows to see if there is a differences between the patients with high proteins levels and low proteins levels in term of risk of recurrence. 

We observe then the differences between the cold and diffuse expression, with first a boxplot of the proteins levels for each status, and then a scatter plot of the expression of each patients: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="cold vs diffuse expression"}



gene=norm_counts_sans[,colnames(norm_counts_sans)=="Q9NWS8"]
datas_gene=as.data.frame(cbind(datas[,c(2,3,5,4)],gene))
colnames(datas_gene)=c("ID","Cold_diffuse","Time","Event","counts")



ID2=rep("",length(datas_gene[,2]))
for (i in 1:length(ID2)){
  if (datas_gene[i,5]>quantile(datas_gene[,5],0.98)|(datas_gene[i,5]<quantile(datas_gene[,5],0.02))){
    ID2[i]=paste(datas_gene[i,1],as.character(Surv(time = round(datas_gene[i,3],2),event=datas_gene[i,4])))
  }else{
    ID2[i]=""
  }
}
datas_gene=as.data.frame(cbind(datas_gene,ID2))

ggplot(datas_gene, aes(x=Cold_diffuse, y=counts,fill=Cold_diffuse)) + scale_fill_manual(values= c("#E7B800", "#2E9FDF")) +
  geom_boxplot()+theme_classic()



datas_cold=datas_gene[order(datas_gene[,2]),]
datas_cold[,1]=factor(datas_cold[,1],levels=datas_cold[,1])
ggplot(datas_cold, aes(x=ID, y=counts,color=Cold_diffuse,label=ID2)) + scale_color_manual(values= c("#E7B800", "#2E9FDF")) +  geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q9NWS8") #+theme_classic(axis.text.x=element_blank())
```

We can do the same for the oligo/astro status: 

```{r,echo=F,message=FALSE,warning=FALSE,cache=FALSE,fig.cap="astro vs oligo expression"}

gene=norm_counts_sans[,colnames(norm_counts_sans)=="Q9NWS8"]
datas_gene=as.data.frame(cbind(datas[,c(1,8)],gene))
colnames(datas_gene)=c("ID","Astro_Oligo","counts")
ggplot(datas_gene, aes(x=Astro_Oligo, y=counts,fill=Astro_Oligo)) +
  geom_boxplot()+theme_classic()

datas_gene=as.data.frame(cbind(datas_gene,ID2))
datas_astro=datas_gene[order(datas_gene[,2]),]
datas_astro[,1]=factor(datas_astro[,1],levels=datas_astro[,1])
ggplot(datas_astro, aes(x=ID, y=counts,color=Astro_Oligo,label=ID2))  + geom_text_repel() +
  geom_point()+theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+xlab("patients")+ylab("counts value for Q9NWS8") #+theme_classic(axis.text.x=element_blank())
```



# Summary 

We have a total of 163 patients with lower grade glioma, we have RNAseq data on 59 patients, proteomics data on 98 patients (minus 2 patients that were discarded as outliers) and imaging data on 158 patients. We have also 5 patients that are "G4", wich might be a different grade of tumor.  
We are interested in finding the genes and proteins associated with the status cold/diffuse, astro/oligo and the risk of recurrence. 

In the total data, we could see that the status cold/diffuse was associated with survival (significant with a multivariate cox model), as well as with the proteomics data, but not the RNA data. The astro/oligo status was not signiciant. These results were not changed by the absence of G4 patients.

## RNA data

We have a total of 59 patients, with 24 cold patients and 35 diffuse, 24 astro and 35 oligo patients. We looked at the difference of gene expression between cold and diffuse, astro and oligo: 

* Cold vs Diffuse: dearseq found 1212 differentially expressed genes (1978 without G4 patients)
* Astro vs Oligo: dearseq found 117 differentially expressed genes (119 without G4 patients)

Even if the number of significant genes is not the same, in term of pvalues, the absence or presence of G4 did not change the results a lot.  


In term of survival, we performed multiple times the random forest, and selected: 

* The gene SLC17A8, always first in term of importance with the total patients, and also associated with cold/diffuse
* The genes CABIN1, SLC22A11, RUNX3, RPS23P1 and RPS2P36 that were found in all random forest, with or without G4 patients

The genes SLC17A8 and RUNX3 are negatively correlated.

## Proteomics data

We have a total of 96 patients because the patiens TOR-ROB and BAL-PIE were discarded as they were outliers, with 37 cold patients and 59 diffuse, 37 astro and 56 oligo patients. We looked at the difference of proteins expression between cold and diffuse, astro and oligo: 

* Cold vs Diffuse: dearseq found 14 differentially expressed proteins (222 without G4 patients)
* Astro vs Oligo: dearseq found no proteins

Even if the number of significant proteins is not the same, in term of pvalues, the absence or presence of G4 did not change the results a lot.  


In term of survival, we performed multiple times the random forest, and selected the proteins Q5HYK3, O43639, Q8N608, Q13126, Q9HA77, Q9NWS8, O14734, Q96BJ3, P30048, Q9Y5A9, Q9NQC3, Q9H0V9, Q9H479, Q9Y5L0, O95825, O15173


